<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>NRNB: Projects</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="">
<meta name="author" content="">

<!-- Le styles -->
<link href="assets/css/bootstrap.css" rel="stylesheet">
<script type="text/javascript"
	src="http://ajax.googleapis.com/ajax/libs/jquery/1.7/jquery.js"></script>
<script>
$(function () {
	   var activeTab = $('[href=' + location.hash + ']');
	   activeTab && activeTab.tab('show');
	   window.scrollTo(0, 0);
	});
</script>

<style type="text/css">
body {
	padding-top: 60px;
	padding-bottom: 40px;
}
</style>
<link href="assets/css/bootstrap-responsive.css" rel="stylesheet">


<!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
<!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

<!-- Fav and touch icons -->
<link rel="shortcut icon" href="assets/ico/favicon.ico">
<link rel="apple-touch-icon-precomposed" sizes="144x144"
	href="assets/ico/apple-touch-icon-144-precomposed.png">
<link rel="apple-touch-icon-precomposed" sizes="114x114"
	href="assets/ico/apple-touch-icon-114-precomposed.png">
<link rel="apple-touch-icon-precomposed" sizes="72x72"
	href="assets/ico/apple-touch-icon-72-precomposed.png">
<link rel="apple-touch-icon-precomposed"
	href="assets/ico/apple-touch-icon-57-precomposed.png">
</head>

<body>

	<div class="navbar navbar-inverse navbar-fixed-top">
		<div class="navbar-inner">
			<div class="container">
				<a class="brand" href="index.html"><img
					src="images/NRNB_white_22.png" alt="image" /></a>
				<ul class="nav">
					<li><a href="index.html">Home</a></li>
					<li><a href="outreach.html">Collaborate</a></li>
					<li><a href="training.html">Learn</a></li>
					<li><a href="tools.html">Gallery</a></li>
					<li class="active"><a href="#">Projects</a></li>
					<li><a href="people.html">People</a></li>
				</ul>

			</div>
		</div>
	</div>

	<div class="container">
		<H3>Technology Research and Development</H3>
		<a class="btn btn-primary pull-right"
			href="presentations.html#nrnbannual">Annual Reports &raquo;</a>

		<p>We are conducting an aggressive development program centered on
			Technology and Research Development projects (TRDs) that will extend
			the Cytoscape infrastructure to make it significantly more useful to
			researchers.</p>

		<ul class="nav nav-tabs" id="tabs">
			<li class="active"><a href="#a-tab" data-toggle="tab">TRD-A</a></li>
			<li><a href="#b-tab" data-toggle="tab">TRD-B</a></li>
			<li><a href="#c-tab" data-toggle="tab">TRD-C</a></li>
			<li><a href="#d-tab" data-toggle="tab">TRD-D</a></li>
		</ul>
		<div class="tab-content">
			<a class="btn btn-info btn-mini" href="project-tracking.html">Submit
				Project Update</a>
			<div class="tab-pane active" id="a-tab">
				<h3>Project A: Identification of network-based biomarkers for
					molecular classification</h3>
				<h4>
					Project Leaders: <a href="people.html#sander">Chris Sander</a>
					(MSKCC) and <a href="people.html#trey">Trey Ideker</a> (UCSD)
				</h4>

				<div class="well">
					<h4>A Gene Ontology Extracted from Molecular Networks (Trey Ideker, UCSD):</h4>
					<p>Ontologies are of key importance to many domains of biological research. 
					The Gene Ontology (GO), in particular, has been instrumental in unifying knowledge about biological processes, 
					cellular components, and molecular functions through a hierarchy of concepts and their 
					interrelationships. However, given only partial biological knowledge and inconsistency in how 
					this knowledge is curated, it has been difficult to construct, extend and validate GO in an 
					unbiased manner. We have recently showed that the existing collection of high-throughput 
					network maps, as are now becoming available, can be analyzed to automatically assemble an 
					ontology of gene function that rivals manually curated efforts [Dutkowski et al. Nat Biotechnol, 
					2013 2013. 31(1): p. 38-45]. Our systematic computational approach (Fig. 1) combines evidence from 
					physical, genetic and transcriptional networks to produce an ontology comprised of 4,123 biological 
					concepts and 5,766 hierarchical concept relations (Fig. 2). Using a new ontology alignment procedure 
					(Fig. 1), we found that the network-based ontology captures the majority of known cellular components 
					and identifies approximately 600 new cellular components and component relations – many of which we 
					were able to validate either experimentally or bioinformatically. By working closely with the GO curators, 
					we were able to incorporate selected new components and relations into the Gene Ontology, thus providing 
					proof-of-principle for how to systematically update and revise the GO structure based on large-scale data. 
					</p>
					<p>The network-extracted ontology is a new resource for systems and synthetic biology – i.e. a data-driven 
					catalogue of cellular machinery, from genes, to complexes, to pathways and higher-order processes. It 
					provides a powerful tool for performing multi-scale analysis of biological networks, including automatically 
					identifying, annotating and visualizing the complete hierarchical structure. We also show how integrating 
					the ontology with additional high-throughput datasets leads to identification of new components and 
					processes altered in human disease. Based on our results, we suggest a new role for ontologies in 
					bioinformatics: rather than merely being used as a gold-standard for performing functional enrichment, 
					ontologies should serve as evolvable models that are validated, revised, and expanded based on new genomic data.</p>
					<p>Moving forward, it will be interesting to see how the network-extracted ontology can further be extended. 
					For instance while NeXO represents a rigorous approach to capture ontology terms and term relations, the 
					ability to systematically annotate the type of relation that occurs between terms (e.g. “is a”, “part of”, 
					“regulates”) poses a separate and very interesting challenge. An in-depth investigation is needed to assess 
					which network properties are best at separating the different types of relations, and whether there are 
					additional data sets that might be brought to bear on this problem (Stage 3). Similarly, while NeXO 
					identifies the majority of known cellular components, it will be interesting to further investigate 
					what types of network data could be used to increase the coverage of biological processes and molecular 
					functions. Finally, a key question is whether enough high-quality data exist to build NeXO ontologies for 
					other species, particularly human, and, whether it is better to structure a common ontology for all species, 
					as has been done in GO, or to focus on individual species-specific ontologies.
					
					<img src="images/IdekerF1.png" width="700" /><br /> 
					<b>Figure 1: Automated assembly and alignment of gene ontologies.</b> (A) Probabilistic community detection within 
					the input networks yields a binary tree in which nodes correspond to ontology terms and links correspond to 
					parent-child term relations. Unsupported terms are replaced by multi-way joins, and additional parent-child 
					relations are added based on network data. The resulting ontology is aligned against the Gene Ontology, 
					in a way that (B) prohibits non-unique mappings and ancestor-descendant criss-crossing.
					
					<img src="images/IdekerF2.png" width="700" /><br /> <b>Figure 2:</b>
					The NeXO ontology is shown as a tree, with nodes indicating terms and edges indicating hierarchical 
					relations between terms, i.e. that one term contains another. Node sizes indicate the number of genes assigned 
					to a term. Node colors represent the degree of correspon-dence to a term in GO as determined by ontology alignment, 
					with high-level alignments labeled. Insets show the hierarchy identified for the ribosome and actin cytoskeleton.
					</p>
					<h4>Network Analysis Tools for Cancer Genomics (Chris Sander, MSKCC):</h4>
					<p>This project is focused on building 
					network analysis tools for interpreting high-throughput cancer genomic data sets to identify altered disease 
					networks and enable the identification of network-based biomarkers in cancer. Our primary focus is building 
					user-friendly, open source tools for visualizing and analyzing multidimensional cancer genomic data sets 
					(including copy number, mutation, and mRNA expression) in the context of known biological pathways and 
					interaction networks, and making these tools broadly available to clinical, experimental and computational 
					investigators within the cancer research community. Providing such tools to the cancer research community 
					is critical, as numerous large-scale projects, including the Cancer Genome Atlas (TCGA) project and the 
					International Cancer Genome Consortium (ICGC), are profiling dozens of cancer types and subtypes. Identifying 
					altered pathways and networks within each of these cancer types remains a critical and open challenge.</p>
					
					<p>During our first several years of NRNB funding, we completed a prototype project for displaying 
					multi-dimensional cancer genomic data in the context of molecular interaction networks. We chose to 
					implement the prototype in CytoscapeWeb [1], as CytoscapeWeb does not require any additional software 
					installation or require Java Web Start. It therefore significantly lowers the barriers for usage, 
					particularly for biologists and clinical researchers ­­­­- two of our main target user groups.
					We transitioned our tools from prototype to production mode, and have made our software available 
					to the entire cancer research community. Cancer researchers are now using these tools to perform 
					network analysis on up to 20 different cancer types, including TCGA-funded projects, such as glioblastoma 
					multiforme (GBM) [2] and serous ovarian cancer [3].</p>
					
					<p>The cBioPortal for Cancer Genomics code base has recently reached a stable state where it is now being 
					considered as a general framework to build our other NRNB-related tools on. Our recently finalized drug-target 
					data support in the context of cBioPortal’s network analysis is one such example. During the past year, 
					we improved the network analysis capabilities of the cBioPortal by providing query and visualization of 
					aggregated drug data from multiple resources. With this new feature, the portal currently contains 
					gene-centric drug-target information from the following resources: DrugBank [8], KEGG Drug [9], 
					<a href="http://www.cancer.gov/cancertopics/druginfo/alphalist">NCI Cancer Drugs</a>, and Rask-Andersen et al. [10]. 
					Within the network analysis view, drugs are hidden by default, but can be added to the network via the 
					Genes & Drugs menu on the right side of the screen. Users now have the option of displaying FDA-approved 
					drugs, cancer drugs defined by NCI Cancer Drugs, or all drugs targeting the query genes. For example, 
					when the user queries for the gene EGFR in the portal, we not only show the network context of this gene, 
					but also provide information about the drugs targeting the product of this gene: geﬁtinib and erlotinib 
					are tyrosine kinase inhibitors that target the catalytic domain of EGFR, and cetuximab and trastuzumab 
					are monoclonal antibodies that target the extracellular domain of EGFR and ERBB2, respectively (Fig. 3) [11].
					
					<img src="images/SanderF1.png" width="700" /><br /> 
					<b>Figure 3: Improved Network tab: Network analysis of epidermal growth factor receptor networks 
					in serous ovarian cancer.</b> (A) Network view of the EGFR and ERBB2 neighborhood in serous ovarian cancer 
					(TCGA data set) rendered by Cytoscape Web. EGFR and ERBB2 are query genes (thick border), and nearest 
					neighbor genes are color coded by their alteration frequency in ovarian cancer. One can display drugs 
					that target EGFR or ERBB2 (hexagons, orange if FDA approved), as well as details about genomic alterations 
					and links to external resources (lower left panel, example MYC). (B) The portal overlays multidimensional 
					genomic data (copy number, mutation, and mRNA expression) onto all nodes in the network. (C) Edges can 
					represent different interaction types (color-coded, such as “reacts with”). (D) Options for filtering, 
					cropping and searching the network of interest.
					<br /><br />
					Our new drug-target feature is now available as part of the open-access cBioPortal and is helping 
					cancer researchers in exploring the therapy options within the network context of genes of interest. </p>
					
					<p><b>New Driving Biological Projects:</b> In the next year, we are anticipating improving the network 
					analysis feature in two ways: 1) detailed visualization of the pathways and reactions in the network view; 
					2) inference of indirect drug targets, for potentially interesting therapy options, by using genomic 
					alteration and drug-target data. Currently, interaction types that are shown in the network analysis 
					view are derived from the BioPAX to SIF inference rules [7]. For example: In Same Component indicates 
					that Genes A and B are involved in the same biological component, such as a complex; State Change 
					indicates that Gene A causes a state change, such as a phosphorylation change within Gene B. This 
					reduction from BioPAX to SIF was necessary as the Cytoscape Web framework, by then, was not 
					supporting visualization of more complex elements, such as compartments. With the technology being 
					developed as part of the CSP-100 project (Gary Bader), it recently become feasible to visualize 
					biological networks in a more detailed way, therefore enabling the use of Systems Biology Graphical 
					Notation (SBGN) for better representation of BioPAX. As part of our NRNB collaboration with Ugur 
					Dogrusoz (Bilkent University, Turkey), we are aiming to adopt SBGN-complaint views to visualize 
					multi-dimensional cancer genomics data with the network context (see Figure 4). This project has 
					recently been implemented as a proof-of-concept prototype and is now being integrated into cBioPortal. 
					When complete, this new feature will allow better presentation of proteomics data (e.g. Reverse Phase 
					Protein Array data provided as part of the TCGA network) by allowing users to optionally switch from 
					a gene-centric to protein-centric view.
					
					<img src="images/SanderF2.png" width="700" /> <br /> <b>Figure 4:
					Proposed additions to the current simple network view.</b> On the left is the Simple Interaction 
					derived from BioPAX; on the right is an example visualization of a BioPAX network obtained from Pathway 
					Commons. The latter is utilizing the new visualization capabilities, Systems Biology Graphical Notation 
					(SBGN), of the CytoscapeWeb project. The SBGN view provides a more detailed representation of the 
					pathway and also provides protein-centric view with Proteomics data mapped to specific proteins or 
					phospho-proteins.
					
					<p>In the next year, we are also planning to utilize genomic alteration and pathway data to infer 
					clinically relevant uses of drug-target data. For this, we intend to use down- and up-stream relationships 
					between genes to suggest drugs of possible interest that can indirectly target a particular genomic 
					alteration event in cancer samples (see Figure 4). One historical example for such cases is the use of 
					AKT inhibitors in patients who bear a homozygous PTEN deletion. Without the gene PTEN and its product, 
					Akt proteins, which are down-stream of PTEN, cannot be suppressed, and therefore are found to be 
					upregulated in cancer samples that have the homozygous PTEN deletion. In the presence of an AKT 
					inhibitor, this up-regulation effect can be counteracted. Another similar example of this concept is 
					the use of CDK4/6 inhibitors when CDKN2A is either mutated or homozygously deleted in cancer cells. 
					Pathway resources, such as Pathway Commons, already provide this type of relationships between genes; 
					and we plan to extract this information in a systematic way and combine it with the drug-target data 
					in order to infer such therapy options in an automatic manner within the cBioPortal framework. This 
					method and the prototype are currently under development (Stage 2).
					<img src="images/SanderF3.png" width="700" /> <br /> <b>Figure 5:</b>
					Conceptual framework for inferring novel and drug-based therapy options based on specific genomic 
					alteration with the use of pathway context -- e.g. use of AKT inhibitors when PTEN is altered in the tumor.
					</p>
					
					<h4>Network Analysis Methods for Inferring Causality in Signaling Networks (Sander, MSKCC)</h4>
					<p>The goal of our second TRD project is to develop network analysis tools that algorithmically infer 
					causality within signaling networks and make these tools available. High-throughput screens conducted 
					with libraries of small molecules or inhibitory RNAs have the ability to identify compounds that induce 
					tumor suppressive responses in cancer cells [12]. While the effects of such perturbations can be easily 
					linked to transcriptional changes, identifying the causal mechanism is a main challenge. In collaboration 
					with Somwar and colleagues [13], we used a computational approach to predict the target of a small molecule 
					inducing reduced growth in lung adenocarcinoma cell lines. Interestingly, experimental follow up confirmed 
					the prediction.</p>
					<p>Building on this concept, we have been working on computational approaches to model causal signaling 
					cascades inducing observed transcriptional changes within perturbed cancer cell lines. We have been 
					exploring the use of optimization algorithms adapted from statistical physics to identify the minimal 
					set of interactions able to connect genes that are differentially expressed after a perturbation, with 
					candidate targets of the same perturbation (Stage 2). This initial approach relied on an algorithm 
					that solves the Steiner-tree problem. Given a set of “terminal” nodes, the Steiner-tree is defined 
					as the tree of minimum weight connecting these terminals, allowing the inclusion of additional nodes. 
					Differentially expressed genes after a perturbation and/or candidate targets of the same perturbation 
					can be used as terminals. Our prediction was that the resulting Steiner-tree could therefore contain 
					both gene interactions able to explain the observed transcriptional changes and the putative target 
					of the perturbation. Within this past year, we determined that this approach does not work as well 
					as expected, and are now in the process of exploring a new algorithmic framework that combines 
					Gaussian graphical models with maximum entropy methods.</p>
					<p><b>New Driving Biological Projects</b><br />A new biological driver for deriving causality networks 
					is inferring causal relationships within data types and between data types, such as copy number 
					changes and cancer genomics. For example, we would like to investigate the relationship between 
					mutations in the TP53 tumor repressor and the complex copy number profile in ovarian cancer. Another 
					example is the exploration of causal relationships between gene mutations. For example, mutations in 
					the POLE gene lead to a characteristic spectrum of mutations in other proteins. We have preliminary 
					results and plan to develop a network analysis approach to identify causal relationships.
					</p><p>We are also considering looking at interactions between microbial subpopulations, starting with 
					the gut microbiome, where a set of interacting bacterial populations change under fluctuating constraints 
					provided by the host and nutrient intake. Recent work has shown the precise composition and evolution 
					of this population is closely coupled to the state of health of the host. Certain deviations from 
					equilibrium present a significant risk of invasion by pathogenic bacteria, as seen with some cancer 
					patients receiving bone marrow transplantations [23]. A more detailed understanding of the relationships 
					between gut microbial subpopulations following such aggressive treatments in the host could inform 
					therapeutic development leading to improved outcomes.</p>
				</div>
			</div>
			<div class="tab-pane" id="b-tab">
				<h3>Project B: Using Cytoscape for Social Network Research </h3>
				<h4>
					Project Leader: <a href="people.html#fowler">James Fowler</a>
					(UCSD)
				</h4>

				<div class="well">

					<p>
						Social networks are increasingly recognized as having an important
						effect on a wide variety of human behaviors, affecting the spread
						of health outcomes (AIDS, obesity, drinking and smoking behavior),
						emotional states (depression, loneliness, and happiness), and
						political outcomes (legislative activity, judicial decisions, and
						terrorist events). However, the statistical tools used to study
						these effects are drawn from a tradition that focuses on dyads, or
						sets of two people who are socially connected. When we find an
						association in behavior within dyads, we typically cannot
						distinguish three possible explanations. 1) <i>Homophily:</i>
						people might choose to be friends with others because they have
						the same behavior. In other words, birds of a feather might flock
						together. 2) <i>Context:</i> there might be variation in the
						environment than causes both individuals in a dyad to exhibit the
						same behavior. For example, poor neighborhoods might create
						environments that promote less healthy behaviors, causing an
						association between friends who live in the same neighborhood. 3)
						<i>Influence:</i> one person in the dyad actually influences the
						other in some way to cause their behavior to become more like
						their own.
					</p>
					<p>
						We are developing a Network Correlation Cytoscape App in collaboration 
						Dr. Alex Pico. In order to understand why so many behaviors are 
						influenced by social connections, a tool must be developed that is 
						able to determine the variable contribution of a given set of potential 
						causes (such as homophily, context, and influence effects). This 
						technology can then be used to determine the underlying causes behind 
						the spread of various behaviors (health, political, emotional) in networks 
						of different types and across different time points. The emerging techniques 
						and tools will be broadly applicable to the study of social network influence 
						on health behaviors such as eating habits, alcohol consumption, and drug use.
					</p>

					<img src="images/F9a.png" width="320" title="A" /><img
						src="images/F9b.png" width="320" title="B" /><br /> <b>Figure:
						Obesity in the Framingham Heart Study Social Network.</b> Panel (a)
					shows a subcomponent of the whole FHS social network that contained
					2,106 individuals in the year 2000. Node border indicates gender
					(red=female, blue=male), node color indicates obesity
					(yellow=BMI>30), node size is proportional to BMI, and arrow colors
					indicate relationship (purple=friend or spouse, orange=family).
					Created using Pajek.131 Panel (b) shows the estimated difference
					between the probability that an alter is obese given that an ego is
					obese in the observed FHS network and an identical network (with
					topology preserved) in which the same number of obese individuals
					is randomly distributed. Alter distance refers to closest social
					distance between the alter and ego (alter = distance 1, alter's
					alter = distance 2, etc.). Error bars show 95% confidence intervals
					based on 1,000 simulations. The waves correspond to survey waves
					spread out between 1971 and 2003.
				</div>
			</div>
			<div class="tab-pane" id="c-tab">
				<h3>Project C: Visualization and Representation of Biological
					Networks</h3>
				<h4>
					Project Leaders: <a href="people.html#bader">Gary Bader</a> (U
					Toronto) and <a href="people.html#alex">Alex Pico</a>
					(Gladstone/UCSF)
				</h4>

				<div class="well">
					<h4>Cytoscape 3.0 and CytoscapeWeb for the Visualization and Representation 
					of Biological Networks 
					(Bader, U Toronto): </h4>
					<p><b>Technologies developed with NRNB funds:</b><br />
					Our goal is to develop new technologies for visualization and representation 
					of biological networks. 
					Our grant aims are:
					<ul>
					<li><b>Aim 1.</b> Simplifying network views by hierarchically organizing 
					networks and their modules.</li>
					<li><b>Aim 2.</b> Showing only the information needed across multiple levels 
					of detail (semantic zooming) 
					and data sources (information layering)</li>
					<li><b>Aim 3.</b> Simultaneously viewing network attributes collected from 
					many biological experiments. </li>	
					</ul>
					<p>Our major activity over the past year has continued to ensure that Cytoscape 
					3.0 supports the advanced visualization and representation features that we 
					proposed in the NRNB grant, both in system design and performance. This has 
					required major effort working on releasing Cytoscape 3.0 and making sure all 
					of the application programming interfaces (APIs) present in that system will 
					support our aims. Cytoscape 3.0 has now been released and we are focusing on 
					developing new features. Over the past year, we implemented support for multiple
					renderers so app authors can contribute their own visualizations. We helped 
					integrate the Metanodes plugin into Cytoscape 3 as CyGroups in collaboration 
					with Scooter Morris at UCSF, which enables hierarchical views. We developed 
					technology to integrate OpenGL into Cytoscape, which required changes to the 
					code building process. This will support faster new visualizations in the future 
					that take advantage of current graphics card functionality. We also mentored a 
					GSoC project for visualizing dynamic networks (e.g. networks that vary over time).
					</p>
					<p>Excitingly, we have also ported over the CytoscapeWeb visualization tool that 
					we originally developed for the GeneMANIA project to use open web standards 
					such as HTML5. This software is now available at 
					github.com/cytoscape/cytoscape.js. This involved testing various HTML5 canvas 
					and WebGL based libraries for rendering networks within the browser, such as 
					Three.js, KineticJS, EaselJS, performing performance testing using WebGL directly 
					for drawing networks using HTML5 technology, performing performance testing using 
					HTML5 canvas directly and finally implementing a complete HTML5 canvas based 
					rendered that is fully compatible with touch devices, like the Apple iPad. 
					This technology will improve accessibility of Cytoscape visualizations by 
					supporting any computing device that runs a modern web browser (many of which 
					can not run the Cytoscape Java application). As more groups develop web-based 
					network visualization and analysis functionality, we expect extensive innovation 
					in user interfaces and visualization techniques to occur on this platform. We 
					are also continuing to lay the groundwork for representation and visualization 
					of detailed biological pathway information in Cytoscape 3. We have completed 
					the following activities in this area:
					<ul>
					<li>Developed a new CyPath2 Cytoscape 3.0 app, which can load biological 
					pathway data from the Pathway Commons web service in BioPAX level 3 format.</li>
					<li>Updated BioPAX and Pathway Commons core plugins/apps for Cy2 and Cy3</li>
					</ul>
					</p>
					<p>Ensuring Cytoscape 3 will enable our stated aims has required tremendous 
					effort, in that we have need to implement a number of prototype features to 
					test that our API designs are robust. We expect this work will pay off in 
					2013 now that we have finally released Cytoscape 3 and have started working 
					on novel visualization features in earnest.</p>
					<p>We continue to maintain our highly successful Enrichment Map visualization 
					plugin for Cytoscape 2.8, responding to frequent requests by users for new 
					features. This visualization tool is heavily used in all of our collaborations 
					with local biology groups and by others (the two papers describing the method 
					garnered over 90 citations since 2010). We have recently established a 
					collaboration with Jill Mesirov’s group at the Broad Institute, MIT to integrate 
					Enrichment Map into their popular Gene Set Enrichment Analysis (GSEA) software. 
					This work, funded by a separate NIH R01 grant, is now complete and will be 
					released in the coming months.</p>
					<p>Over the past year, we have developed strong ties with a new driving 
					biological project – the 
					<a href="http://oicr.on.ca/oicr-programs-and-platforms/innovation-programs/cancer-stem-cells">
					Cancer Stem Cell program</a> at the Ontario Institute for Cancer Research (OICR) 
					in Toronto, led by John Dick, the discoverer of cancer stem cells. Fifteen labs 
					participate in this program, most of which run genomics experiments and are very 
					interested in pathway analysis to help interpret their results. As such, they 
					have funded two full time research associates in my lab, Veronique Voisin who 
					does pathway analysis using Enrichment Map and Shaheena Bashir, who is a 
					biostatistician who processes raw genomics data to prepare for pathway analysis. 
					This has led to the four published projects in the past year.</p>
					
					<h4>Analyzing Complex Networks Using Ontologies and Cytoscape 3.0 (Alex Pico, 
					Gladstone/UCSF):</h4>
					<p>To make use of the vast wealth of data and knowledge to elucidate the 
					function of biological networks is one of the biggest challenges in 
					bioinformatics. The development of high-throughput technology has given rise 
					to an enormous increase of data on biomolecular expression and interactions, 
					which results in protein interaction networks, gene regulatory networks, 
					signaling networks and metabolic networks. One approach to understanding 
					networks relies on ontologies, such as Gene Ontology, and enrichment analysis. 
					Common approaches, such as BiNGO [1], regard a network as a list of genes, 
					performing gene-level annotation and enrichment methods. However, it is 
					apparent that the same list of genes with different interactions may perform 
					different functions. Thus, the functional analysis of networks should take 
					into consideration network topology.</p>
					<p>Following up on last year's work on the Mosaic app for the partitioning 
					and visualization of complex networks using Gene Ontology [2], we focused on 
					the development of the Network Ontology Analysis (NOA) app for Cytoscape. 
					The NOA app implements the NOA algorithm [3] for network-based enrichment 
					analysis, which extends Gene Ontology annotations to network links, or edges. 
					First, NOA assigns ontology terms to interactions based on the 
					known annotations of connected genes via optimizing two novel indexes 
					‘Coverage’ and ‘Diversity’. Then, NOA generates two alternative reference 
					sets to statistically rank the enriched functional terms for a given 
					biological network. NOA was shown to be more efficient not only for dynamic 
					regulatory networks but also for rewired protein interaction networks. 
					However, there are several shortcomings with the initial implementation: 
					no graphical interface or visualization of results; limited support for 
					species, gene names and ontology types; no batch mode for large-scale 
					computation; and no integration or interoperability with other network 
					tools. To overcome these shortcomings, we have reimplemented the NOA 
					algorithm as a Cytoscape app and added interfaces, extensible ontology and 
					identifier support, new visualizations, a batch mode, and interoperability 
					with other Cytoscape apps, such as CyThesaurus and Mosaic. The NOA 
					app facilitates the annotation and analysis of one or more networks in 
					Cytoscape according to user-defined parameters. In addition to tables, the 
					NOA plugin also presents results in the form of heatmaps and overview 
					networks in Cytoscape, which can be exported for publication figures. A 
					manuscript on this work is currently in review. 
					The publication and release of this tool via the Cytoscape App Store brings 
					the project to Stage 4.</p>
					<p><b>Application</b>
					As a proof-of-principle example, we prepared a file containing all the human 
					pathways curated at WikiPathways and translated Ensembl identifiers for batch
					mode analysis by the NOA plugin. We ran edge-based analysis to make use of 
					the interaction content in the pathways, rather than just treating them like 
					disconnected gene sets. Each pathway was analyzed against the entire 
					collection as reference and no correction was applied. As expected, curated 
					pathways of known function were significantly annotated with relevant GO 
					terms (Fig. 1). Users could add novel pathways or uncharacterized interaction
					networks to this batch analysis to assess functional annotation of edges in 
					the context of known biology.
					<img src="images/PicoF1.png" width="700" /><br /> 
					A heatmap of GO-annotated human pathways based on batch mode NOA plugin 
					analysis. Axes are labeled in inset, highlighting apoptosis-related pathways 
					and GO terms.
					</p>
					<p><b>Cytoscape 3.0</b><br />
					Adding Dr. Scooter Morris to the Gladstone subcontract this year, we not only 
					gained a valuable Roving Engineer for training and outreach, but also a key 
					contributor to Cytoscape 3.0 design and development. During this reporting 
					period, Dr. Morris, worked on porting key plugins to 3.0, that other apps 
					depend on as well as many core features, including context menus, Tunables, 
					and custom graphics. As part of this work, he met with Bader lab members at 
					the University of Toronto to work on improving the user interface for the 
					VizMapper and extending Metanodes. He also improved the standardization of 
					the Cytoscape context menus and the esthetics of Cytoscape dialogs by 
					enhancing the Tunables interface. The focus of the Cytoscape 3 effort during 
					the later half of this reporting period has been the 3.0.1 release. 
					This is primarily a "bug-fix" release that follows on the heels of the 
					initial release of Cytoscape 3, and is intended to be the "solid" release we 
					want to advocate for new users. In addition to several general fixes, 
					Dr. Morris' work has addressed several issues with graphical annotations and 
					groups that are important for wider use.  We also began the design for an API 
					to export the functionality of graphical annotations to other apps, in 
					particular apps such as the GPML App, which will use these features to render
					pathway diagrams imported from the WikiPathways resource, which is another 
					NRNB-supported service.</p>
				</div>
			</div>
			<div class="tab-pane" id="d-tab">
				<h3>Project D: Predictive Modeling and Network Inference</h3>
				<h4>
					Project Leaders: <a href="people.html#benno">Benno Schwikowski</a>
					(Institut Pasteur)
				</h4>

				<div class="well">
					<p>The objectives of this TRD are to (1) Create 'fill-in-the-algorithm' infrastructure 
					for network inference and (2) Make methods accessible to biologists within the Cytoscape 
					framework. Our approach is to create a software infrastructure (CYNI) for network inference 
					algorithms.</p>

					<p>The Cyni Toolbox app is available from the Cytoscape App Store (link) and is also available 
					here (http://www.proteomics.fr/Sysbio/CyniProject). Data imputation and discretization techniques 
					are provided along with several known inference algorithms to make this tool fully operational for 
					any kind of inference requirement. While data imputation and discretization techniques allow you 
					modify Cytoscape tables, network inference algorithms produce a new network after applying the chosen technique.
				
					<p>Cyni Toolbox is not only an application to perform a result, but it also provides a framework 
					for other app developers to facilitate the implementation of other algorithms. The goal of the 
					Cyni framework is to provide standardized, extensible and configurable default solutions for all 
					components different from the core algorithms, such as GUI, parameter handling, and configuration 
					of distance/similarity measures. Thereby, developers can focus on their core expertise. Tutorials 
					are available for Cyni Toolbox users as well as for developing new algorithms through Cyni.</p>


					<img src="images/F20.png" width="700" /><br /> <b>Figure:</b>
					The CYNI framework extends Cytoscape to support an iterative
					computational-experimental cycle for network-based modeling.
				</div>
			</div>
			<!-- end tab-pane -->

		</div>
		<!-- end tab-content -->

		<hr>

		<footer>
			<p>
				The National Resource for Network Biology is an NIH Division of
				Biomedical Technology, Bioinformatics, and Computational Biology (<a
					href="http://www.nigms.nih.gov/About/Overview/bbcb.htm"><b>BBCB</b></a>)
				Biomedical Technology Research Center (<a
					href="http://www.nigms.nih.gov/About/Overview/BBCB/BiomedicalTechnology/BiomedicalTechnologyResearchCenters.htm"><b>BTRC</b></a>).
				NRNB is funded by grant GM103504.
			</p>
			<p class="pull-right">&copy; NRNB, 2014</p>
			<p>
				Contact: <a href="mailto:apico@gladstone.ucsf.edu">email us</a>
			</p>
		</footer>

	</div>
	<!-- /container -->



	<script type="text/javascript">
		var _gaq = _gaq || [];
		_gaq.push([ '_setAccount', 'UA-19386719-1' ]);
		_gaq.push([ '_trackPageview' ]);

		(function() {
			var ga = document.createElement('script');
			ga.type = 'text/javascript';
			ga.async = true;
			ga.src = ('https:' == document.location.protocol ? 'https://ssl'
					: 'http://www')
					+ '.google-analytics.com/ga.js';
			var s = document.getElementsByTagName('script')[0];
			s.parentNode.insertBefore(ga, s);
		})();
	</script>
	<script src="assets/js/bootstrap.min.js"></script>
</body>
</html>

