<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>NRNB: GSoc Ideas Page</title>

<!-- Le styles -->
<link href="../assets/css/bootstrap.css" rel="stylesheet">
<link href="../assets/css/bootstrap-responsive.css" rel="stylesheet">
<link type="text/css" rel="stylesheet" href="../assets/css/jquery.tocify.css" />
<link type="text/css" rel="stylesheet" href="../assets/css/jquery-ui.min.css" />
<style type="text/css">
#nav {
     position: fixed;
}
.tocify ul, .tocify li {
        line-height: 15px;
}
</style>
</head>
<body>
<p>The updated NRNB GSoC project ideas list has been moved and is now at <a href="https://github.com/nrnb/GoogleSummerOfCode/issues">GitHub</a>.</p>
<div class="row" height="100%">
    <div class="span4"  height="100%">
         <!-- Our table of contents will be here !-->
         <div id="nav" style="font-size: 12px;" height="100%"></div>
    </div>
    <div id="ideas" class="span8">

<!-- PASTE HERE -->

<p><br />
</p>

<h2 dir="ltr" >Idea 1: Submit your own original proposal</h2>

<p>Feel free to propose your own idea. As long as it relates to one of our projects, we will give it serious consideration. 
Creativity and self-motivation are great traits for open source programmers, but make sure your proposal is also relevant. 
Projects can be on any network biology related software idea. Consider some of the <a href="http://nrnb.org/tools.html" target="_blank">NRNB coordinated technologies</a>.</p>
<br />

<h2 dir="ltr" >Idea 2: Co-authorship Social network App</h2>
<p><b>Background</b>: Biological networks can be visualized and analyzed using Cytoscape. Often researchers want to go beyond 
the network of proteins or genes and also look at the inter-connectedness between colleagues and institutions. Who tends to 
publish together? What institutions are most collaborative? Are there inter-disciplinary connections in my institution?</p>
<p><b>Goal</b>: Create a visual summary of how individuals are connected. An existing App currently displays such a network, 
where nodes are authors and edges the number of publications two authors share, using publication information from querying 
PubMed over the web or from a local file downloaded from Scopus or Incites databases. Different data sources have different 
depths of data, for instance PubMed, the most widely accessible of the three main sources, has citation counts but they are 
limited to only those articles cited in PubMed Central and its institution data is limited to the first author. Given that 
PubMed is the easiest to access source, we would like to expand its functionality to capture more data about author 
institution and publication citation information, as well as H-index calculation for individuals or subsets of papers.</p>
<p>We would like to expand our co-authorship Cytoscape App to include generic file formats to easily create networks from any 
new data source. As well as expanding publication network visualization and summarization by integrating with other Cytoscape 
apps such as WordCloud to create customized node images representing the tag cloud of all the author’s publications, or 
adjusting node size to account for impact factors of the author’s publications.</p>
<p><b>Links</b>:</p>
<ul>
<li><a href="https://github.com/BaderLab/SocialNetworkApp" target="_blank">Existing code</a></li>
<li><a href="http://baderlab.org/UserguideSocialNetworkApp" target="_blank">Existing user guide</a></li>
</ul>
<p><b>Technology and Skills</b>: Cytoscape, Java, XML, PubMed e-utils</p>
<p><b>Potential Mentors</b>: Sergio Baranzini, Gary Bader, Ruth Isserlin</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 3: Upgrading Ontology Tagging at WikiPathways</h2>
<p><b>Background</b>: <a href="http://www.wikipathways.org" target="_blank">WikiPathways</a> is a wiki, like Wikipedia, but 
for biological pathway diagrams. We ripped out the text editor of Mediawiki and put in a custom pathway drawing tool. 
In addition to capturing pathway models, we have a number of systems to annotate pathway using standardized identifiers and 
ontology terms.</p>
<p><b>Goal</b>: To upgrade the current ontology tagging system at WikiPathways to take full advantage of the latest 
services from NCBO’s <a href="http://bioportal.bioontology.org/" target="_blank">BioPortal</a>.</p>
<p><b>Description</b>: The upgrade work will include overhauling the current tagging system from the database schema up to the 
user interface. Promising proposals will include an assessment of the current system (found on any given pathway page and 
WikiPathways), research in to the available web services at BioPortal, and a set of features to overhaul and add.</p>
<p><b>Technology and Skills</b>: WikiPathways, PHP, JavaScript, web services, Mediawiki</p>
<p><b>Potential Mentors</b>: Alex Pico, Sravanthi Sinah, Martina Summer-Kutmon</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 4: Computing patient similarity networks for 100K samples</h2>
<p><b>Background</b>: A major goal of modern biomedical research is to tailor patient treatment based on individual genomic 
profiles and/or other patient information. Network-guided approaches can integrate several types of patient data - e.g. 
genetic, RNA, clinical, brain imaging – to identify networks of patient similarities (PSN). These networks could identify 
biological pathways reflecting different disease subcategories or explain why responses to the same medication vary among 
people. Software for computing PSN exists, but currently requires the user to use a patchwork of different tools. Moreover, 
existing machine learning operations don’t easily scale to the size of contemporary datasets (~100K patients).</p>
<p><b>Goal</b>: To create fast, streamlined software to compute and validate PSN. The software will compute PSN based on 
different types of data (e.g. molecular profiles, pathway information, clinical data), by using a similarity measure, such as 
Pearson correlation. It will perform K-fold cross validation for training. Finally it will generate ROC curves to measure the 
sensitivity-specificity of the prediction system.</p>
<p><b>Existing Software</b>: A software pipeline for this workflow already exists, therefore the algorithm itself will not 
need to be developed. Development will focus on optimizing the workflow, and creating a separate software package with 
computing and visualization functionality. Rcpp (R interfacing with Cpp) will be used to program the software. C++ matrix 
libraries (e.g. Armadillo) will be used to reduce memory footprint and speed computations.</p>
<p><b>Publications</b>: 
     <ul>
          <li>Wang B. et al. (2014). <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24464287" target="_blank">
          Similarity network fusion for aggregating data types on a genomic scale</a>. Nature Methods 11(3): 333-7.</li> 
          <li>Mostafavi S. et al. (2008). <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=18613948" target="_blank">
          GeneMANIA: a real-time multiple association network integration algorithm for predicting gene function</a>. 
          Genome Biol. 9 Suppl 1:S4.</li> 
     </ul>
</p>
<p><b>Technology and Skills</b>: R statistical programming language, C++, Java</p>
<p><b>Potential Mentors</b>: Gary Bader, Shraddha Pai</p>
<p><b>Contact</b>: <a href="mailto:shraddhapai.neuro@gmail.com">Shraddha Pai</a></p>
<br />

<h2 dir="ltr" >Idea 5: Develop a Method for Aligning Pathways and Visualizing these Alignments</h2>
<p><b>Background</b>: There is continued growth in the groups generating machine-readable representations of biological pathways 
(such as <a href="http://www.reactome.org/" target="_blank">Reactome</a> or <a href="http://wikipathways.org/" target="_blank">WikiPathways</a>) 
and subsequent efforts to aggregate these pathways into unified repositories, such as 
<a href="http://www.pathwaycommons.org/" target="_blank">Pathway Commons</a>. Aggregation of these 
pathways suffers from multiple pathways being aggregated that differ due to mismatches in author representations of these pathways.</p>
<p><b>Goal</b>: The goal of this project is to develop a method for aligning pathways and then visualizing the results of these 
alignments. Starting points could be to find interactions that use the same substrates and produce the same products.
There are many uses of such a tool including helping to validate pathways as they are added to repositories, merging 
pathways across different resources, and the comparative analyses of pathways across organisms. One way to think about this 
project, is if you're given multiple representations of the same pathway 
(<a href="http://wikipathways.org//index.php?query=wnt&amp;species=Homo+sapiens&amp;title=Special%3ASearchPathways&amp;doSearch=1&amp;ids=&amp;codes=&amp;type=query" target="_blank">examples of several human Wnt pathways</a>) 
can you develop a method to show a unified pathway and how the original representations map to the unified pathway.</p>
<p><b>Technology and Skills</b>: Reactome, PathwayCommons, WikiPathways, Java, XML</p>
<p><b>Potential Mentors</b>: Augustin Luna (MSKCC)</p>
<p><b>Contact</b>: <a href="mailto:lunaa@cbio.mskcc.org" >lunaa@cbio.mskcc.org</a></p>
<br />

<h2 dir="ltr" >Idea 6: Drag and Drop or Palette Editor for SBGN and other graphical notations in Cytoscape</h2>
<p><b>Description</b>: The goal is to design and implement a drag-and-drop or palette-based editor for Cytoscape that would include nodes, edges, 
and graphical annotations. In the ideal case, this would be extensible so that apps add their own nodes or edges with specific 
graphical attributes. The <a href="http://www.sbgn.org/" target="_blank">Systems Biology Graphical Notation</a> (SBGN) is a 
standard graphical notation for presenting pathways and protein interactions. Currently, Cytoscape lacks the node symbols and 
edge types required for SBGN. In this project, those notations will be added to the Cytoscape core, and a simple palette 
editor developed for user interaction.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris, Alex Pico</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu" >Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 7: Improve and expand Cytoscape’s color mapping options</h2>
<p><b>Description</b>: Currently, the color options for Cytoscape’s discrete color mappings are very limited. Brewer palettes provide a 
luminosity-neutral set of colors that would be extremely useful to add to Cytoscape, and there are several other approaches 
to choosing sets of colors that contrast nicely. Creating these color palettes and providing them for use within Cytoscape’s 
visual style interface would be extremely useful to Cytoscape users.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris, Michael Heuer, Alex Pico</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu" >Scooter Morris</a> </p>
<br />

<h2 dir="ltr" >Idea 8: Constrained layout algorithms in Cytoscape</h2>
<p><b>Description</b>: Most layout algorithms in Cytoscape rely on inherent network topology and only a crude set of spatial constraints 
(e.g., grid, circle). There are many cases when working with networks in Cytoscape where you want to be able to constrain 
layouts to user- or attribute-defined spaces. This idea involves implementing layouts that are constrained by:</p>
<ul>
<li>specific shapes</li>
<ul>
<li>basic shapes</li>
<li>cellular templates (e.g., with subcellular organelles)</li>
<li>attribute-driven Venn diagrams (e.g., to visualize attribute groupings and overlaps)</li>
</ul>
<li>specific “anchor” nodes</li>
<ul>
<li>user-defined (e.g., manually selected)</li>
<li>attribute-defined (e.g., hubs or column values)</li>
</ul>
<li>etc...</li>
</ul>
<p><b>Technology and Skills</b>: Cytoscape, Java, and a strong math background</p>
<p><b>Potential Mentors</b>: Scooter Morris, Alex Pico, Bruce Aronow</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu" >Scooter Morris</a> </p>
<br />

<h2 dir="ltr">Idea 9: Collaborative System for Creating/Sharing Biological Icons</h2>
<p><b>Background</b>: <a href="http://wikipathways.org/index.php/WikiPathways:About" target="_blank">Biological pathways</a> provide intuitive 
diagrams of the interactions underlying biological processes. 
For example, the <a href="http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP706&label[]=L-DOPA&label[]=Dopamine&label[]=DDC&colors=yellow" target="_blank">SIDS Susceptibility Pathways</a> shows interactions between metabolites, genes and proteins, such as DDC 
catalyzing the conversion of L-DOPA to Dopamine (look for the items highlighted in yellow on the lower right). The site 
 <a href="http://www.wikipathways.org" target="_blank">WikiPathways</a> is an open, collaborative platform dedicated to drawing and sharing biological pathways. These pathways often 
use icons to represent items like mitochondria or cardiac myocytes (heart muscle cells). The WikiPathways drawing editor has 
a limited set of icons built in, but it does not allow pathway authors to create and re-use custom icons. For example, the 
cardiac myocyte icon is repeated in the SIDS Susceptibility Pathways for <a href="http://wikipathways.org/index.php/Pathway:WP1266" target="_blank">mouse</a>
and for <a href="http://wikipathways.org/index.php/Pathway:WP706" target="_blank">human</a>, but the only way to re-use the 
icon is to manually redraw or copy/paste it. Updates to the icon in pathway do not appear in the other.</p>
<p><b>Goal</b>: Build a biological icon library that allows pathway authors to create icons, share them with other 
WikiPathways’ users and reuse them across pathways. It could work like Github for icons, allowing users to create, fork and 
merge their work, with the best icons rising to the top via a rating system. The library will work as a plugin for the current 
WikiPathways viewer/editor <a href="https://github.com/wikipathways/pathvisiojs" target="_blank">pathvisiojs</a>, allowing pathway authors to create new icons from scratch or to create icons from 
sections of existing pathways, e.g., the cardiac myocyte from the previously referenced pathways. It will have a backend 
component to store the icons and a front-end component to allow WikiPathways users to view and search for icons.</p>
<p><b>Links</b>: An <a href="https://code.google.com/p/svg-edit/" target="_blank">SVG drawing tool </a>with a large number of icons available might provide some ideas. 
</p>
<p><b>Technology and Skills</b>: WikiPathways, JavaScript required. Some backend experience (PHP, Node.js or Python) to create a backend for saving the 
icons. SVG or Canvas a plus.
</p>
<p><b>Potential Mentors</b>: Anders Riutta, Alex Pico, Kristina Hanspers</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr">Idea 10: Biological Pathway Styling Themes</h2>
<p><b>Background</b>: Description: <a href="http://wikipathways.org/index.php/WikiPathways:About" target="_blank">Biological pathways</a> provide intuitive diagrams of the interactions underlying biological 
processes. For example, the <a href="http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP706&label[]=L-DOPA&label[]=Dopamine&label[]=DDC&colors=yellow" target="_blank">SIDS Susceptibility Pathways</a>
shows interactions between metabolites, genes and proteins, such as 
DDC catalyzing the conversion of L-DOPA to Dopamine (look for the items highlighted in yellow on the lower right). The site 
 <a href="http://www.wikipathways.org" target="_blank">WikiPathways</a> is an open, collaborative platform dedicated to drawing and sharing biological pathways. These diagrams are viewed 
in many different environments, such as laptop screens, presentation slides and printed science journals. At WikiPathways today, 
the default styling theme for these diagrams is best suited for viewing by an individual on a computer screen, but it would be 
useful to export the diagrams with themes optimized for print and for presentations.</p>
<p><b>Goal</b>: Develop a custom theming system to allow for exporting pathways with themes optimized for print and for 
projected presentation slides. For example, the presentation export option might make pathway diagrams easier to read by 
removing some text, enlarging the remaining text and inverting the colors of the diagram. Optionally, the presentation option 
could integrate with one or more existing online presentation framework(s).</p>
<p><b>Links</b>:
<ul>
<li><a href="https://github.com/wikipathways/pathvisiojs" target="_blank">WikiPathways pathway viewer/editor</a></li>
<li>Example online presentation frameworks:</li>
<ul>
     <li><a href="https://github.com/hakimel/reveal.js" target="_blank">github.com/hakimel/reveal.js</a></li>
     <li><a href="https://github.com/bartaz/impress.js" target="_blank">github.com/bartaz/impress.js</a></li>
     <li><a href="https://docs.google.com/presentation" target="_blank">docs.google.com/presentation</a></li>
</ul>
</ul>
</p>
<p><b>Technology and Skills</b>: WikiPathways, User experience, graphical design, CSS, some JavaScript.</p>
<p><b>Potential Mentors</b>: Anders Riutta, Alex Pico, Kristina Hanspers</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 11: Details Panel for Semantic Terms</h2>
<p><b>Background</b>: <a href="http://wikipathways.org/index.php/WikiPathways:About" target="_blank">Biological pathways</a> provide intuitive 
diagrams of the interactions underlying biological processes. For 
example, the <a href="http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP706&label[]=L-DOPA&label[]=Dopamine&label[]=DDC&colors=yellow" target="_blank">SIDS Susceptibility Pathways</a> 
show interactions between metabolites, genes and proteins, such as DDC catalyzing 
the conversion of L-DOPA to Dopamine (look for the items highlighted in yellow on the lower right). The site <a href="http://www.wikipathways.org" target="_blank">WikiPathways</a>
is an open, collaborative platform dedicated to drawing and sharing biological pathways.</p>
<p>A common challenge in biological research is dealing with an ever-growing number of terms, some of them 
synonyms and others almost synonyms but with important differences in meaning. For instance, WikiPathways uses the term 
“GeneProduct” in its annotation panels (to see an example, click the “TNF” box in the upper left corner of the SIDS pathway). 
A WikiPathways user might be unsure of the exact definition of the “GeneProduct”, so it would be useful to point the user to 
the <a href="http://vocabularies.wikipathways.org/wp#GeneProduct" target="_blank">definition of the term</a> in the WikiPathways vocabulary. This could be done with a link or a mouse over display. Another 
example from the SIDS pathway is the Extracellular Space, which could display information from this IRI: 
 <a href="http://identifiers.org/go/GO:0005615" target="_blank">href="http://identifiers.org/go/GO:0005615"</a>.</p>
<p>The first two examples are easy to display because each IRI has an HTML representation, but a more challenging example is 
<a href="http://www.biopax.org/release/biopax-level3.owl#Protein" target="_blank">Protein</a>. 
This IRI returns RDF but does not have an HTML 
representation, so it would be necessary to extract information from the RDF, such as the definition, and display it in a 
small, easy-to-read panel.</p>
<p><b>Goal</b>: Develop a browser-based display system for dereferenced IRIs. The system will use content-type negotiation to 
request an HTML representation of a resource, but it will be able to convert JSON-LD, RDF and OWL representations into an HTML 
representation for display in a small details panel or pop-over. Additionally, develop a plugin for the WikiPathways 
viewer/editor <a href="https://github.com/wikipathways/pathvisiojs" target="_blank">pathvisiojs</a> that integrates the display system. Optionally, if time allows, add functionality to allow 
pathvisiojs users to switch vocabularies for the GUI display, mapping from the built-in vocabularies to the user-preferred 
vocabularies with <a href="http://www.w3.org/TR/skos-primer/" target="_blank">SKOS</a> files.</p>
<p><b>Links</b>:
<ul>
<li><a href="https://semantic-mediawiki.org/" target="_blank">Project with related functionality</a></li>
<li><a href="https://github.com/alexmilowski/green-turtle" target="_blank">An RDFa implementation for browsers</a></li>
<li><a href="https://code.google.com/p/ubiquity-rdfa/wiki/TutorialIntroduction" target="_blank">Possibly relevant project</a></li>
</ul></p>
<p><b>Technology and Skills</b>: WikiPathways, JavaScript, <a href="http://json-ld.org/" target="_blank">JSON-LD</a>, RDF, OWL, optionally SKOS</p>
<p><b>Potential Mentors</b>: Anders Riutta, Alex Pico, Kristina Hanspers</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 12: Development of Experiment Data Curation Tools using the NDEx infrastructure and the OpenBEL Knowledge 
Representation Language</h2>
<p><b>Background</b>: Biological knowledge can be represented in network form, a methodology that has proved valuable in 
communicating and computing on biological mechanisms.  In some cases, the biological relationships encoded in the networks 
correspond directly to experimental results, as in protein binding assays that result in protein-protein interaction 
assertions. In other cases, scientific findings have been encoded as qualitative causal relationships, providing networks 
of facts that can be used to build causal reasoning models. These encodings have been frequently performed by manual 
curation or ad hoc software, so there is an opportunity to develop more general tools for abstracting experimental results 
into reusable portable knowledge encoded as biological networks. Recent technological progress in modern experimental OMICS 
techniques allow scientists to generate large data sets that need to be carefully validated and curated, therefore the 
automation of the interpretation and abstraction of these data to computable scientific findings is a valuable technique to 
bridge the gap between data generation and its availability for the scientific community.</p>
<p><b>Goal</b>: To develop and test a set of utilities to process selected types of experimental data into more abstract 
biological knowledge as networks in the OpenBEL language and to compare the resulting networks with existing, manually 
curated knowledge networks.  This transformation of data to computable scientific findings represents the automation of a 
basic scientific process; the abstraction of data tables into statements of inferred causality in a given biological context 
makes those results more reusable, more computable, and easier to integrate with findings from other studies. This facilitates 
scientists to take better advantage of one class of “big data” and can potentially help use the experimental data to identify 
new hypotheses in the biomedical/pharmaceutical  fields.</p>
<p><b>Project breakdown</b>:</p>
<p><u>Phase 1</u> of the project aims to develop new software utilities to <u>capture</u> experimental data as OpenBEL statements.</p>
<p>In <u>Phase 2</u>, the newly developed utilities will be tested with experimental data sets derived from microarray and 
mass spec analysis. One source of testing will be a set of experimental data generated and validated by the Itkin-Ansari lab 
at UC San Diego and focused on the involvement of bHLH transcription factor Id3 in pancreatic cancer.</p>
<p><u>Phase 3</u> will focus on generating additional utilities for the <u>comparison</u> of OpenBEL networks.</p>
<p>Finally, <u>Phase 4</u> will use these tools to compare and contrast the OpenBEL networks captured from experimental data with 
existing OpenBEL networks created by manual curation.</p>
<p>Project milestones and deliverables will include: 
     <ul>
          <li>Update/expansion of existing related networks available in public databases and repositories.</li>
          <li>Pre-publication of all new network data using the NDEx infrastructure 
          (<a href="http://www.ndexbio.org" target="_blank">www.ndexbio.org</a>). </li>
          <li>Generation of new leads to support research project progress.</li>
          <li>Publication of at least 1 research article in an international peer-reviewed journal 
          showcasing the newly developed tools, their use and potential applications.</li>
     </ul>
</p>

<p><b>Technology and Skills</b>: NDex, Python, JavaScript</p>
<p><b>Potential Mentors</b>: Dexter Pratt, Pamela Itkin-Ansari</p>
<p><b>Contact</b>: <a href="mailto:rudi@ucsd.edu">Rudolf T. Pillich</a></p>
<br />

<h2 dir="ltr" >Idea 13: Integration of WikiPathways Visualization with PathVisio into Bioclipse</h2>
<p><b>Background</b>: Bioclipse is a powerful platform combining a powerful scripting language to write data analysis and 
visualization workflows, which can easily be shared on, for example, <a href="http://myexperiment.org/" target="_blank">MyExperiment.org</a>. 
While it has various components to support metabolomics in general, it lacks support for visualization of metabolic pathways.</p>

<p><b>Goal</b>: To integrate PathVisio visualization widget into Bioclipse for visualization of pathway information from 
WikiPathways. Selecting protein and metabolites in the pathway visualization, will show that chemical entity in other 
Bioclipse components, using the Jmol and JChemPaint plugins for Bioclipse. Preferably, interaction can be scripted, allowing 
integration with other metabolomics features in Bioclipse.</p>

<p><b>Description</b>: You will use Java to write a plugin for <a href="http://www.bioclipse.net/" target="_blank">Bioclipse</a> that uses 
<a href="http://www.pathvisio.org/" target="_blank">PathVisio</a> as a library to visualize 
metabolic pathways from WikiPathways inside Bioclipse. You will use the 
<a href="http://www.eclipse.org/articles/Article-WorkbenchSelections/article.html">Eclipse selection event propagation mechanism</a> so 
that when users select protein and metabolites, Bioclipse will automatically visualize 2D or 3D structural information about 
those molecules (only the event creation has to be implemented). Preferably, a 
<a href="http://www.biomedcentral.com/1471-2105/10/397" target="_blank">scripting language extension</a> is written, 
to allow extracting information from the pathway and highlighting of parts of the pathway. This involves extending an existing 
interface with methods that can be run from the scripting language.</p>
<p><b>Technology and Skills</b>: WikiPathways, Bioclipse, Java</p>
<p><b>Potential Mentors</b>: Egon Willighagen, Jonathan Alvarsson</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 14: Help scientists share data with the Synapse client for Cytoscape</h2>
<p><b>Background</b>: <a href="http://synapse.org" target="_blank">Synapse</a> is an web-based platform
that makes it easy for scientists to share datasets. Users can request a DOI number for datasets, which allows them to be 
cited and promoted in publications. This encourages scientists to share data with the public, even after the data is deemed 
no longer relevant to them -- but could be to others. Synapse also provides extensive _provenance_ tools for tracking 
progressive changes to datasets as they are transformed from raw to publishable data. This allows for scientists to 
collectively establish, review, and share data processing pipelines.</p>
<p>The <a href="http://apps.cytoscape.org/apps/synapseclient" target="_blank">Synapse client app</a> for Cytoscape allows 
users browse and search their Synapse files and to import them into Cytoscape. Cytoscape is a powerful analysis network tool 
that can play an important role in data processing pipelines. The current Synapse app lacks facilities for putting processed 
data back into Synapse.</p>
<p><b>Goal</b>: The goal of this project is to expand the Synapse app's abilities so that it can submit data back to Synapse 
and annotate it using its provenance tools.</p>
<p>First phase: create menu items and commands for submitting networks, tables, and sessions to a Synapse account. The app 
would present a very simple UI for asking the user to which project the file belongs.</p>
<p>Second phase: provide an intuitive, easy-to-use UI for annotating submissions with Synapse's provenance tools. When the 
user submits files to Synapse, the user can specify a name and description of what was done to the submission and what files 
were used to create the submission. These files could be from Synapse or from any URL.</p>
<p><b>Links</b>: 
<ul>
     <li><a href="http://apps.cytoscape.org/apps/synapseclient" target="_blank">Synapse app page</a></li>
     <li><a href="https://github.com/samadlotia/SynapseClientForCytoscape" target="_blank">Synapse app code</a></li>
     <li><a href="http://hud.rel.rest.doc.sagebase.org.s3-website-us-east-1.amazonaws.com/index.html" target="_blank">Synapse REST API reference</a></li>
</ul>
</p>
<p><b>Technology and Skills</b>: Cytoscape, Java, OSGi, Using REST APIs in Java, Swing</p>
<p><b>Potential Mentors</b>: Alex Pico</p>
<p><b>Contact</b>: <a href="mailto:apico@gladstone.ucsf.edu" >Alex Pico</a></p>
<br />

<h2 dir="ltr" >Idea 15: Export as D3.js-Based Web Application</h2>
<p><b>Background</b>: From Cytoscape 3.2, users can <a href="http://wiki.cytoscape.org/Cytoscape_3/UserManual/Publish" target="_blank">export 
their network visualizations as a simple web application</a>. This feature uses <a href="http://js.cytoscape.org/" target="_blank">cytoscape.js</a> 
and other popular libraries such as <a href="https://angularjs.org/" target="_blank">Angular.js</a> to reproduce visualizations generated with 
Cytoscape 3 Desktop application using modern web technologies.</p>
<p><b>Goal</b>: This project is extending this idea. In addition to exporting Cytoscape-generated visualizations as simple 
node-link diagrams, you will add new options for this feature, including:</p>
<ul>
<li>TreeMap</li>
<li>Dendrogram</li>
<li>Circle Packing</li>
<li>Node-Link Tree</li>
<li>Adj. Matrix</li>
</ul>
<p>This enables users to visualize the same data set in different ways. In this project, you need to extend the existing 
<a href="http://apps.cytoscape.org/apps/d3jsexporter" target="_blank">D3.js</a> Exporter App and add new export options. Note that you need 
both of Java and JavaScript knowledge to implement this feature. Also, we assume you are comfortable with modern web 
application development tools, such as command-line tools running on node.js, JavaScript frameworks like Angular.js, testing 
frameworks, and of course, <a href="http://d3js.org/" target="_blank">D3.js</a>.</p>
<p><b>Technology and Skills</b>: Cytoscape, JavaScript, D3.js, Java, knowledge of modern web development tools such as node.js/grunt/gulp/yeoman.</p>
<p><b>Potential Mentors</b>: Kei Ono</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 16: BridgeDb config plugin in PathVisio</h2>
<p><b>Background</b>: <a href="http://www.bridgedb.org" target="_blank">BridgeDb</a> is an identifier mapping framework used in PathVisio and WikiPathways. 
Currently it is possible to <a href="http://www.pathvisio.org/wiki/BridgeDbConfigPluginHelp" target="_blank"> load a mapping databases for gene products and metabolites</a>. 
This should be extended with interaction mapping databases.</p>
<p><b>Goal</b>: The existing <a href="http://www.pathvisio.org/plugin/bridgedbconfig-plugin/" target="_blank">BridgeDbConfig plugin</a> provides 
functionality to load more mapping databases or to use other web services for the mappings. This functionality should be 
extended and integrated into the core PathVisio application as another core module. The configuration dialog has to be 
improved so it is more intuitive and user friendly. Furthermore the plugin should allow users to download and update BridgeDb 
mapping databases from within PathVisio. Currently the users have to download the databases first and then load them into 
PathVisio. This step should be automatized by the plugin.</p>
<p><b>Technology and Skills</b>: BridgeDb, PathVisio, Java</p>
<p><b>Potential Mentors</b>: Kutmon, Jonathan Melius</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com" >wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 17: Citing publications in biological pathways - PathVisio and WikiPathways</h2>
<p><b>Background</b>: <a href="http://www.pathvisio.org" target="_blank">PathVisio</a> is a commonly-used pathway editor, visualization and analysis tool implemented in Java and is 
linked with the collaborative pathway database <a href="http://www.wikipathways.org" target="_blank">WikiPathways</a>. Currently, PathVisio and WikiPathways 
allow users to add publication references to <a href="http://www.ncbi.nlm.nih.gov/pubmed" target="_blank">PubMed</a>.</p>
<p><b>Goal</b>: The goal of this project is to extend the functionality to allow the reference of other identifiers like DOIs, 
export references in standard formats like Bibtex or EndNote and provide statistics about citations used on WikiPathways 
(total number of citations, avg. number of citations per pathway, top cited papers/journals, ...).</p>
<p><b>Technology and Skills</b>: PathVisio, WikiPathways, Java, PHP, Mediawiki</p>
<p><b>Potential Mentors</b>: Sravanthi Sinah, Martina Summer-Kutmon, Anwesha Bohler</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com" >wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 18: Enhanced Layout Algorithms</h2>
<p><b>Description</b>: As Cytoscape is used for increasingly large networks, the performance of our layout algorithms becomes 
a limiting factor. There are a number of layout algorithms that perform very well in real-world scenarios that have not been 
implemented in Cytoscape. For example, the <a href="http://yifanhu.net/PUB/graph_draw_small.pdf" target="_blank">Yifan Hu Multi-level layout 
algorithm</a>, a Fast Multipole Multilevel layout (See  <a href="http://www.ogdf.net/doc-ogdf/classogdf_1_1_f_m_m_m_layout.html" target="_blank">OGDF</a> 
for example), Yifan Hu proportional, and potentially some of the better algorithms from Gephi. Cytoscape provides a 
framework for adding new layout algorithms as Apps, so this work could all be done as a series of separate apps.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 19: Compound Graph Visualization in Cytoscape 3</h2>
<p><b>Description</b>: Cytoscape 3 has introduced two concepts into the core, groups and annotations, that could be combined 
to provide a compound graph visualization. Currently groups can be either in a “collapsed” or “expanded” state. 
<p><b>Goal</b>: For a compound graph, a new visualization state needs to be introduced that allows the compound node 
(group) to be visualized as a shape (concave or 
convex hull) around it’s children. Moving that shape would move all of it’s children, etc.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 20: Visualization of Time-series Cluster Data</h2>
<p><b>Description</b>: Currently, the <a href="http://apps.cytoscape.org/apps/clustermaker" target="_blank">clusterMaker app</a> 
uses standard heat maps to visualize attribute cluster data (based on Java TreeView). Increasingly, time series data is 
becoming important for clustering applications and it would be useful to expand the provided
visualization options to explicitly support time series data. For these purposes, we’re thinking about time-series that have 
multiple conditions in each series, which means we need to move beyond the standard two-dimensional heat map. 
<p><b>Goal</b>: This project can explore utilizing animation, small multiples, or other standard visualization techniques (or your own novel techniques) 
to provide new visualization opportunities to users. Ideally, this would be integrated with network-based animation tools 
such as <a href="http://apps.cytoscape.org/apps/dynnetwork" target="_blank">DynNetwork</a> or 
<a href="http://apps.cytoscape.org/apps/cyanimator" target="_blank">CyAnimator</a>.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 21: Develop a Converter between Visual and Textual Representations of Biological Pathways</h2>
<p><b>Background</b>: <a href="http://www.pathwaycommons.org/" target="_blank">Pathway Commons</a> seeks to aggregate all 
pathway data available in literature, public databases, and curation/crowd sourced efforts. Our goal is to accelerate this 
process by leveraging pathway diagram editors that support a widely used visual representation for biological pathways.
<p><b>Problem</b>: </p>Pathway Commons data is represented in the 
<a href="http://www.ncbi.nlm.nih.gov/pubmed/20829833" target="_blank">Biological Pathway Exchange (BioPAX)</a> 
community standard format, but the BioPAX format lacks a simple graphical editor. On the other hand, the System Biology 
Graphical Notation (SBGN) is a visual representation for pathways that is supported in about 20 tools.</p>
<p><b>Goal</b>: We'd like you to build a converter between the BioPAX and SBGN formats. A similar project has been developed 
with the editor CellDesigner (that shares a format similar to SBGN) making us believe that this should be a reasonably 
straight forward project; many of the mapping issues have already been addressed. The project will make use of two Java 
libraries <a href="http://www.biopax.org/paxtools.php" target="_blank">Paxtools</a> (for BioPAX) and 
<a href="http://www.sbgn.org/LibSBGN" target="_blank">LibSBGN</a> (for SBGN).</p>
<p><b>Links</b>:
     <ul>
          <li><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232372/" target="_blank">BioPAX support in CellDesigner</a></li>
          <li><a href="ftp://ftp.pantherdb.org/biopax" target="_blank">CellDesigner to BioPAX source code</a></li>
          <li><a href="http://sourceforge.net/p/libsbgn/code/HEAD/tree/trunk/" target="_blank">libSBGN source code and example SBGNML files</a></li>
     </ul>
</p>
<p><b>Technology and Skills</b>: BioPAX, SBGN, PathwayCommons, Java and XML</p>
<p><b>Potential Mentors</b>: Augustin Luna</p>
<p><b>Contact</b>: <a href="mailto:lunaa@cbio.mskcc.org" >Augustin Luna</a></p>
<br />

<h2 dir="ltr" >Idea 22: Visualizing complex data with sets-aware layout algorithms</h2>
<p><b>Background</b>: Sets app for <a href="http://www.cytoscape.org" target="_blank">Cytoscape</a> is a powerful tool where 
users can define individual sets for nodes and edges. 
Users create sets from a selection or from some discrete criteria. For instance, given a network that consists of gene and 
protein interactions, the user can define a set of all gene nodes and another set of protein nodes. Selecting all gene nodes 
is a snap, as the user can just click on the gene set. New sets can be made from existing ones by using the union, intersect, 
and subtract operations.</p>
<p>The app also provides two set-aware layout algorithms:
     <ul>
          <li>Grid layout: creates grids of node sets without taking network topology into account.</li>
          <li>FDL: strike a balance between set membership and network topology, so unconnected nodes that belong in the same 
          set move closer together.</li>
     </ul>
</p>
<p><b>Goal</b>: The goal of this project is to expand sets-based layout algorithms. Layout algorithms that take user-defined 
set membership into consideration can help elucidate complex network data in ways that purely automatic layout algorithms 
cannot. Here are some planned sets-based layout algorithms:
     <ul>
          <li>Inner-FDL: Creates a grid of each set, then applies FDL on each set individually.</li>
          <li>Ordered grid: Within a set, order each node based on an attribute. This is useful for visualizing genomic 
          locations, so that nodes in a set will be ordered by where their representative genes are in the genome. The layout 
          algorithm could also place nodes in some other shape besides a grid, like a circle or spiral.</li>
     </ul>
</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Alex Pico, Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:apico@gladstone.ucsf.edu" >Alex Pico</a></p>
<br />

<h2 dir="ltr" >Idea 23: Improve the Text-Mining Functionality of Factoid</h2>
<p><b>Background</b>: <a href="http://pathwaycommons.github.io/factoid/" target="_blank">Factoid</a> aims to be a tool that helps authors 
annotate biological interactions that occur in the text of academic publications in a machine-readable manner. Currently, 
Factoid uses basic entity-recognition to identify genes, proteins, biological processes to identify interactions between 
these entities. 
<p><b>Goal</b>: The goal of this project is to integrate more sophisticated algorithms into the extraction of 
interactions from text content. A starting point for some of these would be the algorithms outlined by the EBI website 
WhatIzIt.</p>
<p><b>Technology and Skills</b>: PathwayCommons, Experience connecting web services, Javascript</p>
<p><b>Potential Mentors</b>: Augustin Luna (MSKCC), Max Franz (U of Toronto) </p>
<p><b>Contact</b>: <a href="mailto:lunaa@cbio.mskcc.org" >Augustin Luna</a>, <a href="mailto:maxkfranz@gmail.com" >Max Franz</a></p>
<br />

<h2 dir="ltr" >Idea 24: Export to GPML with WikiPathways app</h2>
<p><b>Background</b>: <a href="http://wikipathways.org" target="_blank">WikiPathways</a> is an open, collaborative wiki for 
biological pathways. (Pathways are diagrams representing biochemical processes.) Like Wikipedia articles, all pathways are 
open and free to everyone, and anyone can contribute content.</p>
<p>The <a href="http://apps.cytoscape.org/apps/wikipathways" target="_blank">WikiPathways app for Cytoscape</a> allows users 
to take pathway content from WikiPathways and import it into Cytoscape. Users can then modify the pathway's appearance using 
Cytoscape's myriad visualization tools.</p>
<p>The native format for representing pathway content in WikiPathways is GPML, an XML-based format. GPML files can be edited 
online on WikiPathways or can be edited using the <a href="http://pathvisio.org" target="_blank">PathVisio</a> application. 
The WikiPathways Cytoscape app provides the means to import GPML files into Cytoscape networks.</p>
<p><b>Goal</b>: Users often need to update pathway content to reflect new insights from literature or data gleaned from 
Cytoscape's powerful visualization and data integration tools. The goal of this project is to add the ability to export GPML 
files, so that users can effortlessly submit new pathway content to the WikiPathways website.</p>
<p>When pathway content is imported from GPML into Cytoscape, the WikiPathways app reads the GPML content and creates a series 
of columns in the node table representing the pathway node's visual information. This information is then used by Cytoscape to
style the network. The exporter would use this node table data to reconstruct the GPML file. Additional node table data could 
be needed to complete the translation to and from Cytoscape and GPML.</p>
<p><b>Technology and Skills</b>: WikiPathways, Cytoscape, Java, XML</p>
<p><b>Potential Mentors</b>: Alex Pico</p>
<p><b>Contact</b>: <a href="mailto:apico@gladstone.ucsf.edu" >Alex Pico</a></p>
<br />

<h2 dir="ltr">Idea 25: Developing Web Interface For The Online Resource for Interface- Based Interaction Networks (TOR-IBIN) 
Using Cytoscape.js</h2>
<p><b>Background</b>: The protein-protein interaction (PPI) is a physical contact between two proteins, or more, resulted from 
biochemical reaction or electrostatic forces. The interaction usually takes place through an Interface, which is part of the 
protein (e.g. protein domain) with special molecular structure that enables and mediates the interaction between proteins. 
Interface-based PPI predictions help in understanding the mechanisms of interactions between proteins as well as helps in 
deciphering the functional roles of the proteins.</p>
<p><b>Focus</b>: TOR-IBIN is an online resource for gathering and storing the published interface-based (IB) protein-protein 
interaction networks (IN) predicted using any PPI prediction method. TOR-IBIN aims to make these networks available in a 
central repository with unified format and comprehensive search functions that maximize the use of these data. TOR-IBIN 
web-based interface aims to implement the browse and search functions and well as network visualization using Cytoscape.JS and 
network download in several formats including JSON and tab-delimited formats. TOR-IBIN is a relational database implemented in 
MySQL.
- The web interface and functionality will be similar to the web interface of the cancer variants database (Can-VD) 
canvd.baderlab.org.
</p>
<p><b>Goals</b>:
<ol>
<li>Develop a web-based interface for interface-based PPI data retrieval with a comprehensive search functions and data 
download options.</li>
<li>Develop a visualization and annotation module using Cytoscape.js to visualize the retrieved PPI data as a network with 
annotations that highlights the features of the predicted PPI such as the prediction confidence and validation...etc.</li>
</ol></p>
<p><b>Inputs</b>: The student will be provided with the database design and sample data as well as an introduction to the 
topic and the different possible visualizations and annotations needed. The student will have the freedom to choose among 
them or to propose other options.</p>
<p><b>Technology and Skills</b>: Cytoscape.js, Javascript, JSON, PHP, JQuery</p>
<p><b>Potential Mentors</b>: Mohamed Helmy, Gary Bader</p>
<p><b>Contact</b>: <a href="mailto:mohamed.attiashahata@utoronto.ca">Mohamed Helmy</a>, <a href="mailto:gary.bader@utoronto.ca">Gary Bader</a></p>

<br />
<h2 dir="ltr">Idea 26: Full Compound Support in Cytoscape 3</h2>
<p><b>Background</b>: The notion of compound graphs is used to represent more complex types of relationships or varying levels 
of abstractions in data visualization. Networks from almost all domains need such abstractions and nesting, including 
biological networks. The recently developed standard notation of <a href="http://www.sbgn.org/" target="_blank">SBGN</a> makes 
heavy use of such structures for molecular complexes and cellular locations.</p>
<p><b>Goal</b>: We would like to first implement full compound support in 
<a href="http://www.cytoscape.org/" target="_blank">Cytoscape</a> 3 core. This will not only allow users 
to nest nodes of a network but also automatically adjust size of a compound node as its child nodes get moved, resized or 
deleted. Then, we'd like to implement a native spring embedder based layout algorithm named 
<a href="http://www.cs.bilkent.edu.tr/~ivis/layout/cose-animated-demo/cose.html" target="_blank">CoSE</a> that takes such 
structures 
into account during layout. A <a href="http://www.cs.bilkent.edu.tr/~ivis/chilay.html" target="_blank">Java implementation of 
CoSE</a> is already publicly available and needs to be integrated into Cytoscape 3.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java, Data Structures and Algorithms</p>
<p><b>Potential Mentors</b>: Ugur Dogrusoz (Bilkent University & MSKCC) and Gerardo Huck (U of Toronto)</p>
<p><b>Contact</b>: <a href="mailto:ugur@cs.bilkent.edu.tr">ugur@cs.bilkent.edu.tr</a>, <a href="gerardohuck@gmail.com">gerardohuck@gmail.com</a></p>
<br />

<h2 dir="ltr">Idea 27: Extending and improving PathVisio Data Import</h2>
<p><b>Background</b>: <a href="http://pathvisio.org" target="_blank">PathVisio</a> is a pathway editor and analysis tool. PathVisio allows users to upload and visualize data on 
biological pathways. Currently, it is possible to upload only one dataset at a time as a comma separated file. </p>
<p><b>Goal</b>: This functionality should be extended to allow uploading a number of data files at the same time as well 
as uploading of multiple data file formats (for eg. Excel sheets) should be allowed.
This will involve extending the <a href="http://developers.pathvisio.org/browser/trunk/modules/org.pathvisio.gex" target="_blank">gex</a> and 
<a href="http://developers.pathvisio.org/browser/trunk/modules/org.pathvisio.data" target="_blank">data</a> modules of PathVisio. These modules are part of the PathVisio core and take care 
of importing experimental data and mapping it to the elements in a pathway. </p>
<p>Features of the new plugin:
<ul>
<li>Uploading multiple datasets at once : these could be any kind of quantitative or qualitative data about any pathway 
element, such as genes, proteins, metabolites, and interactions.</li>
<li>Linking each individual dataset to a particular element on a pathway, for eg. linking a gene expression dataset to genes 
on pathways and a flux data set to interactions on pathways.</li>
<li>Importing data from multiple file formats (For eg: .xls, .ods).</li>
<li>Create a pop-up panel accessible from the menu bar of the PathVisio main window to show:
     <ul>
          <li>all the currently loaded datasets</li>
          <li>the biological elements they are linked to</li>
          <li>the currently loaded annotation databases</li>
     </ul>
</ul></p>
<p><b>Technology and Skills</b>: PathVisio, Java</p>
<p><b>Potential Mentors</b>: Anwesha Bohler, Martina Summer-Kutmon</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 28: Over-representation module for WikiPathways</h2>
<p><b>Background</b>: <a href="http://www.pathvisio.org" target="_blank">PathVisio</a> is a commonly-used pathway editor, visualization and analysis tool 
implemented in Java. Currently, PathVisio provides an advanced statistics module that allows users to define criteria 
to perform pathway statistics, which is sometimes not needed.</p>
<p><b>Goal</b>: In this project, the student will extend the statistics module with a basic over-representation methods 
that uses a gene list as input and provides ranked pathways as output. The genes in the pathways are highlighted when they 
are in the input gene list. Additionally this basic analysis module will also be integrated on <a href="http://www.wikipathways.org" target="_blank">WikiPathways</a>. The student 
will develop a page where the user can provide a list of genes, then the PathVisio 
statistics module gets called to calculate the over-representation and the result will be shown on WikiPathways. The pathways 
will be shown in the pathvisiojs viewer and the genes in the input list will be highlighted.</p>
<p><b>Technology and Skills</b>: WikiPathways, Java, PHP</p>
<p><b>Potential Mentors</b>: Martina Summer-Kutmon, Anwesha Bohler</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br/ >

<h2 dir="ltr" >Idea 29: Tutorial mode for PathVisio</h2>
<p><b>Background</b>: <a href="http://www.pathvisio.org" target="_blank">PathVisio</a> is a commonly-used pathway editor, visualization and analysis tool 
implemented in Java. The analyses in PathVisio and the different advanced visualization methods can become very complex 
and having an integrated tutorial mode in the application is a powerful tool to give detailed instructions while performing 
a task.</p>
<p><b>Goal</b>: This will give the users a step-by-step guide with hints and additional information while learning how to use 
PathVisio. Documentation and tutorials are crucial elements of software development and the goal of this project is to 
implement a tutorial mode in PathVisio. PathVisio can be extended through plugins, so the tutorial mode should allow other 
modules to provide additional tutorials that can be selected if the plugins are installed.</p>
<p><b>Technology and Skills</b>: PathVisio, Java, development of tutorials</p>
<p><b>Potential Mentors</b>: Martina Summer-Kutmon, Kristina Hanspers, Sravanthi Sinha</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br/ >

<h2 dir="ltr" >Idea 30: Principal Component Analysis</h2>
<p><b>Description</b>: One of the common multidimensional analysis tools is principal component analysis. At this point, there 
is no good PCA tool integrated with Cytoscape. The ideal tool would support performing the analysis, visualizing the results 
of the analysis using standard approaches, and then visualizing the results in the context of the Cytoscape network itself.
There are many tools that could assist in this task and the analysis itself is pretty well understood. The challenge to this 
idea is looking for ways to integrate the results of the analysis into Cytoscape.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 31: Incremental Loading of Networks</h2>
<p><b>Background</b>: Cytoscape now supports using JSON as a format for loading networks. In an effort to improve exploration 
of extremely large networks, it would be extremely useful to be able to load parts of the network on request. One use case 
for this is loading networks from the <a href="http://sfld.rbvi.ucsf.edu/django/" target="_blank">Structure-Function Linkage 
Database</a>, which creates Cytoscape networks representing similarity between groups of proteins. Some of these networks are extremely large (millions of edges) and usually users are 
only interested in local neighborhoods of the network.</p>
<p><b>Goal</b>: This project would involve writing a Cytoscape app that would load an 
abstraction of these networks (where multiple nodes are represented by a single "meta" node) into Cytoscape and then allow 
users to incrementally load all of the nodes within that metanode.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris, Samad Lotia</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 32: Add t-SNE to clusterMaker</h2>
<p><b>Background</b>: <a href="http://apps.cytoscape.org/apps/clustermaker" target="_blank">clusterMaker</a> is a Cytoscape 
app that unifies different clustering techniques and displays into a single interface. Current clustering algorithms 
include hierarchical, k-medoid, AutoSOME, and k-means for clustering expression or genetic data; and MCL, transitivity 
clustering, affinity propagation, MCODE, community clustering (GLAY), SCPS, and AutoSOME for partitioning networks based on 
similarity or distance values. Hierarchical, k-medoid, AutoSOME, and k-means clusters may be displayed as hierarchical groups 
of nodes or as heat maps. All of the network partitioning cluster algorithms create collapsible "meta nodes" to allow 
interactive exploration of the putative family associations within the Cytoscape network, and results may also be shown as a 
separate network containing only the intra-cluster edges, or with inter-cluster edges added back.</p>
<p><b>Goal</b>: <a href="http://lvdmaaten.github.io/tsne/" target="_blank">t-SNE</a> is a technique for dimensionality 
reduction that is widely used. It would be extremely useful to add this algorithm into the existing clusterMaker app.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 33: cyRecommender - New Cytoscape 3 app for recommending network ontologies</h2>
<p><b>Background</b>: Cytoscape 3 is a powerful software for networks that provides diverse apps in order to combine 
external knowledge with the network connectivity. There are a few apps that allow to link ontology details with node labels. 
However, existing apps require the user to decide what ontology or ontologies are best for each network, which may be a 
hard and time‑consuming task. The NCBO Recommender is a service developed by Stanford University that has been designed to 
recommend the most appropriate ontologies for a corpus or a list of keywords. Integrating the NCBO Recommender with Cytoscape 
will make it possible to automatically obtain the best ontological representation for a set of network nodes. The NCBO is 
developing a new version of the Recommender, which will be available in a few months. The proposed Cytoscape app will be 
one of the first tools using the new version of the Recommender.</p>
<p><b>Goal</b>: To integrate the NCBO Recommender service into a new Cytoscape app. After selecting a set of network nodes, 
the Recommender will be invoked to obtain the best ontologies to represent the node labels, and the network will be presented 
using the recommended ontologies. The integration will consist on an easy to use graphical interface for any level of ontology
experience.</p>
<p><b>Description</b>: You will use Java to write an app for <a href="http://www.cytoscape.org/cy3.html" target="_blank">Cytoscape 3</a> that will access the NCBO Recommender REST services. 
Your tool will make it possible to:
<ul>
<li>Get the keywords from the selected node labels, send them to <a href="http://bioportal.bioontology.org/recommender" target="_blank">NCBO Recommender</a>
<a href="http://data.bioontology.org/documentation" target="_blank">REST services</a>, and obtain the recommended 
ontologies (the selection could be a subnetwork or the entire network).</li>
<li>Visualize the network using the recommended ontologies.</li>
<li>Save the recommendations as CSV or TAB-separated files.</li>
<li>Save the network with the new added ontology information for each node.</li>
</ul>
This work involves programming a new app that will interact with Cytoscape in the node selection, network plotting, etc. 
The code will be available as an open source GitHub project.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java, GitHub</p>
<p><b>Potential Mentors</b>: Cristian R Munteanu, Marcos Martinez, Janna Hastings and Alejandro Pazos</p>
<p><b>Contact</b>: <a href="mailto:muntisa@gmail.com">Cristian R Munteanu</a></p>
<br />

<h2 dir="ltr" >Idea 34: Advanced search on WikiPathways</h2>
<p><b>Background</b>: <a href="http://www.wikipathways.org" target="_blank">WikiPathways</a> is a collaborative pathway database that is build on Mediawiki. 
Currently our pathways can be annotated with ontology tags from the pathway ontology, cell type ontology and disease ontology.</p> 
<p><b>Goal</b>: The advanced search should provide an interface to search for pathways containing one or more pathway 
elements (genes, proteins, metabolites), for pathways that are annotated with specific ontology tags or pathways that contain 
a specific interactions. The advanced search allows users to filter pathways based on species, curation tags, authors, 
creation date, number of edits, etc.</p>
<p><b>Technology and Skills</b>: WikiPathways, PHP, Mediawiki</p>
<p><b>Potential Mentors</b>: Martina Summer-Kutmon, Alex Pico</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 35: SIR model for CentiScaPe</h2>
<p><b>Background</b>: <a href="http://apps.cytoscape.org/apps/centiscape" target="_blank">CentiScaPe</a> is a Cytoscape app 
that allows to compute network centrality parameters. It is the most 
complete Cytoscape app for network centralities calculating Degree, Average Shortest Path, Eccentricity, Closeness, 
Betweenness, Centroid, Stress, Radiality, Eigenvector, Edge Betweenness, Bridging Centrality, for directed and undirected 
networks. CentiScaPe allows identifying network nodes that are relevant from both experimental and topological viewpoints. 
CentiScaPe also provides a Boolean logic- based tool that allows easy characterization of nodes whose topological relevance 
depends on more than one centrality. Finally, different graphic outputs and the included description of biological 
significance for each computed centrality facilitate the analysis by the end users not expert in graph theory, thus allowing 
easy node categorization and experimental prioritization.</p>
<p><b>Goal</b>: CentiScaPe can provide only static analysis of a network. We want to develop a tool that can compute 
semi-dynamic simulation, particularly the <a href="http://www.ncbi.nlm.nih.gov/pubmed/2059741" target="_blank">SIR model</a>. The SIR model is a Compartmental model used in epidemiology to 
simulate the establishment and spread of infectious diseases. It serves as a base mathematical framework for understanding the 
complex dynamics of such system. Population is divided in three states: susceptible to the infection of the pathogen (often 
denoted by S), infected by the pathogen (given the symbol I) and recovered/removed/immune (denoted R). The way that these 
compartments interact is often based on a network model, hence the idea to include the SIR model in the CentiScaPe tool for 
network analysis</p>
<p><b>Technology and Skills</b>: Cytoscape, CentiScaPe, Java</p>
<p><b>Potential Mentors</b>: Giovanni Scardoni (CBMC, University of Verona)</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 36: Random networks for CentiScaPe</h2>
<p><b>Background</b>: <a href="http://apps.cytoscape.org/apps/centiscape" target="_blank">CentiScaPe</a> is a Cytoscape app 
that allows, by using a very plain interface, to compute different network 
centrality parameters in single and in multiple networks at the same time. It allows to plot the results by using real and 
normalized values in different plots. Several centralities are available: Degree, Average Shortest Path, Eccentricity, 
Closeness, Betweenness, Centroid, Stress, Radiality, Eigenvector, Edge Betweenness, Bridging Centrality. Each one could be 
computed for directed and undirected networks; CentiScaPe gives the possibility to assign a weight to the edges and, currently, 
we added a new feature that permits to assign a weight to each node by using an attribute.</p>
<p><b>Goal</b>: The new version permits an in-depth analysis of biological networks, from the topological point of view. The 
main issue with this methodology is that it requires a lot of literature mining in order to validate and verify the results 
that, in general, remain speculative. In order to overcome this limitation we decided to add a feature that allows to validate
the results by using random networks, i.e. by adding a sort of statistical evaluation. By randomizing a network we expect to 
highlight all the differences between the real and the random process[1]: this step gives strength to all the findings that 
come from a topological analysis and adds a statistical background that could integrate the biological information from 
experimental data or text mining.</p>
<p>We will integrate CentiScaPe with some algorithms that were already developed in an existing, but no more supported, app
for the Cytoscape platform, version 2.6, that is RandomNetworks [2][3]. We will develop some algorithms that are able to generate 
networks by following a specific model: Watt-Strogatz, Erdos-Renyi, Barabasi-Albert and some others. Another interesting 
feature will be the degree-preserving algorithm that is able to randomize the edges by keeping the degree of each node fixed. 
Finally the comparison between the centralities before and after the randomization could be analyzed by using the topological 
measures of CentiScaPe.</p>
<p>[1] <a href="http://www.ncbi.nlm.nih.gov/pubmed/12399590" target="_blank">Network Motifs: Simple Building Blocks of Complex Networks. Milo et al, Science 25 October 2002.</a></p>
<p>[2] <a href="http://apps.cytoscape.org/apps/randomnetworks" target="_blank">RandomNetworks App</a></p>
<p>[3] <a href="https://sites.google.com/site/randomnetworkplugin/" target="_blank">RandomNetworks Project Website</a></p>
<p><b>Technology and Skills</b>: Cytoscape, CentiScaPe, Java</p>
<p><b>Potential Mentors</b>: Giovanni Scardoni (CBMC, University of Verona)</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 37: Implement support for the SBML Multistate/Multicomponent Species package</h2>
<p><b>Description</b>: One of the <a href="http://sbml.org/Community/Wiki#SBML_Level_3_Status_Summary_Table" target="_blank">many packages</a> 
for SBML Level 3 is <a href="http://sbml.org/Community/Wiki/SBML_Level_3_Proposals/Multistate_and_Multicomponent_Species" target="_blank">
Multistate and multicomponent species</a>. This package 
defines constructs for models and modelers to represent biochemical species that have internal structure or state properties. 
These may involve molecules that have multiple potential states, such as a protein that may be covalently modified, and 
molecules that combine to form heterogeneous complexes located among multiple compartments. The JSBML team has already 
initiated an implementation of the multi package, which will be a useful starting point for this project.</p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java, some exposure to biochemistry.</p>
<p><b>Potential Mentors</b>: Nicolas Rodriguez, Nicolas Le Novère</p>
<p><b>Contact</b>: <a href="mailto:rodrigue@ebi.ac.uk">Nicolas Rodriguez</a>, <a href="mailto:nicolas.lenovere@babraham.ac.uk">Nicolas Le Novère</a></p> 
<br />

<h2 dir="ltr" >Idea 38: Add support for Schema-based validation of SBML</h2>
<p><b>Description</b>: SBML files need to be validated carefully to ensure that they conform to the specification. Currently, 
the most complete implementation of SBML validation is embodied in libSBML, although the rules of SBML validity are defined 
in the SBML specification documents. It is possible to validate SBML from JSBML using either the 
<a href="http://sbml.org/Facilities/Validator/" target="_blank">Online SBML Validator</a> or a 
Java package we provide for calling libSBML locally (i.e., without a network connection) but we want to move toward capturing 
all of the SBML's validity rules in schema languages such as <a href="http://relaxng.org/" target="_blank">RELAX NG</a> and 
<a href="http://www.schematron.com/" target="_blank">Schematron</a>, then having both libSBML and JSBML (and 
any other SBML-using system) use schema validation engines instead of hardcoded validation. This will be especially important 
as more SBML Level 3 packages become implemented. We have already started to define the RELAX NG schemas for SBML Level 3, but 
we need to implement all of the validations rules for SBML core and the different packages and provide the hooks in JSBML to 
using those schemas for validating SBML files.</p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015#" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java, XML, RELAX NG, Schematron, SBML</p>
<p><b>Potential Mentors</b>: Sarah Keating, Andreas Dräger</p>
<p><b>Contact</b>: <a href="mailto:skeating@ebi.ac.uk">Sarah Keating</a>, <a href="mailto:andraeger@eng.ucsd.edu">Andreas Dräger</a></p> 
<br />

<h2 dir="ltr" >Idea 39: Representation and evaluation of mathematical formulas</h2>
<p><b>Description</b>: JSBML uses the concept of abstract syntax trees to work with mathematical expressions. In the Google 
Summer of Code 2014, a new math package was implemented that captures all different kinds of tree nodes that can occur in 
formulas (e.g., real numbers or algebraic symbols such as 'plus' or 'minus') an own, specialized class, i.e., a specific 
abstract syntax tree node. In this way, the handling of formulas has become much more straightforward and even more efficient. 
However, this new package requires some adaption of dependent parts of JSBML. This project focuses on the optimization of the 
performance of mathematical operations when used in large-scale simulations, as this is, for instance, done by the 
<a href="http://simulation-core.sourceforge.net/" target="_blank">Systems 
Biology Simulation Core Library</a>. When parsing infix formulas JSBML relies on the Java Compiler Compiler (JavaCC) project to 
create abstract syntax trees. With the availability of the new math package, some adaptation of existing grammar files for 
the JavaCC parser generator will also be required and will be part of this project.</p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015#" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java, <a href="https://javacc.java.net/" target="_blank">JavaCC</a></p> 
<p><b>Potential Mentors</b>: Victor Kofia, Andreas Dräger, Alex Thomas, Sarah Keating</p>
<p><b>Contact</b>: <a href="mailto:victor.kofia@gmail.com">Victor Kofia</a>, <a href="mailto:andraeger@eng.ucsd.edu">Andreas Dräger</a>, 
     <a href="mailto:althomas@ucsd.edu">Alex Thomas</a>, <a href="mailto:skeating@ebi.ac.uk">Sarah Keating</a></p> 
<br />

<h2 dir="ltr" >Idea 40: Improving the plugin interface for CellDesigner</h2>
<p><b>Description</b>: One of the most frequently used programs in computational systems biology is 
<a href="http://celldesigner.org/" target="_blank">CellDesigner</a>. JSBML 
provides an interface that facilitates the development of plugins for this program. This interface has recently been revised 
and improved. Now, test cases and plugins for CellDesigner are to be implemented in order to make use of it and ensure its 
correct behavior. It is, for instance, possible to use CellDesigner's complex canvas user interface to create or manipulate 
biochemical networks and to conduct numerical computation. Complex drawing styles for biological networks can be encoded in 
SBML through its rendering extension. JSBML's current plugin interface to CellDesigner does, however, not support the 
rendering extension. In this project, important rendering styles that are used in CellDesigner should be implemented in the 
plugin interface.</p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015#" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java, some basic understanding of visualization algorithms</p> 
<p><b>Potential Mentors</b>: Akira Funahashi, Ibrahim Y. Vazirabad, Andreas Dräger</p>
<p><b>Contact</b>: <a href="mailto:funa@bio.keio.ac.jp">Akira Funahashi</a>, <a href="mailto:yvazirabad@gmail.com">Ibrahim Y. Vazirabad</a>,
     <a href="mailto:andraeger@eng.ucsd.edu">Andreas Dräger</a></p> 
<br />

<h2 dir="ltr" >Idea 41: Tests for JSBML</h2>
<p><b>Description</b>: JSBML class and method structure largely corresponds with 
<a href="http://sbml.org/Software/libSBML" target="_blank">libSBML</a>, our C++ language counterpart. One area that will need 
to be standardized between the two parsing libraries is the test cases for proper SBML syntax. LibSBML has a set of files 
(and corresponding tests) which test a large set of these SBML syntax rules. This project will require the programmer to 
create important tests for core and extension libraries for JSBML which correspond to the libSBML rules. The programmer could 
also suggest further tests, and he/she would be essential in building the code base that enacts quality control for JSBML.</p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015#" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java</p> 
<p><b>Potential Mentors</b>: Andreas Dräger, Alex Thomas</p>
<p><b>Contact</b>: <a href="mailto:andraeger@eng.ucsd.edu">Andreas Dräger</a>, <a href="mailto:althomas@ucsd.edu">Alex Thomas</a></p> 
<br />

<h2 dir="ltr" >Idea 42: Flattening of hierarchically structured models</h2>
<p><b>Description</b>: The <a href="http://sbml.org/Documents/Specifications/SBML_Level_3/Packages/comp" target="_blank">hierarchical modeling package</a> 
enables SBML files to be composed of multiple individual submodels. These submodels can be stored in the same file, or be 
distributed in a file system, even be linked through resource identifiers. Hence, comp allows models in SBML to be similarly 
composed as HTML documents. While this format adds a large number of benefits and freedom to the model creator, it also 
significantly increases the complexity of the SBML documents. In order to run regular analysis methods on those composite 
models, a flattening procedure, i.e., a conversion into one large single model, is prerequisite. The specification of the comp 
package describes how such a flattening algorithm can work. The aim of this project is now to implement such a flattening 
routine in JSBML in order to create an SBML file without comp. An additional goal of this project is the implementation of 
basic validation rules for the comp package.</p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015#" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java</p> 
<p><b>Potential Mentors</b>: Lucian Smith, Chris Myers, Leandro Watanabe</p>
<p><b>Contact</b>: <a href="mailto:lucianoelsmitho@gmail.com">Lucian Smith</a>, <a href="mailto:myers@ece.utah.edu">Chris Myers</a>,
     <a href="mailto:l.watanabe@utah.edu">Leandro Watanabe</a></p> 
<br />

<h2 dir="ltr" >Idea 43: Better support for constraint-based modeling</h2>
<p><b>Description</b>: One popular modeling technique in systems biology is 
<a href="http://cobramethods.wikidot.com/" target="_blank">Constraint-Based Reconstruction and Analysis (COBRA)</a>. 
<a href="http://sbml.org/Documents/Specifications/SBML_Level_3/Packages/fbc" target="_blank">The Flux Balance Constraints</a>
(fbc) package has been defined in order to represent the specific requirements of this modeling technique in SBML. The 
<a href="http://opencobra.github.io/" target="_blank">COBRA toolbox</a> for <a href="http://www.mathworks.com/products/matlab/" target="_blank">MATLAB</a> 
and its <a href="https://www.python.org/" target="_blank">Python</a> counterpart <a href="https://github.com/opencobra/cobrapy" target="_blank">COBRApy</a> 
belong to the most widely used programs for this modeling approach. Since these programs preceded the development of fbc, an 
SBML dialect has evolved. The aim of this project is to better support constraint-based modeling in JSBML. To this end, a 
converter from COBRA's SBML dialect to fbc needs to be implemented. This also includes an update of the fbc package implementation in JSBML to the next version.</p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015#" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java</p> 
<p><b>Potential Mentors</b>: Alex Thomas, Andreas Dräger</p>
<p><b>Contact</b>: <a href="mailto:althomas@ucsd.edu">Alex Thomas</a>, <a href="mailto:andraeger@eng.ucsd.edu">Andreas Dräger</a></p> 
<br />

<h2 dir="ltr" >Idea 44: Converter of CellDesigner's annotations to pure SBML</h2>
<p><b>Description</b>: One very popular software in systems biology is <a href="http://celldesigner.org/" target="_blank">CellDesigner</a>. 
Long before the SBML extensions for 
 <a href="http://sbml.org/Documents/Specifications/SBML_Level_3/Packages/layout" target="_blank">Layout</a> and 
<a href="http://sbml.org/Documents/Specifications/SBML_Level_3/Packages/render" target="_blank">Render</a> have 
been defined, this program already provided mechanisms to display network diagrams. To this end, the 
developers of CellDesigner have created their specific SBML dialect. In this project, a parser should be created that 
understands CellDesigner's SBML dialect and converts it to SBML with layout and render extension without the need to use a 
CellDesigner plug-in. If time permits, it would also be nice to have as well a converter from SBML with layout and render to 
<a href="http://www.sbgn.org/LibSBGN" target="_blank">SBGN-ML</a> and vice-versa. Those could be integrated into the 
<a href="http://sourceforge.net/projects/sbfc/" target="_blank">Systems Biology Format Converter (SBFC)</a></p>
<p>More information about the SBML and JSBML projects at <a href="http://sbml.org/GSoC2015#" target="_blank">SBML.org</a>.</p>
<p><b>Technology and Skills</b>: JSBML, SBML, Java</p> 
<p><b>Potential Mentors</b>: Akira Funahashi, Andreas Dräger</p>
<p><b>Contact</b>: <a href="mailto:funa@bio.keio.ac.jp">Akira Funahashi</a>, <a href="mailto:andraeger@eng.ucsd.edu">Andreas Dräger</a></p> 
<br />

<h2 dir="ltr" >Idea 45: Spatial Analysis of Functional Enrichment in Biological Networks</h2>
<p><b>Background</b>: Biological networks connect genes, proteins and other biological entities that interact with one other 
to perform their function. These networks tend to have a non-random structure and often form densely connected structures 
(clusters) that involve closely related elements. Depending on the nature of the network, these clusters may correspond to, 
for example, genes acting in the same biological pathway or proteins sharing the same evolutionary ancestor. Identifying 
clusters and uncovering their biological role is an important and challenging aspect of understanding biological networks and 
making use of the information they contain.</p>
<p>To address this challenge directly, we have developed an algorithm to test for local functional enrichment in biological 
networks. Here, functional enrichment indicates statistical overrepresentation of members of a known functional group within 
a particular area of the network. Using this method we are able to quickly and automatically annotate a biological network 
without defining cluster boundaries. This method, referred to as Spatial Analysis of Functional Enrichment (SAFE), is 
currently being used to understand the structure of genetic interaction networks in yeast Saccharomyces cerevisiae (figure 
attached).</p>
<p><b>Goal</b>: To make a SAFE app for Cytoscape that would allow researchers to run the SAFE method on any network of 
their interest and/or with any set of functional annotations. Specifically, the app should:
     <ul>
          <li>Use the pre-loaded network</li>
          <li>Load <a href="http://www.geneontology.org" target="_blank">Gene Ontology</a> or a custom set of functional annotations (e.g., a list of node 
          labels with categorical assignment to any number of groups)</li>
          <li>Perform functional enrichment on the network using the annotations provided (the algorithm will be available 
          in Matlab and/or in pseudo-code)</li>
          <li>Visually represent the enrichment results on the network (the algorithm will be available in Matlab and/or 
          in pseudo-code)</li>
          <li>Provide the enrichment results in a list form for browsing and/or export as text file</li>
     </ul>
</p>
<p><b>Links</b>: <a href="http://www.baryshnikova-lab.org" target="_blank">Baryshnikova lab</a></p>
<p><b>Technology and Skills</b>: Cytoscape, Java, basic understanding of biological networks</p> 
<p><b>Potential Mentors</b>: Anastasia Baryshnikova (Princeton), Dmitriy Gorenshteyn (Princeton)</p>
<p><b>Contact</b>: <a href="mailto:abarysh@princeton.edu">Anastasia Baryshnikova</a></p> 
<br/ >

<h2 dir="ltr" >Idea 46: Cytoscape.js quadtree support</h2>
<p><b>Background</b>: Cytoscape.js is a JS library that allows for the analysis and visualization of graphs (a.k.a. networks).  
This project would give the opportunity for the student to work on and with Cytoscape.js, as follows:</p>
<p><b>Goal</b>: A <a href="https://en.wikipedia.org/wiki/Quadtree" target="_blank">quadtree</a> is a data structure that 
allows for efficient lookup of sets of spatial co-ordinates. For Cytoscape.js, this could be applied to element (node and 
edge) hit tests: The quadtree could be queried to determine the closest element to the cursor/mouse or finger in an efficient 
way. It might not be warranted for many use cases because of the additional cost of maintaining the tree, but it could be 
useful in other cases. This project need not be constrained to quadtrees specifically, as a different particular tree 
structure may be more appropriate.</p>
<p><b>Technology and Skills</b>: Cytoscape.js, JavaScript</p> 
<p><b>Potential Mentors</b>: Max Franz, Christian Lopes</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 47: Scrollbar in Cytoscape.js</h2>
<p><b>Background</b>: Cytoscape.js is a JS library that allows for the analysis and visualization of graphs (a.k.a. networks).  
This project would give the opportunity for the student to work on and with Cytoscape.js, as follows:</p>
<p><b>Goal</b>: This project would entail the creation of a Cytoscape.js extension that would show scrollbars in a manner 
similar to that of touch devices; i.e. the scrollbar would show momentarily on scroll movement and fadeout thereafter. This 
extension would be useful for showing the relative panning position of the graph in an economical way with regards to 
computation expense and screen space. You would have to create new scrollbar logic and possibly also packaging code
as a reusable extension.</p>
<p><b>Links</b>: </p>
<ul>
     <li><a href=" http://iscrolljs.com/" target="_blank">iScroll</a></li>
     <li><a href="https://github.com/cytoscape/cytoscape.js-qtip" target="_blank">qTip extension</a></li>
</ul>
<p><b>Technology and Skills</b>: Cytoscape.js, JavaScript</p> 
<p><b>Potential Mentors</b>: Max Franz, Christian Lopes</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 48: PubMANIA: Advanced literature search using network inference</h2>
<p><b>Background</b>: Finding all the relevant literature when researching a topic is a common problem. There are many 
potentially useful methods for finding relevant works, including: using search terms, looking through papers that cite a 
publication of particular interest, and looking at papers written by a specific researcher. However, all of these approaches 
can be extremely slow, and will often miss many relevant works. A similar problem exists in genetics; often, you have a set 
of related genes and want to find all the genes that are related to these genes in the same way. 
<a href="http://www.genemania.org/" target="_blank">GeneMANIA</a> solves this 
problem by integrating different types of gene-gene interaction networks, learning what networks are most important in 
connecting your set of genes, and suggesting new genes that appear to be related based on the network structure.</p>
<p><b>Goal</b>: We would like to apply the GeneMANIA approach to publications, creating a service that can recommend 
publications to the user based on a set of provided publications. This requires that we extend the GeneMANIA codebase to 
handle extremely large, sparsely-connected networks, and apply this to our collection of publication-interaction networks 
(based on <a href="http://www.ncbi.nlm.nih.gov/pubmed" target="_blank">PubMed</a>). By integrating networks of citations, 
journals, authors, and common terms, we hope to assist researchers in quickly and easily finding all of the publications 
relevant to their work.</p>
<p><b>Technology and Skills</b>: Java, SQL, Javascript, Data Structures and Algorithms</p>
<p><b>Potential Mentors</b>: Carl de Boer (the Broad Institute)</p>
<p><b>Contact</b>: <a href="mailto:cgdeboer@broadinstitute.org">Carl de Boer</a></p>
<br />

<h2 dir="ltr" >Idea 49: Analyzing bio-network dynamics on the web</h2>
<p><b>Background</b>: <a href="http://apps.cytoscape.org/apps/animo">ANIMO</a> is a Cytoscape App to analyse the dynamic 
behavior of biological networks. As an example of its usefulness, consider a biologist studying the signaling events of a 
cell. ANIMO is built to help answer questions like "If I now give this stimulus to the cell, what happens? Which pathways get 
activated?". ANIMO can show how signals travel along the pathways represented by Cytoscape networks: nodes are colored to 
reflect the activity changes during the course of a simulated experiment. Graphs and answers to non-trivial questions can 
also be obtained, and all of this is done without the need to understand the complexities of mathematical modeling.
However, the power of ANIMO comes at a cost: an additional tool needs to be installed in order to automatically analyse the 
network models. Some biologists are not at home with software requirements, so it would be much easier if the biologist could 
access their tools directly from a web page, without the need to install anything.
Cytoscape provides a way to export networks as (static) web pages, thanks to the features of Cytoscape.js. It would be great 
if these web pages could be extended with the dynamic information generated by ANIMO. Things would get even more interesting 
if the biologist could analyse their network models through the same web interface.</p>
<p><b>Goal</b>: Convert ANIMO into a web-app based on Cytoscape.js, making the most useful features of ANIMO easily accessible 
from any web-enabled device without requiring the installation of additional software.</p>
<p><b>Technology and Skills</b>: Cytoscape.js, JavaScript, Cytoscape, Java, basic client-server interactions</p>
<p><b>Potential Mentors</b>: Stefano Schivo, Rom Langerak, Max Franz</p>
<p><b>Contact</b>: <a href="mailto:s.schivo@ewi.utwente.nl">Stefano Schivo</a></p>

<h2 dir="ltr" >Idea 50: New Layout for Cytoscape.js and esyN Optimised for Large Networks</h2>
<p><b>Background</b>: <a href="http://www.esyn.org">esyN</a> is a web tool to facilitate the exchange of biological network 
models between researchers. esyN acts as a searchable database of user-created networks from any field. esyN aims at making 
it very easy for everybody to build biological networks. Users can upload a list of gene and retrieve interactions from 
Biogrid to populate the network. One major limitation at the moment is represented by the ability of the current layouts to 
display medium and large networks. </p>
<p><b>Goal</b>: Implement a new cityscape.js layout optimised for large networks but yet is fast enough to run on the user’s 
browser. </p>
<p><b>Technology and Skills</b>: Cytoscape.js, JavaScript</p>
<p><b>Potential Mentors</b>: Dan Bean, Giorgio Favrin</p>
<p><b>Contact</b>: <a href="mailto:gf247@cam.ac.uk">Giorgio Favrin</a></p>
<br />

<h2 dir="ltr" >Idea 51: Cytoscape App for Downloading and Uploading esyN Networks</h2>
<p><b>Background</b>: <a href="http://www.esyn.org">esyN</a> is a web tool to facilitate the exchange of biological network 
models between researchers. esyN acts as a searchable database of user-created networks from any field. esyN allows the user 
to easily construct networks by uploading list of gene and retrieving their interactions from BioGRID.  In addition to its 
basic tools, esyN contains a number of logical templates that can be used to create models more easily. The ability to use 
previously published models as building blocks makes esyN a powerful tool for the construction of models and network graphs. 
Users are able to save their own projects online and share them either publicly or with a list of collaborators. </p>
<p><b>Goal</b>: Implement a Cytoscape app to be able to download esyN networks (both published and private ones) into 
cytoscape and vice versa upload cytoscape networks in esyN.</p>
<p><b>Technology and Skills</b>: Cytoscape, Java</p>
<p><b>Potential Mentors</b>: Dan Bean, Giorgio Favrin, Giovanni Scardoni</p>
<p><b>Contact</b>: <a href="mailto:gf247@cam.ac.uk">Giorgio Favrin</a></p>
<br />
<h2 dir="ltr" >Idea 52: Enhancing Cytoscape Network Data Import</h2>
<p><b>Background</b>: <a href="http://www.cytoscape.org">Cytoscape</a> is a network analysis and visualization tool. Users 
commonly want to import data files that contain both network interactions and node annotations. Currently, you have to invoke
two different import dialogs multiple times to accomplish this task.</p>
<p><b>Goal</b>: The network import dialog and file processing should be overhauled and enhanced to provide a single interface
to perform network and node data import. After choosing a source file, the UI would present an interactive preview (like it 
does currently) with additional features like specifying columns that are associated with either source or target nodes, in 
addition to edge (interaction) annotation columns. The execution of the dialog would then trigger network and data table 
import tasks.  There are many subtle aspects to this project:
     <ul>
          <li>Designing UI and UX</li>
          <li>Using Cytoscape Tasks to support command line control</li>
          <li>Improving logic behind data type and list handling</li>
          <liImplementing graceful fail states, e.g., friendly UX for fixing a minor error and trying again></li>
     </ul>
And there are many extensions to this project that might be proposed:
<ul>
 <li>Improvements to the data table importer</li>
 <li>Expanding supported file formats</li>
 <li>Batch import of multiple files sharing a common column structure</li>
 <li>Persistent UI settings within a session</li>
</ul>
</p>
<p><b>Technology and Skills</b>: Cytoscape, Java, Swing</p>
<p><b>Potential Mentors</b>: Scooter Morris, Alex Pico, Christian Lopes</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 53: Implement a PathLinker Annotation Plugin for PathVisio</h2>
<p><b>Background</b>: Cells respond to their environment through a series of molecular interactions that form biological 
pathways; understanding these pathways is a major area of research in Systems Biology. PathLinker is an algorithm that 
reconstructs biological pathways from experimental data, and it has been very useful in annotating existing biological 
pathways as well as discovering new pathway additions. <a href="http://www.pathvisio.org">PathVisio</a> is free open-source 
software to build, visualize, annotate, and analyze pathways.  It integrates tightly with <a href="http://www.wikipathways.org">WikiPathways</a>, 
a "wikipedia'' for biological pathways. </p>
<p><b>Goal</b>: To implement PathLinker as an annotation plugin for PathVisio.</p>
<p>This plugin will assist a pathway author using who is using PathVisio to create or expand a pathway. When the author 
selects a protein, the plugin will suggest possible physical, regulatory, and signaling interactions involving that protein 
and the literature supporting each interaction. The author will then click or drag the suggestion over into the pathway to 
confirm its inclusion. The plugin will collect statistics on the number of suggestions accepted, ignored, or rejected by 
authors.</p>
<p>The PathLinker graph algorithm is be the technology underlying this plugin. PathLinker takes as input a comprehensive 
network of physical, regulatory, and signaling interactions among proteins. PathLinker also requires a set of "sources" 
(e.g., receptors) and a set of "targets" (e.g., transcription factors) that the pathway author will provide. PathLinker 
computes multiple shortest paths from any source to any target and ranks these paths in increasing order of length. A 
user-defined parameter controls the number of paths computed. If the pathway author asks for (say) 50 paths, then for each 
protein, the plugin will store which interactions in these 50 paths connect with the protein, the ranks of these paths, and 
the literature support for each interaction.</p>
<p><b>Links</b>
     <ul>
          <li><a href="https://github.com/tmmurali/PathLinker/">PathLinker implementation in Python</a></li>
          <li>PathLinker implementation in Java (release planned by the end of May 2015)</li>
     </ul>
</p>
<p><b>Technology and Skills</b>: graph algorithms, Java, PathVisio, WikiPathways</p>
<p><b>Potential Mentors</b>: T. M. Murali and Anna Ritz</p>
<p><b>Contact</b>: <a href="murali@cs.vt.edu">T.M. Murali</a>, <a href="annaritz@vt.edu">Anna Ritz</a></p>
<br />

<!-- END PASTE -->

    </div> <!-- .span9 !-->
</div> <!-- .row !-->

<script src="../assets/js/jquery-1.11.0.min.js"></script>
<script src="../assets/js/jquery-ui.min.js"></script>
<script src="../assets/js/jquery.tocify.min.js"></script>
<script>
$("#nav").tocify({
    context: "#ideas",
    showAndHide: false,
    selectors: "h2",
});
</script>
</body>
</html>
