<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>NRNB: GSoc Ideas Page</title>

<!-- Le styles -->
<link href="../assets/css/bootstrap.css" rel="stylesheet">
<link href="../assets/css/bootstrap-responsive.css" rel="stylesheet">
<link type="text/css" rel="stylesheet" href="../assets/css/jquery.tocify.css" />
<link type="text/css" rel="stylesheet" href="../assets/css/jquery-ui.min.css" />
<style type="text/css">
#nav {
     position: fixed;
}
.tocify ul, .tocify li {
        line-height: 15px;
}
</style>
</head>
<body>
<div class="row" height="100%">
    <div class="span4"  height="100%">
         <!-- Our table of contents will be here !-->
         <div id="nav" style="font-size: 12px;" height="100%"></div>
    </div>
    <div id="ideas" class="span8">

<!-- PASTE HERE -->

<p><br />
</p>

<h2 dir="ltr" >Idea 1: Submit your own original proposal</h2>

<p>Feel free to propose your own idea. As long as it relates to one of our projects, we will give it serious consideration. 
Creativity and self-motivation are great traits for open source programmers, but make sure your proposal is also relevant.</p>

<br />
<h2 dir="ltr" >Idea 2: Co-authorship Social network App</h2>

<p>Description: Biological networks can be visualized and analyzed using Cytoscape. Often researchers want to go beyond the 
network of proteins or genes and also look at the inter-connectedness between colleagues and institutions. Who tends to 
publish together? What institutions are most collaborative? Are there inter-disciplinary connections in my institution?</p>

<br />
<p>Goal: Create a visual summary of how individuals are connected. An existing App currently queries PubMed directly or takes in a file with a set of publications (from scopus or incites) and builds a co-publication network where nodes are authors and edges the number of publications two authors share. Different sources have different depths of data, for instance PubMed, the most widely accessible of the three main sources, has citation counts but they are limited to only those articles cited in pubmed central and its institution data is limited to the first author. Given that PubMed is the most accessible source we want to expand its functionality to include computed institution data as well as better citation count approximations, and H-index calculation for individuals or subsets of papers. </p>

<p>This functionality can be applied to several data types outside academia, whenever relationships between people is of interest. An example is professional social networks (e.g. linked-in), where individuals are connected by who they have worked with in the past or the twitter-verse where individuals are connected when they each reference the same (#) hash tag. This project will start with support for co-publication networks and extend to other networks like twitter and linked-in as time permits. Additionally, we want to expand our co-authorship cytoscape App to include generic file formats to easily create the above networks.</p>

<br />
<p>Links:</p>

<p>Existing code - <a href="https://github.com/BaderLab/SocialNetworkApp" >https://github.com/BaderLab/SocialNetworkApp</a></p>

<p>Existing userguide - <a href="http://baderlab.org/UserguideSocialNetworkApp" >http://baderlab.org/UserguideSocialNetworkApp</a></p>

<br />
<p>Language and Skills: Java, XML, PubMed e-utils</p>

<p>Potential Mentors: Sergio Baranzini, Gary Bader, Ruth Isserlin</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 3: User Interaction on Biomedical Linked Data</h2>

<p>During the GSoC 2012, we developed<a href="http://vsdlc3.blogspot.com" > SemScape</a>, a plugin that allows the use of Cytoscape as a browser for Semantic Web (Linked Data) knowledge bases. We are currently in the process of updating libs and packaging the plugin for the Cytoscape App Store.</p>

<p>The idea for this year’s project is to go one step further and to allow the use of Cytoscape to visually define queries in SPARQL. The basic idea is that from a graph, a sub-graph can be defined where a few nodes or edges can be marked as variables (and eventually filters can be associated to them): this is equivalent to a (simple) SPARQL query. The user can then define queries in a simple and interactive way, without knowing the underlying syntax, that can be stored and applied on different data sources.</p>

<p>The project can then be extended in three directions, depending on the interest of the student:</p>

<ul>
<li><p>allowing users to push such queries to a repositories and hence share them (the current plugin support updates of queries from remote repositories)</p>

</li>
<li><p>Implementing more complex queries (unions, negation, …)</p>

</li>
<li><p>Implementing heuristics for query-selection: in the current plugin, queries can be performed on demand, but they are also associated to specific actions. For instance, right-selecting a node of type protein can present as an option to expend the graph with the results of a query which is can include the specific protein as a value. This is possible through a simple markup through which queries are annotated by the user. Such markup could be included in the visual query/construction process and/or extracted from simple query patterns.</p>

</li>
</ul>

<br />
<p>Some of the ideas in this project<a href="http://www.biomedcentral.com/1471-2105/9/S4/S6" > have been already prototyped</a>, so we have a good understanding of the difficulty and timing required. </p>

<p>This project has the potential to make Cytoscape an ideal tool to navigate and query Linked Data resources (both in biomedical and, more generally, in Open Data domains). It also opens interesting possibilities in analyzing graphs from graph knowledge bases that can be represented as RDF (e.g.: dbpedia, freebase,...).</p>

<br />
<p>Language and Skills: Java/Maven. Knowledge of OSGi would be helpful. Knowledge of Semantic Web technologies would be a plus.</p>

<p>Potential Mentors: Andrea Splendiani, Andra Waagmeester</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 4: CentiScaPe for multiple networks analysis and graphical output enhancement</h2>

<p>Background: CentiScaPe is a Cytoscape plugin that allows to compute network centrality parameters. It is the most complete Cytoscape plugin for network centralities calculating Degree, Average Shortest Path, Eccentricity, Closeness, Betweenness, Centroid, Stress, Radiality, Eigenvector, Edge Betweenness, Bridging Centrality, for directed and undirected networks. CentiScaPe allows identifying network nodes that are relevant from both experimental and topological viewpoints. CentiScaPe also provides a Boolean logic- based tool that allows easy characterization of nodes whose topological relevance depends on more than one centrality. Finally, different graphic outputs and the included description of biological significance for each computed centrality facilitate the analysis by the end users not expert in graph theory, thus allowing easy node categorization and experimental prioritization.</p>

<br />
<p>Goal: The new beta version of CentiScaPe provides also the possibilities to analyze multiple networks and to integrate the analysis with experimental data set. This implies that a user can automatically start a computation of several networks and waits for the results. Anyway, the output of the analysis of a lot of networks (say 100) is useless if not supported by a smart and user friendly system to analyze the results. The Idea is to implements some techniques used in software analysis to extract information from the results of multiple computation in order to compare different networks and to extract topological based biomarkers:</p>

<ul>
<li><p>The main framework for the results analysis is the Abstract Interpretation theory, particularly the interval analysis (and eventually others) [1].</p>

</li>
<li><p>Some statistical parameters, automatic analysis of the output, and new kinds of graphical outputs that can help to extract information from a multiple networks centrality analysis should be provided. The tools should for example be able to answer to questions as “How does the centroid value of the protein A chenge in the submitted 100 different networks?” or “How does the betweenness of protein B change if we consider 30 different phosporilation values for each proteins in the network N (this means we have 30 different experimental data set for the same network that must be included in the computation)?”.</p>

</li>
<li><p>The tools should also provide the implementation of the integration of experimental data-sets for the eigenvector, bridging centrality and edge betweenness (they are now available for all the other centralities).</p>

</li>
</ul>

<br />
<p>[1] Patrick Cousot and Radhia Cousot. 1977. Abstract interpretation: a unified lattice model for static analysis of programs by construction or approximation of fixpoints. In Proceedings of the 4th ACM SIGACT-SIGPLAN symposium on Principles of programming languages (POPL ‘77). <a href="http://doi.acm.org/10.1145/512950.512973" >http://doi.acm.org/10.1145/512950.512973</a></p>

<br />
<a href="http://doi.acm.org/10.1145/512950.512973" ></a><p>Language and Skills: Java, software analysis (possibly abstract interpretation theory)</p>

<p>Potential Mentors: Giovanni Scardoni (CBMC, University of Verona), Fausto Spoto (University of Verona)</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 5: Improving Cytoscape.js Based Viewer for SBGN Process Description Diagrams with Better Layout and 
Advanced Complexity Management Operations</h2>

<p><b>Background</b>:
Cytoscape.js is a JavaScript graph library for analysis and visualization of relational information [1]. The Systems Biology 
Graphical Notation (SBGN) is a visual language developed by the community with the intention of enabling scientists to 
represent networks, including models of cellular processes, in a standard, unambiguous way [2]. SBGN comprises three 
complementary languages suited to represent different types of pathways. SBGN Process Descriptions is a language that permits 
the description of the processes (e.g. biochemical reactions) taking place in a biological system and makes heavy use of 
compound structures (e.g. compartments and molecular complexes). 
Often times, SBGN PD networks are too large to visualize and details become obscured. Strategies to conceal the "unwanted" 
details of a network, allowing one to temporarily remove certain nodes and edges and subsequently reintroduce the removed 
details. Complexity management techniques in graph visualization such as filtering, hiding, folding, and nesting play an 
important role in effective analysis of biological networks as much as any other domain.
Last year, support for compound structures has been added to Cytoscape.js core and an SBGN viewer named SBGNViz has been 
built [3]. Some complexity management operations have also been designed and implemented within this tool to hide unwanted 
details on demand without violating SBGN PD network invariants (e.g. all substrates and products of a process must be shown 
at any point).</p>

<p><b>Goal</b>:
This year, we'd like to improve this tool in certain ways. First, it'd be great to improve the layout implementation that 
supports compound graphs. Secondly, existing complexity management operations could be extended to include folding/unfolding. 
Notice however that this is not a straightforward topology change. For instance collapsing a compound node into a single "meta" 
node requires calculation of any inter-graph edges going into the compound node and representing them with "meta" edges in the 
collapsed network. Similarly, expanding a collapsed node back requires undoing it. The situation becomes cumbersome since 
arbitrary level of nesting is allowed and folding can take place at an arbitrary level.</p>

<p>[1] <a href="http://cytoscape.github.io/cytoscape.js/" >http://cytoscape.github.io/cytoscape.js/</a></p>

<p>[2] <a href="http://www.sbgn.org/" >http://www.sbgn.org/</a></p>

<p>[3] <a href="https://code.google.com/p/sbgnviz-js/">https://code.google.com/p/sbgnviz-js/</a></p>

<p><b>Language and Skills</b>: JavaScript, Java, Data Structures and Algorithms</p>

<p><b>Potential Mentors</b>: Ugur Dogrusoz (Bilkent University & MSKCC) and S. Onur Sumer (MSKCC)</p>

<p><b>Contact</b>: <a href="mailto:ugur@cs.bilkent.edu.tr" >ugur@cs.bilkent.edu.tr</a> and <a href="mailto:onur@cbio.mskcc.org" >onur@cbio.mskcc.org</a> </p>

<br />
<h2 dir="ltr" >Idea 6: Advanced data visualization in PathVisio</h2>

<p><a href="http://pathvisio.org/" >PathVisio</a> is a pathway editor and analysis tool. PathVisio allows users to upload and visualize data on biological pathways. We want to extend the visualization module in PathVisio which is part of the PathVisio core application. A central visualization registry should allow plugins to create and register new visualization options. The current visualization interface in PathVisio is very complex and should be made more user-friendly and intuitive for the users. </p>

<br />
<p>Goals:</p>

<p>- Independent visualization module and registry</p>

<p>- More user-friendly visualization interface</p>

<p>- Possibility for users to define visualization templates that can be used for different datasets (currently a visualization is linked to a dataset and has to be created again for a new dataset)</p>

<br />
<p>Language and Skills: Java, OSGi</p>

<p>Potential Mentors: Martina Kutmon, Anwesha Bohler (Unimaas, NL)<br />
Contact: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 7: Develop a Method for Aligning Pathways and Visualizing these Alignments</h2>

<p>There is continued growth in the groups generating machine-readable representations of biological pathways (such as <a href="http://www.reactome.org/" >Reactome</a> or <a href="http://wikipathways.org/" >WikiPathways</a>) and subsequent efforts to aggregate these pathways into unified repositories, such as Pathway Commons. Aggregation of these pathways suffers from multiple pathways being aggregated that differ due to mismatches in author representations of these pathways. The goal of this project is to develop a method for aligning pathways and then visualizing the results of these alignment. Starting points could be to find interactions that use the same substrates and produce the same products. </p>

<br />
<p>There are many uses of such a tool including helping to validate pathways as they are added to repositories, merging pathways across different resources, and the comparative analyses of pathways across organisms. One way to think about this project, is if you're given multiple representations of the same pathway (<a href="http://wikipathways.org//index.php?query=wnt&amp;species=Homo+sapiens&amp;title=Special%3ASearchPathways&amp;doSearch=1&amp;ids=&amp;codes=&amp;type=query" >examples of several human Wnt pathways</a>) can you develop a method to show a unified pathway and how the original representations map to the unified pathway. </p>

<br />
<p>Language and Skills: Java, XML</p>

<p>Potential Mentors: Augustin Luna (MSKCC)</p>

<p>Contact: <a href="mailto:lunaa@cbio.mskcc.org" >lunaa@cbio.mskcc.org</a></p>

<br />

<h2 dir="ltr" >Idea 8: Drag &amp; Drop or Palette Editor for SBGN and other graphical notations in Cytoscape</h2>

<p>The goal is to design and implement a drag-and-drop or palette-based editor for Cytoscape that would include nodes, edges, and graphical annotations. In the ideal case, this would be extensible so that apps add their own nodes or edges with specific graphical attributes. The Systems Biology Graphical Notation (SBGN) is a standard graphical notation for presenting pathways and protein interactions. Currently, Cytoscape lacks the node symbols and edge types required for SBGN. In this project, those notations will be added to the Cytoscape core, and a simple palette editor developed for user interaction.</p>

<br />
<p>Language and Skills: Java</p>

<p>Potential Mentors: Scooter Morri, Alex Pico</p>

<p>Contact: <a href="mailto:scooter@cgl.ucsf.edu" >scooter@cgl.ucsf.edu</a> </p>

<br />
<h2 dir="ltr" >Idea 9: Improve and expand Cytoscape’s color mapping options</h2>

<p>Currently, the color options for Cytoscape’s discrete color mappings are very limited. Brewer palettes provide a luminosity-neutral set of colors that would be extremely useful to add to Cytoscape, and there are several other approaches to choosing sets of colors that contrast nicely. Creating these color palettes and providing them for use within Cytoscape’s visual style interface would be extremely useful to Cytoscape users.</p>

<br />
<p>Language and Skills: Java</p>

<p>Potential Mentors: Scooter Morris, Michael Heuer, Alex Pico</p>

<p>Contact: <a href="mailto:scooter@cgl.ucsf.edu" >scooter@cgl.ucsf.edu</a> </p>

<br />

<h2 dir="ltr" >Idea 10: Constrained layout algorithms in Cytoscape</h2>

<p>Most layout algorithms in Cytoscape rely on inherent network topology and only a crude set of spatial constraints (e.g., grid, circle). There are many cases when working with networks in Cytoscape where you want to be able to constrain layouts to user- or attribute-defined spaces. This idea involves implementing layouts that are constrained by:</p>

<ul>
<li><p>specific shapes </p>

</li>
<ul>
<li><p>basic shapes </p>

</li>
<li><p>cellular templates (e.g., with subcellular organelles)</p>

</li>
<li><p>attribute-driven Venn diagrams (e.g., to visualize attribute groupings and overlaps)</p>

</li>
</ul>

<li><p>specific “anchor” nodes</p>

</li>
<ul>
<li><p>user-defined (e.g., manually selected)</p>

</li>
<li><p>attribute-defined (e.g., hubs or column values)</p>

</li>
</ul>

<li><p>etc...</p>

</li>
</ul>

<br />
<p>Language and Skills: Java, and a strong math background</p>

<p>Potential Mentors: Scooter Morris, Alex Pico, Bruce Aronow</p>

<p>Contact: <a href="mailto:scooter@cgl.ucsf.edu" >scooter@cgl.ucsf.edu</a> </p>

<br />
<h2 dir="ltr" >Idea 11: A SPARQL query plugin for Pathvisio</h2>

<p>Background: The semantic web is becoming a valuable resource for Pathway knowledge. The challenge still remain to smoothly integrate this knowledge into biological frameworks. With their SPARQL endpoints Uniprot (<a href="http://beta.sparql.uniprot.org/" >http://beta.sparql.uniprot.org/</a>) or the EBI rdf Platform (<a href="https://www.ebi.ac.uk/rdf/" >https://www.ebi.ac.uk/rdf/</a>) , biological data is becoming available as semantic web data. To use this knowledge one need not only to learn SPARQL queries, but also be prepared to convert the semantic data in different formats that comes with the various biological platforms. Using semantic web data in Pathvisio, we currently need to compose the appropriate SPARQL queries, get the data and write various parser to change the data in to the different data models. </p>

<br />
<p>Goal: To make a Pathvisio plugin that allows a pathway curator to write SPARQL queries against a variety of semantic web based pathways resources and include the obtained triples into the Pathway under scrutiny. The plugin should also provide an interface to SPARQL results for other Pathvisio plugins. Having such SPARQL interface still requires a pathway editor to become knowledgeable in writing SPARQL queries. The burden of having to write various parsers for semantic web data however, is taken away. The plugin should:</p>

<ul>
<li><p>Provide a SPARQL query editor, from where the user could compose queries, which are then subsequently launched to the respective SPARQL endpoints.</p>

</li>
<li><p>Provide an interface that wrap specific SPARQL queries for other Pathvisio plugins. </p>

</li>
<li><p>Link to a centralized repository (e.g. Github) where SPARQL queries can be stored for future reference and others to use.</p>

</li>
<li><p>Link to a centralized index of relevant SPARQL endpoints. This index might again be stored on platforms like Github. </p>

</li>
<li><p>Import the resulting triples, which can then be linked to recognizable pathway components, or as data elements to use in Pathway analysis. </p>

</li>
<li><p>Make a RDF representation of a Pathway opened in Pathvisio. These triples are to be used as template for SPARQL queries. The user should be able to click on a triple which is then added to the WHERE block in the SPARQL query.</p>

</li>
</ul>

<br />
<p>Potential students: This idea is for a student who has solid background in SPARQL queries, or the Semantic web in general, and who is looking into opening up the Semantic Web to specific knowledge domains (I.e. Biology and/or Medicine). </p>

<br />
<p>Language and Skills: SPARQL, RDF, and Java</p>

<p>Potential Mentors: Andra Waagmeester, Anders Riutta, Martina Kutmon </p>

<p>Contact: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 12: Web publishing of models from Virtual Cell database</h2>

<p>The<a href="http://vcell.org" > Virtual Cell</a> modeling and simulation framework is an open source software for modeling and simulations of biological systems using Java RMI, providing both client- and server- based modeling and simulation capabilities. The biologically oriented user interface allows experimentalists to create models, define cellular geometry, specify simulations and analyze the simulation results. The models and results are stored on a remote server in ORACLE database and presently can be reviewed only in the VCell software. You need to implement a content management system that will search and retrieve model information from ORACLE or MySQL database and publish it online. The model structure stored in a relationship database is available at<a href="http://vcell.org/technology/VcmlSpec.html" > http://vcell.org/technology/VcmlSpec.html</a></p>

<br />
<p>Language and Skills: Java</p>

<p>Possible Mentors: Michael Blinov, Jim Schaff</p>

<p>Contact: <a href="mailto:vcell-discuss@googlegroups.com" >vcell-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 13: New Features for Online Tutorial System</h2>

<p><a href="http://opentutorials.cgl.ucsf.edu" >OpenTutorials</a> is a free, online tutorial resource for open source and NRNB-supported tools. It is built on top of MediaWiki and a handful of extensions. It’s basic function is to provide a wiki-based collection of tutorials that can be viewed as a web page, slide show or printed handout. The wiki approach makes it easy to build and maintain content. The slide show and handout features make it useful for presenters.</p>

<p>There are a number of features that need to be implemented to improve the site and really make it effective to a larger community. What we’re looking for here with this project is an application that is very clear about the use cases of tutorials:</p>

<ul>
<li><p>for developers wanting to create and maintain tutorial materials</p>

</li>
<li><p>for presenters of tutorials</p>

</li>
<li><p>and for users learning from tutorials.</p>

</li>
</ul>

<br />
<p>The application should include a number of features that enhance these use cases. When possible, the features should point to other examples were similar approaches have proven successful (e.g., in other popular software). After we discuss the features, your application should then include implementation details and a timeline. These would be implemented as one or more PHP extensions, for example. Experience or some demonstration of familiarity with MediaWiki or PHP would be a plus. Here are some feature ideas to get you started:</p>

<ul>
<li><p>Mix-and-match: We need a mechanism to allow presenters to mix and match tutorials (or sections of tutorials) into sets that are tailored for their class or workshop audience.</p>

</li>
<li><p>Transient locks: We need a mechanism for temporarily locking content while it is being presented (so that someone isn’t editing a wiki page while you are presenting it to a class of 100 students!). This can be done either by supporting temporary presenter “copies” that are not publicly editable, or simply locks that presenters can set for specific dates and times.</p>

</li>
<li><p>Enhanced image replacement tools to make it easier to update screenshots.</p>

</li>
<li><p>Tutorial builder tools using templates and forms to get a new tutorial started, or an integrated html-to-wiki converter like<a href="http://labs.seapine.com/htmltowiki.cgi" > this</a>.</p>

</li>
<li><p>Tweaks to the wiki page display of header/meta data and of slide breaks (e.g., a clear, yet unobtrusive icon).</p>

</li>
</ul>

<br />
<p>Language and Skills: PHP, MediaWiki</p>

<p>Potential Mentors: Alex Pico, Kristina Hanspers</p>

<p>Contact: <a href="mailto:apico@gladstone.ucsf.edu" >apico@gladstone.ucsf.edu</a></p>

<br />
<h2 dir="ltr" >Idea 14: Custom Export Editor for WikiPathways Using SVG</h2>

<p>Background:<a href="http://wikipathways.org" > WikiPathways</a> is a collaborative pathway editing and sharing platform. There are many ways to use the pathway content including data visualization and analysis. Another use is as illustrations in presentations and publications. We currently provide SVG export for this purpose. However, the images usually require quite a bit of “touch up” in secondary tools like Illustrator, which inhibits this use case.</p>

<p>Goal: Customize a basic SVG Editor to insert into the export process to allow basic visual style updates and custom annotations on top of the pathway image. Pre-configured options would include:</p>

<ul>
<li><p>thicker lines and bold font (for better display in powerpoint)</p>

</li>
<li><p>swap basic cellular component shapes with nice graphics (provided from an existing repository)</p>

</li>
<li><p>color schemes</p>

</li>
</ul>

<p>This simple “touch up” editor could dramatically increase the audience for WikiPathways to include anyone looking for an easy tool to generate nice looking pathway images for presentations. Ideally, this editor would work on tablets as well as across major browsers.</p>

<br />
<p>Language and Skills: SVG,<a href="http://code.google.com/p/svg-edit/" > SVG-edit</a> (or alternative)</p>

<p>Potential Mentors: Alexander Pico, Anders Riutta</p>

<p>Contact: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 15: Integration of WikiPathways Visualization with PathVisio into Bioclipse</h2>

<p>Goal: To integrate PathVisio visualization widget into Bioclipse for visualization of pathway information from WikiPathways. Selecting protein and metabolites in the pathway visualization, will shows of that chemical entity in other Bioclipse components, using the Jmol and JChemPaint plugins for Bioclipse. Preferably, interaction can be scripted, allowing integration with other metabolomics features in Bioclipse.</p>

<br />
<p>Motivation: Bioclipse is a powerful platform combining a powerful scripting language to write data analysis and visualization workflows, which can easily be shared on, for example, MyExperiment.org [1]. While it has various components to support metabolomics in general, it lacks support for visualization of metabolic pathways.</p>

<br />
<p>Description: You will use Java to write a plugin for Bioclipse [2] that uses PathVisio [3] as a library to visualize metabolic pathways from WikiPathways inside Bioclipse. You will use the Eclipse selection event propagation mechanism [4] so that when users select protein and metabolites, Bioclipse will automatically visualize 2D or 3D structural information about those molecules (only the event creation has to be implemented). Preferably, a scripting language extension is written [5], to allow extracting information from the pathway and highlighting of parts of the pathway. This involves extending an existing interface with methods that can be run from the scripting language.</p>

<ol ><li><p><a href="http://myexperiment.org/" >myexperiment.org</a></p>

</li>
<li><p><a href="http://www.bioclipse.net/" >www.bioclipse.net</a></p>

</li>
<li><p><a href="http://www.pathvisio.org/" >www.pathvisio.org</a></p>

</li>
<li><p><a href="http://www.eclipse.org/articles/Article-WorkbenchSelections/article.html" >www.eclipse.org/articles/Article-WorkbenchSelections/article.html</a></p>

</li>
<li><p><a href="http://www.biomedcentral.com/1471-2105/10/397" >www.biomedcentral.com/1471-2105/10/397</a></p>

</li>
</ol><br />
<a href="http://www.biomedcentral.com/1471-2105/10/397" ></a><p>Language and Skills: Java</p>

<p>Potential Mentors: Egon Willighagen, Jonathan Alvarsson</p>

<p>Contact: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 16: Integration of Cytoscape into Bioclipse</h2>

<p>Goal: To integrate the new Cytoscape OSGi visualization widgets into Bioclipse for visualization of large interaction networks. Selecting protein and metabolites in the pathway visualization, will shows of that chemical entity in other Bioclipse components, using the Jmol and JChemPaint plugins for Bioclipse. Preferably, interaction can be scripted, allowing integration with other metabolomics features in Bioclipse.</p>

<br />
<p>Motivation: Bioclipse is a powerful platform combining a powerful scripting language to write data analysis and visualization workflows, which can easily be shared on, for example, MyExperiment.org [1]. While it has been used in drug discovery and predictive toxicology in pharmaceutical industry [2], it currently does not have functionality to visualize large graph structures as Cytoscape has.</p>

<br />
<p>Description: You will use Java to write a plugin for Bioclipse [3] that uses Cytoscape. No particular graph is aimed at, but the plugin will use the Eclipse selection event propagation mechanism [4] so that when users select protein and metabolites, Bioclipse will automatically visualize 2D or 3D structural information about those molecules (only the event creation has to be implemented). Preferably, a scripting language extension is written, to allow extracting information from the visualized graphs.</p>

<ol ><li><p><a href="http://myexperiment.org/" >myexperiment.org</a></p>

</li>
<li><p><a href="http://pubs.acs.org/doi/abs/10.1021/ci200242c" >pubs.acs.org/doi/abs/10.1021/ci200242c</a></p>

</li>
<li><p><a href="http://www.bioclipse.net/" >www.bioclipse.net</a></p>

</li>
<li><p><a href="http://www.eclipse.org/articles/Article-WorkbenchSelections/article.html" >www.eclipse.org/articles/Article-WorkbenchSelections/article.html</a></p>

</li>
</ol><br />
<a href="http://www.eclipse.org/articles/Article-WorkbenchSelections/article.html" ></a><p>Language and Skills: Java</p>

<p>Potential Mentors: Egon Willighagen, someone from Bioclipse team</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 17: Publish Cytoscape 3 Session to cytoscape.js</h2>

<p><a href="http://www.cytoscape.org/cy3.html" >Cytoscape 3</a> and<a href="http://cytoscape.github.com/cytoscape.js/" > cytoscape.js</a> are completely independent software packages; 3 is a Java Desktop application and .js is a JavaScript library to visualize network data. However, it shares some basic concepts, such as Visual Styles. It is very useful if we can reproduce same/similar Cytoscape 3 network views in web browsers to share/publish data sets.</p>

<p>The goal of this project is exporting Cytoscape 3 data as JSON, and reproducing same/similar interactive network views with cytoscape.js and other JavaScript libraries.</p>

<ul>
<li><p>Export Cytoscape 3 data as JSON</p>

</li>
<ul>
<li><p>Networks</p>

</li>
<li><p>Data Tables</p>

</li>
<li><p>Visual Styles</p>

</li>
</ul>

<li><p>This is a JAVA programming project. JavaScript knowledge is optional, but Java/OSGi knowledge is REQUIRED. You need to write a lot of Java classes to convert Cytoscape 3 data objects into JSON using Jackson library.</p>

</li>
<li><p><a href="https://github.com/idekerlab/json-impl" >Prototype is available</a></p>

</li>
<li><p>Create a test HTML5 page to visualize Cytoscape 3 network views with cytoscape.js</p>

</li>
<li><p>(Optional) Add import JSON function for Cytoscape 3</p>

</li>
</ul>

<br />
<p>Language and Skills: Java, JavaScript, HTML5, Git/GitHub</p>

<p>Potential Mentors: Kei Ono, Christian Lopes, Max Franz, Gary Bader</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 18: BridgeDb config plugin in PathVisio</h2>

<p><a href="http://www.bridgedb.org" >BridgeDb</a> is an identifier mapping framework used in PathVisio and WikiPathways. Currently it is possible to load a mapping databases for gene products and metabolites. The existing<a href="http://www.pathvisio.org/wiki/BridgeDbConfigPluginHelp" > BridgeDb config plugin</a> provides functionality to load more mapping databases or to use other webservices for the mappings. This functionality should be extended and integrated into the core PathVisio application as another core module. The configuration dialog has to be improved so it is more intuitive and user friendly.</p>

<p>The main features of this module should be:</p>

<ul>
<li><p>user friendly dialog to configure mapping services</p>

</li>
<li><p>centralized module for identifier mapping in PathVisio</p>

</li>
<li><p>system to define which databases are used for which species and/or data node type</p>

</li>
</ul>

<br />
<p>Language and Skills: Java, OSGi, BridgeDb</p>

<p>Potential Mentors: Martina Kutmon, Anwesha Bohler</p>

<p>Contact: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 19: Publication Reference Plugin for PathVisio</h2>

<p>Currently<a href="http://www.pathvisio.org" > PathVisio</a> supports publication reference annotation of pathways, data nodes and interactions with PubMed identifiers. The goal of this plugin is to provide a more advanced framework for reference annotations.</p>

<p>This plugin should provide (can be extended):</p>

<ul>
<li><p>functionality to use other identifiers than pubmed (e.g. DOI) (might be possible to use an existing java library for that (e.g. Jabref))</p>

</li>
<li><p>show all references in a side tab with linkouts</p>

</li>
<li><p>exporting all references in different reference style formats</p>

</li>
</ul>

<br />
<p>Language and Skills: Java, OSGi</p>

<p>Potential Mentors: Sravanthi Sinha, Martina Kutmon</p>

<p>Contact: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 20: Improved Layering/Interaction in Cytoscape 3 Networks</h2>

<p>This project involves working on the visualization core in Cytoscape 3 (ding) to significantly improve the user experience. Currently, Cytoscape supports a single canvas to visualize the network. When moving or changing graphic objects (nodes and edges), the entire visible portion of the canvas needs to be updated. It has been pointed out by visualization experts that this problem can be easily solved by adding an additional “selection” canvas. Motion of nodes and edges could be performed in this canvas rather than the main canvas, significantly improving the interactivity on dense networks. The addition of a “selection” canvas also enables a number of visualization techniques that are too computationally expensive in the current Cytoscape model.</p>

<br />
<p>Language and Skills: Java</p>

<p>Potential Mentors: Scooter Morris</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 21: Compound Graph Visualization in Cytoscape 3</h2>

<p>Cytoscape 3 has introduce two concepts into the core, groups and annotations, that could be combined to provide a compound graph visualization. Currently groups can be either in a “collapsed” or “expanded” state. For a compound graph, a new visualization state needs to be introduced that allows the compound node (group) to be visualized as a shape (concave or convex hull) around it’s children. Moving that shape would move all of it’s children, etc.</p>

<br />
<p>Language and Skills: Java</p>

<p>Potential Mentors: Scooter Morris</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 22: Visualization of Time-series Cluster Data</h2>

<p>Currently, the clusterMaker plugin uses standard heat maps to visualize attribute cluster data (based on Java TreeView). Increasingly, time series data is becoming important for clustering applications and it would be useful to expand the provided visualization options to explicitly support time series data. For these purposes, we’re thinking about time-series that have multiple conditions in each series, which means we need to move beyond the standard two-dimensional heat map. This project can explore utilizing animation, small multiples, or other standard visualization techniques (or your own novel techniques) to provide new visualization opportunities to users. Ideally, this would be integrated with network-based animation tools such as DynNetwork or CyAnimator.</p>

<br />
<p>Language and Skills: Java</p>

<p>Potential Mentors: Scooter Morris</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 23: Develop a Converter between Visual and Textual Representations of Biological Pathways</h2>

<p><b>Goal</b>: Pathway Commons [1] seeks to aggregate all pathway data available in literature, public databases, and 
curation/crowdsourced efforts. Our goal is to accelerate this process by leveraging pathway diagram editors that support a 
widely used visual representation for biological pathways.</p>
<p><b>Problem</b>: Pathway Commons data is represented in the Biological Pathway Exchange (BioPAX) community standard format [2], 
but the BioPAX format lacks a simple graphical editor. On the other hand, the System Biology Graphical Notation (SBGN) is a 
visual representation for pathways that is supported in about 20 tools. We'd like you to build a converter between the BioPAX 
and SBGN formats.</p>
<p>A similar project has been developed with the editor CellDesigner (that shares a format similar to SBGN) making us believe 
that this should be a reasonably straight forward project; many of the mapping issues have already been addressed. The project 
will make use of two Java libraries Paxtools ([3], for BioPAX) and libSBGN ([4], for SBGN).</p

<p>[1]<a href="http://www.pathwaycommons.org/">Pathway Commons</a></li></p>
<p>[2]<a href="http://www.ncbi.nlm.nih.gov/pubmed/20829833">BioPAX</a></p>
<p>[3]<a href="http://www.biopax.org/paxtools.php">Paxtools</a></p>
<p>[4]<a href="http://www.sbgn.org/LibSBGN">LibSBGN</a></p>

<p><b>Language and Skills</b>: Java and XML</p>
<p><b>Potential Mentors</b>: Augustin Luna</p>
<p><b>Contact</b>: <a href="mailto:lunaa@cbio.mskcc.org" >Augustin Luna</a></p>

<br />

<h2 dir="ltr" >Idea 24: Touch-based Exploration of Large Data Collections and Networks</h2>

<p>The goal of this project is to develop a general, open source framework for performing iterative, faceted search on large data collections and networks specifically for touch-based input devices. The project will leverage ideas from existing desktop applications (see<a href="http://www.youtube.com/watch?v=BZuFUZpEZ-A" > Microsoft Pivot</a>) but will include experimentation and innovation in interaction techniques in order to create a more engaging user experience. Ultimately the framework will be applied in one of potentially many biological applications, including MedSavant, which is described below.</p>

<p>MedSavant is a high-performance search engine for genomic datasets, which is primarily aimed to help researchers identify genetic conditions that cause disease. The desktop client allows users to</p>

<p>incrementally construct faceted search queries and visualize their results virtually in realtime. Similar workflows enabled by the desktop client will be made possible by integration with the proposed</p>

<p>framework. For more information, visit <a href="http://www.medsavant.com" >www.medsavant.com</a>.</p>

<br />
<p>Language and Skills: Android or iOS development; AJAX, JSON, Java, and Javascript; UNIX</p>

<p>Potential Mentors: Marc Fiume, Marta Girdea</p>

<p>Contact: <a href="mailto:mfiume@cs.toronto.edu" >mfiume@cs.toronto.edu</a></p>

<br />
<h2 dir="ltr" >Idea 25: Improve the Text-Mining Functionality of Factoid</h2>

<p><a href="http://pathwaycommons.github.io/factoid/" >Factoid</a> aims to be a tool that helps authors annotate biological interactions that occur in the text of academic publications in a machine-readable manner. Currently, Factoid uses basic entity-recognition to identify genes, proteins, biological processes to identify interactions between these entities. The goal of this project would be to integrate more sophisticated algorithms into the extraction of interactions from text content. A starting point for some of these would be the algorithms outlined by the EBI website WhatIzIt.</p>
<br />
<p>Languages and Skills: Experience connecting web services, Javascript</p>

<p>Potential Mentors: Augustin Luna (MSKCC), Max Franz (U of Toronto) </p>

<p>Contact: <a href="mailto:lunaa@cbio.mskcc.org" >Augustin Luna</a>, <a href="mailto:maxkfranz@gmail.com" >Max Franz</a></p>

<br />
<h2 dir="ltr">Idea 26: Optimize Semi-supervised Learning Algorithm for Genomic Datasets and Inject Prior Knowledge from 
Biological Pathways</h2>
<p><b>Background</b>: A major problem in bioinformatics is that the number of features collected by modern high-throughput 
technologies severely overwhelms the number of samples for which such features are available. In a typical dataset, it is not 
uncommon to have tens of thousands of molecular measurements, but only a few tens or hundreds of samples. This severe 
undersampling creates a significant challenge for researchers who want to mine such datasets for biological or clinical 
insight. For example, when looking to predict sensitivity of cancer cells to various compounds, it is not uncommon to come up 
with a collection of molecular features that are highly predictive on a given dataset, only to have the accuracy fail to 
generalize to new, previously unseen samples.</p>
<p>A natural way to approach the issue is to bring in additional samples to the dataset, e.g., other cancer cells from the same 
tissue in the compound sensitivity example above. However, the phenotypic information or "labels" for such samples may not be 
readily available. Semi-supervised learning is a branch of machine learning that deals with constructing predictive models 
using both labeled and unlabeled data. It is particularly well-suited to biological applications, because while there is 
typically only a small handful of labeled samples, the unlabeled data is aplenty. Furthermore, we can expect some shared 
underlying biology between labeled and unlabeled data, due the fact that molecular measurements capture some underlying 
activity of various signaling and metabolic pathways in a cell. Semi-supervised learning can tap into the vast amount of 
unlabeled data to uncover such pathways and then utilize the labeled samples to correlate these pathways to phenotypic 
outcomes.</p>
<p><b>Goal</b>: To implement a specific semi-supervised learning algorithm and to work with the mentor on modifying this 
algorithm to tailor to biological data. The algorithm was originally proposed by Ando & Zhang (2005) and operates by 
constructing multiple predictive tasks from the unlabeled data and learning across all those tasks simultaneously. The 
intuition behind the algorithm is that, if a molecular feature X is highly correlated with the phenotype F, then learning to 
predict X is just as valuable as learning to predict F directly. The advantage of predicting X is the fact that it's observed 
across all data (both labeled and unlabeled), while the measurements of F are, by definition, available for the labeled data 
only. Continuing with the example above, if a compound targets the epidermal growth factor receptor (EGFR) in a cell, then 
predicting whether EGFR is highly expressed is a task that should be highly related to directly predicting the sensitivity of 
the cell to that compound.</p>
<p>To simultaneously learn across multiple tasks, Ando & Zhang's algorithm makes an assumption that the tasks share some 
underlying lower-dimensional representation. This provides a perfect place to inject prior biological knowledge about 
feature-feature relationships. Specifically, a sub-goal of the project is to modify the lower-dimensional constraints in the 
algorithm to align with known information about biological pathways, as taken from web resources, such as WikiPathways, 
BIOCARTA, PathwayCommons, etc.</p>
<p>The basic implementation of the algorithm is already in place. It remains to optimize the implementation for handling 
full-scale genomic datasets (~20k features and thousands of samples) by moving parts of the code from R to C / C++, and to 
implement the pathway-based component discussed above.</p>
<p>Language and Skills: R, C / C++, Linear Algebra</p>
<p>Potential Mentors: Artem Sokolov (UCSC)</p>
<p>Contact: <a href="mailto:sokolov@soe.ucsc.edu">Artem Sokolov</a></p>
<p>References:<br />
Ando, R. K., & Zhang, T. (2005). A framework for learning predictive structures from multiple tasks and unlabeled data. The 
Journal of Machine Learning Research, 6, 1817-1853.</p>

<br />
<h2 dir="ltr">Idea 27: Developing Web Interface For The Online Resource for Interface- Based Interaction Networks (TOR-IBIN) 
Using Cytoscape.js</h2>
<p><b>Background</b>: The protein-protein interaction (PPI) is a physical contact between two proteins, or more, resulted from 
biochemical reaction or electrostatic forces. The interaction usually takes place through an Interface, which is part of the 
protein (e.g. protein domain) with special molecular structure that enables and mediates the interaction between proteins. 
Interface-based PPI predictions help in understanding the mechanisms of interactions between proteins as well as helps in 
deciphering the functional roles of the proteins.</p>
<p><b>Focus</b>: TOR-IBIN is an online resource for gathering and storing the published interface-based (IB) protein-protein 
interaction networks (IN) predicted using any PPI prediction method. TOR-IBIN aims to make these networks available in a 
central repository with unified format and comprehensive search functions that maximize the use of these data. TOR-IBIN 
web-based interface aims to implement the browse and search functions and well as network visualization using Cytoscape.JS and 
network download in several formats including JSON and tab-delimited formats. TOR-IBIN is a relational database implemented in 
MySQL.
- The web interface and functionality will be similar to the web interface of the cancer variants database (Can-VD) 
canvd.baderlab.org.
</p>
<p><b>Goals</b>:
<ol>
<li>Develop a web-based interface for interface-based PPI data retrieval with a comprehensive search functions and data 
download options.</li>
<li>Develop a visualization and annotation module using Cytoscape.js to visualize the retrieved PPI data as a network with 
annotations that highlights the features of the predicted PPI such as the prediction confidence and validation...etc.</li>
</ol></p>

<p><b>Inputs</b>: The student will be provided with the database design and sample data as well as an introduction to the 
topic and the different possible visualizations and annotations needed. The student will have the freedom to choose among 
them or to propose other options.</p>

<p><b>Language and Skills</b>: Javascript, JSON, PHP, JQuery</p>
<p><b>Potential Mentors</b>: Mohamed Helmy, Gary Bader</p>
<p><b>Contact</b>: mohamed.attiashahata@utoronto.ca, gary.bader@utoronto.ca</p>

<br />
<h2 dir="ltr">Idea 28: Full Compound Support in Cytoscape 3</h2>
<p><b>Background</b>: The notion of compound graphs is used to represent more complex types of relationships or varying levels 
of abstractions in data visualization. Networks from almost all domains need such abstractions and nesting, including 
biological networks. The recently developed standard notation of SBGN [1] makes heavy use of such structures for molecular 
complexes and cellular locations.</p>
<p><b>Goal</b>: We would like to first implement full compound support in Cytoscape 3 [2] core. This will not only allow users 
to nest nodes of a network but also automatically adjust size of a compound node as its child nodes get  moved, resized or  
deleted. Then, we'd like to implement a native spring embedder based layout algorithm named CoSE [3] that takes such structures 
into account during layout. A Java implementation of CoSE is already publicly available [4] and needs to be integrated into 
Cytoscape 3.</p>
<p>[1]<a href="http://www.sbgn.org/">http://www.sbgn.org/</a></p>
<p>[2]<a href="http://www.cytoscape.org/">http://www.cytoscape.org/</a></p>
<p>[3] U. Dogrusoz, E. Giral, A. Cetintas, A. Civril, and E. Demir, "A Layout Algorithm For Undirected Compound Graphs", 
Information Sciences, 179, pp. 980-994, 2009. <a href="http://www.cs.bilkent.edu.tr/~ivis/layout/cose-animated-demo/cose.html">
     http://www.cs.bilkent.edu.tr/~ivis/layout/cose-animated-demo/cose.html</a></p>
<p>[4]<a href="http://www.cs.bilkent.edu.tr/~ivis/chilay.html">http://www.cs.bilkent.edu.tr/~ivis/chilay.html</a></p>

<p><b>Language and Skills</b>: Java, Data Structures and Algorithms</p>
<p><b>Potential Mentors</b>: Ugur Dogrusoz (Bilkent University & MSKCC) and Gerardo Huck (U of Toronto)</p>
<p><b>Contact</b>: ugur@cs.bilkent.edu.tr and onur@cbio.mskcc.org </p>
<br />

<!-- END PASTE -->

    </div> <!-- .span9 !-->
</div> <!-- .row !-->

<script src="../assets/js/jquery-1.11.0.min.js"></script>
<script src="../assets/js/jquery-ui.min.js"></script>
<script src="../assets/js/jquery.tocify.min.js"></script>
<script>
$("#nav").tocify({
    context: "#ideas",
    showAndHide: false,
    selectors: "h2",
});
</script>
</body>
</html>
