<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>NRNB: GSoc Ideas Page</title>

<!-- Le styles -->
<link href="../assets/css/bootstrap.css" rel="stylesheet">
<link href="../assets/css/bootstrap-responsive.css" rel="stylesheet">
<link type="text/css" rel="stylesheet" href="../assets/css/jquery.tocify.css" />
<link type="text/css" rel="stylesheet" href="../assets/css/jquery-ui.min.css" />
<style type="text/css">
#nav {
     position: fixed;
}
.tocify ul, .tocify li {
        line-height: 15px;
}
</style>
</head>
<body>
<div class="row" height="100%">
    <div class="span4"  height="100%">
         <!-- Our table of contents will be here !-->
         <div id="nav" style="font-size: 12px;" height="100%"></div>
    </div>
    <div id="ideas" class="span8">

<!-- PASTE HERE -->

<p><br />
</p>

<h2 dir="ltr" >Idea 1: Submit your own original proposal</h2>

<p>Feel free to propose your own idea. As long as it relates to one of our projects, we will give it serious consideration. 
Creativity and self-motivation are great traits for open source programmers, but make sure your proposal is also relevant.</p>
<br />

<h2 dir="ltr" >Idea 2: Co-authorship Social network App</h2>
<p>Description: Biological networks can be visualized and analyzed using Cytoscape. Often researchers want to go beyond the 
network of proteins or genes and also look at the inter-connectedness between colleagues and institutions. Who tends to 
publish together? What institutions are most collaborative? Are there inter-disciplinary connections in my institution?</p>

<p>Goal: Create a visual summary of how individuals are connected. An existing App currently queries PubMed directly or take
in a file with a set of publications (from scopus or incites) and builds a co-publication network where nodes are authors and 
edges the number of publications two authors share. Different sources have different depths of data, for instance PubMed, the 
most widely accessible of the three main sources, has citation counts but they are limited to only those articles cited in 
pubmed central and its institution data is limited to the first author. Given that PubMed is the most accessible source we 
want to expand its functionality to include computed institution data as well as better citation count approximations, and 
H-index calculation for individuals or subsets of papers. </p>

<p>This functionality can be applied to several data types outside academia, whenever relationships between people is of 
interest. An example is professional social networks (e.g. linked-in), where individuals are connected by who they have worked 
with in the past or the twitter-verse where individuals are connected when they each reference the same (#) hash tag. This 
project will start with support for co-publication networks and extend to other networks like twitter and linked-in as time 
permits. Additionally, we want to expand our co-authorship cytoscape App to include generic file formats to easily create the 
above networks.</p>

<p>Links:</p>

<p>Existing code - <a href="https://github.com/BaderLab/SocialNetworkApp" >https://github.com/BaderLab/SocialNetworkApp</a></p>

<p>Existing userguide - <a href="http://baderlab.org/UserguideSocialNetworkApp" >http://baderlab.org/UserguideSocialNetworkApp</a></p>

<p>Language and Skills: Java, XML, PubMed e-utils</p>

<p>Potential Mentors: Sergio Baranzini, Gary Bader, Ruth Isserlin</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 3: User Interaction on Biomedical Linked Data</h2>

<p>During the GSoC 2012, we developed<a href="http://vsdlc3.blogspot.com" > SemScape</a>, a plugin that allows the use of 
Cytoscape as a browser for Semantic Web (Linked Data) knowledge bases. We are currently in the process of updating libs and 
packaging the plugin for the Cytoscape App Store.</p>

<p>The idea for this year’s project is to go one step further and to allow the use of Cytoscape to visually define queries in 
SPARQL. The basic idea is that from a graph, a sub-graph can be defined where a few nodes or edges can be marked as variables 
(and eventually filters can be associated to them): this is equivalent to a (simple) SPARQL query. The user can then define 
queries in a simple and interactive way, without knowing the underlying syntax, that can be stored and applied on different
data sources.</p>

<p>The project can then be extended in three directions, depending on the interest of the student:</p>

<ul>
<li><p>allowing users to push such queries to a repositories and hence share them (the current plugin 
support updates of queries from remote repositories)</p></li>
<li><p>Implementing more complex queries (unions, negation, …)</p></li>
<li><p>Implementing heuristics for query-selection: in the current plugin, queries can be performed on demand, but they are 
also associated to specific actions. For instance, right-selecting a node of type protein can present as an option to expend 
the graph with the results of a query which is can include the specific protein as a value. This is possible through a simple 
markup through which queries are annotated by the user. Such markup could be included in the visual query/construction process
and/or extracted from simple query patterns.</p></li>
</ul>

<p>Some of the ideas in this project<a href="http://www.biomedcentral.com/1471-2105/9/S4/S6" > have been already prototyped</a>, so we have a good understanding of the difficulty and timing required. </p>

<p>This project has the potential to make Cytoscape an ideal tool to navigate and query Linked Data resources (both in biomedical and, more generally, in Open Data domains). It also opens interesting possibilities in analyzing graphs from graph knowledge bases that can be represented as RDF (e.g.: dbpedia, freebase,...).</p>

<p>Language and Skills: Java/Maven. Knowledge of OSGi would be helpful. Knowledge of Semantic Web technologies would be a plus.</p>

<p>Potential Mentors: Andrea Splendiani, Andra Waagmeester</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 4: CentiScaPe for multiple networks analysis and graphical output enhancement</h2>

<p>Background: CentiScaPe is a Cytoscape plugin that allows to compute network centrality parameters. It is the most complete 
Cytoscape plugin for network centralities calculating Degree, Average Shortest Path, Eccentricity, Closeness, Betweenness, 
Centroid, Stress, Radiality, Eigenvector, Edge Betweenness, Bridging Centrality, for directed and undirected networks. 
CentiScaPe allows identifying network nodes that are relevant from both experimental and topological viewpoints. CentiScaPe 
also provides a Boolean logic- based tool that allows easy characterization of nodes whose topological relevance depends on 
more than one centrality. Finally, different graphic outputs and the included description of biological significance for each 
computed centrality facilitate the analysis by the end users not expert in graph theory, thus allowing easy node 
categorization and experimental prioritization.</p>
<p>Goal: The new beta version of CentiScaPe provides also the possibilities to analyze multiple networks and to integrate the 
analysis with experimental data set. This implies that a user can automatically start a computation of several networks and 
waits for the results. Anyway, the output of the analysis of a lot of networks (say 100) is useless if not supported by a 
smart and user friendly system to analyze the results. The Idea is to implements some techniques used in software analysis, 
some graphical outputs and some statistical parameters to extract information from the results of multiple computation in 
order to compare different networks and to extract topological based biomarkers:</p>
<ul>
<li>Some statistical parameters, automatic analysis of the output, and new kinds of graphical outputs that can help to extract 
information from a multiple networks centrality analysis should be provided. The tools should for example be able to answer to
questions as “How does the centroid value of the protein A chenge in the submitted 100 different networks?” or “How does the 
betweenness of protein B change if we consider 30 different phosporilation values for each proteins in the network N (this 
means we have 30 different experimental data set for the same network that must be included in the computation)?”.</li>
<li>The tool should also provide the implementation of the integration of experimental data-sets in order to include them in 
the topological analysis (more info will be provided to the interested students)</li>
</ul>
<p><b>Technology and Skills</b>: Java, software analysis (possibly abstract interpretation theory)</p>
<p><b>Potential Mentors</b>: Giovanni Scardoni (CBMC, University of Verona), Fausto Spoto (University of Verona)</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 5: Improving Cytoscape.js Based Viewer for SBGN Process Description Diagrams with Better Layout and 
Advanced Complexity Management Operations</h2>

<p><b>Background</b>:
<a href="http://cytoscape.github.io/cytoscape.js/">Cytoscape.js</a> is a JavaScript graph library for analysis and 
visualization of relational information. The <a href="http://www.sbgn.org/">Systems Biology 
Graphical Notation</a> (SBGN) is a visual language developed by the community with the intention of enabling scientists to 
represent networks, including models of cellular processes, in a standard, unambiguous way. SBGN comprises three 
complementary languages suited to represent different types of pathways. SBGN Process Descriptions is a language that permits 
the description of the processes (e.g. biochemical reactions) taking place in a biological system and makes heavy use of 
compound structures (e.g. compartments and molecular complexes). 
Often times, SBGN PD networks are too large to visualize and details become obscured. Strategies to conceal the "unwanted" 
details of a network, allowing one to temporarily remove certain nodes and edges and subsequently reintroduce the removed 
details. Complexity management techniques in graph visualization such as filtering, hiding, folding, and nesting play an 
important role in effective analysis of biological networks as much as any other domain.
Last year, support for compound structures has been added to Cytoscape.js core and an SBGN viewer named 
<a href="https://code.google.com/p/sbgnviz-js/">SBGNViz</a> has been 
built [3]. Some complexity management operations have also been designed and implemented within this tool to hide unwanted 
details on demand without violating SBGN PD network invariants (e.g. all substrates and products of a process must be shown 
at any point).</p>

<p><b>Goal</b>:
This year, we'd like to improve this tool in certain ways. First, it'd be great to improve the layout implementation that 
supports compound graphs. Secondly, existing complexity management operations could be extended to include folding/unfolding. 
Notice however that this is not a straightforward topology change. For instance collapsing a compound node into a single "meta" 
node requires calculation of any inter-graph edges going into the compound node and representing them with "meta" edges in the 
collapsed network. Similarly, expanding a collapsed node back requires undoing it. The situation becomes cumbersome since 
arbitrary level of nesting is allowed and folding can take place at an arbitrary level.</p>

<p><b>Technology and Skills</b>: JavaScript, Java, Data Structures and Algorithms</p>
<p><b>Potential Mentors</b>: Ugur Dogrusoz (Bilkent University & MSKCC) and S. Onur Sumer (MSKCC)</p>
<p><b>Contact</b>: <a href="mailto:ugur@cs.bilkent.edu.tr" >ugur@cs.bilkent.edu.tr</a> and <a href="mailto:onur@cbio.mskcc.org" >onur@cbio.mskcc.org</a> </p>
<br />

<h2 dir="ltr" >Idea 6: Computing patient similarity networks for 100K samples</h2>
<p><b>Background</b>: A major goal of modern biomedical research is to tailor patient treatment based on individual genomic 
profiles and/or other patient information. Network-guided approaches can integrate several types of patient data  - e.g. 
genetic, RNA, clinical, brain imaging – to identify networks of patient similarities (PSN). These networks could identify 
biological pathways reflecting different disease subcategories or explain why responses to the same medication vary among 
people. Software for computing PSN exists, but currently requires the user to use a patchwork of different tools. Moreover, 
existing machine learning operations don’t easily scale to the size of contemporary datasets (~100K patients).</p>
<p><b>Goal</b>: To create fast, streamlined software to compute and validate PSN.  The software will compute PSN based on 
different types of data (e.g. molecular profiles, pathway information, clinical data), by using a similarity measure, such as 
Pearson correlation. It will perform K-fold cross validation for training. Finally it will generate ROC curves to measure the 
sensitivity-specificity of the prediction system.</p>
<p><b>Existing Software</b>: A software pipeline for this workflow already exists, therefore the algorithm itself will not 
need to be developed. Development will focus on optimizing the workflow, and creating a separate software package with 
computing and visualization functionality. Rcpp (R interfacing with Cpp) will be used to program the software. C++ matrix 
libraries (e.g. Armadillo) will be used to reduce memory footprint and speed computations.</p>
<p><b>Publications</b>: 
     <ul>
          <li>Wang B. et al. (2014). Similarity network fusion for aggregating data types on a genomic scale. Nature 
          Methods 11(3): 333-7.</li> 
          <li>Mostafavi S. et al. (2008). GeneMANIA: a real-time multiple association network integration algorithm for 
          predicting gene function. Genome Biol. 9 Suppl 1:S4.</li> 

     </ul>
</p>
<p><b>Technology and Skills</b>: R statistical programming language, C++, Java</p>
<p><b>Potential Mentors</b>: Gary Bader, Shraddha Pai</p>
<p><b>Contact</b>: <a href="mailto:shraddhapai.neuro@gmail.com">Shraddha Pai</a></p>
<br />

<h2 dir="ltr" >Idea 7: Develop a Method for Aligning Pathways and Visualizing these Alignments</h2>
<p>There is continued growth in the groups generating machine-readable representations of biological pathways 
(such as <a href="http://www.reactome.org/" >Reactome</a> or <a href="http://wikipathways.org/" >WikiPathways</a>) and 
subsequent efforts to aggregate these pathways into unified repositories, such as Pathway Commons. Aggregation of these 
pathways suffers from multiple pathways being aggregated that differ due to mismatches in author representations of these 
pathways. The goal of this project is to develop a method for aligning pathways and then visualizing the results of these 
alignment. Starting points could be to find interactions that use the same substrates and produce the same products. </p>
<p>There are many uses of such a tool including helping to validate pathways as they are added to repositories, merging 
pathways across different resources, and the comparative analyses of pathways across organisms. One way to think about this 
project, is if you're given multiple representations of the same pathway 
(<a href="http://wikipathways.org//index.php?query=wnt&amp;species=Homo+sapiens&amp;title=Special%3ASearchPathways&amp;doSearch=1&amp;ids=&amp;codes=&amp;type=query" >examples of several human Wnt pathways</a>) can you develop a method to show a unified pathway and how the original representations map to the unified pathway. </p>
<p><b>Technology and Skills</b>: Java, XML</p>
<p><b>Potential Mentors</b>: Augustin Luna (MSKCC)</p>
<p><b>Contact</b>: <a href="mailto:lunaa@cbio.mskcc.org" >lunaa@cbio.mskcc.org</a></p>
<br />

<h2 dir="ltr" >Idea 8: Drag and Drop or Palette Editor for SBGN and other graphical notations in Cytoscape</h2>
<p>The goal is to design and implement a drag-and-drop or palette-based editor for Cytoscape that would include nodes, edges, 
and graphical annotations. In the ideal case, this would be extensible so that apps add their own nodes or edges with specific 
graphical attributes. The Systems Biology Graphical Notation (SBGN) is a standard graphical notation for presenting pathways 
and protein interactions. Currently, Cytoscape lacks the node symbols and edge types required for SBGN. In this project, those
notations will be added to the Cytoscape core, and a simple palette editor developed for user interaction.</p>
<p>Language and Skills: Java</p>
<p>Potential Mentors: Scooter Morris, Alex Pico</p>
<p>Contact: <a href="mailto:scooter@cgl.ucsf.edu" >Scooter Morris</a> </p>
<br />

<h2 dir="ltr" >Idea 9: Improve and expand Cytoscape’s color mapping options</h2>
<p>Currently, the color options for Cytoscape’s discrete color mappings are very limited. Brewer palettes provide a 
luminosity-neutral set of colors that would be extremely useful to add to Cytoscape, and there are several other approaches 
to choosing sets of colors that contrast nicely. Creating these color palettes and providing them for use within Cytoscape’s 
visual style interface would be extremely useful to Cytoscape users.</p>
<p><b>Language and Skills</b>: Java</p>
<p><b>Potential Mentors</b>: Scooter Morris, Michael Heuer, Alex Pico</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu" >Scooter Morris</a> </p>
<br />

<h2 dir="ltr" >Idea 10: Constrained layout algorithms in Cytoscape</h2>

<p>Most layout algorithms in Cytoscape rely on inherent network topology and only a crude set of spatial constraints (e.g., grid, circle). There are many cases when working with networks in Cytoscape where you want to be able to constrain layouts to user- or attribute-defined spaces. This idea involves implementing layouts that are constrained by:</p>

<ul>
<li><p>specific shapes </p>

</li>
<ul>
<li><p>basic shapes </p>

</li>
<li><p>cellular templates (e.g., with subcellular organelles)</p>

</li>
<li><p>attribute-driven Venn diagrams (e.g., to visualize attribute groupings and overlaps)</p>

</li>
</ul>

<li><p>specific “anchor” nodes</p>

</li>
<ul>
<li><p>user-defined (e.g., manually selected)</p>

</li>
<li><p>attribute-defined (e.g., hubs or column values)</p>

</li>
</ul>

<li><p>etc...</p>

</li>
</ul>

<br />
<p>Language and Skills: Java, and a strong math background</p>

<p>Potential Mentors: Scooter Morris, Alex Pico, Bruce Aronow</p>

<p>Contact: <a href="mailto:scooter@cgl.ucsf.edu" >Scooter Morris</a> </p>

<br />
<h2 dir="ltr" >Idea 11: A SPARQL query plugin for Pathvisio</h2>

<p><b>Background</b>: The semantic web is becoming a valuable resource for Pathway knowledge. The challenge still remain to 
smoothly integrate this knowledge into biological frameworks. With their SPARQL endpoints Uniprot 
(<a href="http://beta.sparql.uniprot.org/" >http://beta.sparql.uniprot.org/</a>) or the EBI rdf Platform 
(<a href="https://www.ebi.ac.uk/rdf/" >https://www.ebi.ac.uk/rdf/</a>) , biological data is becoming available as semantic 
web data. To use this knowledge one need not only to learn SPARQL queries, but also be prepared to convert the semantic data 
in different formats that comes with the various biological platforms. Using semantic web data in Pathvisio, we currently 
need to compose the appropriate SPARQL queries, get the data and write various parser to change the data in to the different 
data models. </p>

<p><b></b>: To make a Pathvisio plugin that allows a pathway curator to write SPARQL queries against a variety of semantic 
web based pathways resources and include the obtained triples into the Pathway under scrutiny. The plugin should also provide 
an interface to SPARQL results for other Pathvisio plugins. Having such SPARQL interface still requires a pathway editor to 
become knowledgeable in writing SPARQL queries. The burden of having to write various parsers for semantic web data however, 
is taken away. The plugin should:</p>

<ul>
<li>Provide a SPARQL query editor, from where the user could compose queries, which are then subsequently launched to the 
respective SPARQL endpoints.</li>
<li>Provide an interface that wrap specific SPARQL queries for other Pathvisio plugins.</li>
<li>Link to a centralized repository (e.g. Github) where SPARQL queries can be stored for future reference and others to use.</li>
<li>Link to a centralized index of relevant SPARQL endpoints. This index might again be stored on platforms like Github.</li>
<li>Import the resulting triples, which can then be linked to recognizable pathway components, or as data elements to use in 
Pathway analysis.</li>
<li>Make a RDF representation of a Pathway opened in Pathvisio. These triples are to be used as template for SPARQL queries. 
The user should be able to click on a triple which is then added to the WHERE block in the SPARQL query.</li>
</ul>

<p><b>Language and Skills</b>: SPARQL, RDF, and Java. This idea is for a student who has solid background in SPARQL queries, or the Semantic web in 
general, and who is looking into opening up the Semantic Web to specific knowledge domains (I.e. Biology and/or Medicine).</p>

<p><b>Potential Mentors</b>: Andra Waagmeester, Anders Riutta, Martina Kutmon </p>

<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 12: Web publishing of models from Virtual Cell database</h2>

<p>The<a href="http://vcell.org" > Virtual Cell</a> modeling and simulation framework is an open source software for modeling and simulations of biological systems using Java RMI, providing both client- and server- based modeling and simulation capabilities. The biologically oriented user interface allows experimentalists to create models, define cellular geometry, specify simulations and analyze the simulation results. The models and results are stored on a remote server in ORACLE database and presently can be reviewed only in the VCell software. You need to implement a content management system that will search and retrieve model information from ORACLE or MySQL database and publish it online. The model structure stored in a relationship database is available at<a href="http://vcell.org/technology/VcmlSpec.html" > http://vcell.org/technology/VcmlSpec.html</a></p>

<br />
<p>Language and Skills: Java</p>

<p>Possible Mentors: Michael Blinov, Jim Schaff</p>

<p>Contact: <a href="mailto:vcell-discuss@googlegroups.com" >vcell-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 13: New Features for Online Tutorial System</h2>

<p><a href="http://opentutorials.cgl.ucsf.edu" >OpenTutorials</a> is a free, online tutorial resource for open source and NRNB-supported tools. It is built on top of MediaWiki and a handful of extensions. It’s basic function is to provide a wiki-based collection of tutorials that can be viewed as a web page, slide show or printed handout. The wiki approach makes it easy to build and maintain content. The slide show and handout features make it useful for presenters.</p>

<p>There are a number of features that need to be implemented to improve the site and really make it effective to a larger community. What we’re looking for here with this project is an application that is very clear about the use cases of tutorials:</p>

<ul>
<li><p>for developers wanting to create and maintain tutorial materials</p>

</li>
<li><p>for presenters of tutorials</p>

</li>
<li><p>and for users learning from tutorials.</p>

</li>
</ul>

<br />
<p>The application should include a number of features that enhance these use cases. When possible, the features should point to other examples were similar approaches have proven successful (e.g., in other popular software). After we discuss the features, your application should then include implementation details and a timeline. These would be implemented as one or more PHP extensions, for example. Experience or some demonstration of familiarity with MediaWiki or PHP would be a plus. Here are some feature ideas to get you started:</p>

<ul>
<li><p>Mix-and-match: We need a mechanism to allow presenters to mix and match tutorials (or sections of tutorials) into sets that are tailored for their class or workshop audience.</p>

</li>
<li><p>Transient locks: We need a mechanism for temporarily locking content while it is being presented (so that someone isn’t editing a wiki page while you are presenting it to a class of 100 students!). This can be done either by supporting temporary presenter “copies” that are not publicly editable, or simply locks that presenters can set for specific dates and times.</p>

</li>
<li><p>Enhanced image replacement tools to make it easier to update screenshots.</p>

</li>
<li><p>Tutorial builder tools using templates and forms to get a new tutorial started, or an integrated html-to-wiki converter like<a href="http://labs.seapine.com/htmltowiki.cgi" > this</a>.</p>

</li>
<li><p>Tweaks to the wiki page display of header/meta data and of slide breaks (e.g., a clear, yet unobtrusive icon).</p>

</li>
</ul>

<br />
<p>Language and Skills: PHP, MediaWiki</p>

<p>Potential Mentors: Alex Pico, Kristina Hanspers</p>

<p>Contact: <a href="mailto:apico@gladstone.ucsf.edu" >apico@gladstone.ucsf.edu</a></p>

<br />
<h2 dir="ltr" >Idea 14: Custom Export Editor for WikiPathways Using SVG</h2>

<p>Background:<a href="http://wikipathways.org" > WikiPathways</a> is a collaborative pathway editing and sharing platform. There are many ways to use the pathway content including data visualization and analysis. Another use is as illustrations in presentations and publications. We currently provide SVG export for this purpose. However, the images usually require quite a bit of “touch up” in secondary tools like Illustrator, which inhibits this use case.</p>

<p>Goal: Customize a basic SVG Editor to insert into the export process to allow basic visual style updates and custom annotations on top of the pathway image. Pre-configured options would include:</p>

<ul>
<li><p>thicker lines and bold font (for better display in powerpoint)</p>

</li>
<li><p>swap basic cellular component shapes with nice graphics (provided from an existing repository)</p>

</li>
<li><p>color schemes</p>

</li>
</ul>

<p>This simple “touch up” editor could dramatically increase the audience for WikiPathways to include anyone looking for an easy tool to generate nice looking pathway images for presentations. Ideally, this editor would work on tablets as well as across major browsers.</p>

<br />
<p>Language and Skills: SVG,<a href="http://code.google.com/p/svg-edit/" > SVG-edit</a> (or alternative)</p>

<p>Potential Mentors: Alexander Pico, Anders Riutta</p>

<p>Contact: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 15: Integration of WikiPathways Visualization with PathVisio into Bioclipse</h2>

<p><b>Goal</b>: To integrate PathVisio visualization widget into Bioclipse for visualization of pathway information from 
WikiPathways. Selecting protein and metabolites in the pathway visualization, will shows of that chemical entity in other 
Bioclipse components, using the Jmol and JChemPaint plugins for Bioclipse. Preferably, interaction can be scripted, allowing 
integration with other metabolomics features in Bioclipse.</p>

<p><b>Motivation</b>: Bioclipse is a powerful platform combining a powerful scripting language to write data analysis and 
visualization workflows, which can easily be shared on, for example, MyExperiment.org [1]. While it has various components 
to support metabolomics in general, it lacks support for visualization of metabolic pathways.</p>

<p><b>Description</b>: You will use Java to write a plugin for Bioclipse [2] that uses PathVisio [3] as a library to visualize 
metabolic pathways from WikiPathways inside Bioclipse. You will use the Eclipse selection event propagation mechanism [4] so 
that when users select protein and metabolites, Bioclipse will automatically visualize 2D or 3D structural information about 
those molecules (only the event creation has to be implemented). Preferably, a scripting language extension is written [5], 
to allow extracting information from the pathway and highlighting of parts of the pathway. This involves extending an existing 
interface with methods that can be run from the scripting language.</p>

<ol><li><a href="http://myexperiment.org/" >myexperiment.org</a></li>
<li><a href="http://www.bioclipse.net/" >www.bioclipse.net</a></li>
<li><a href="http://www.pathvisio.org/" >www.pathvisio.org</a></li>
<li><a href="http://www.eclipse.org/articles/Article-WorkbenchSelections/article.html" >www.eclipse.org/articles/Article-WorkbenchSelections/article.html</a></li>
<li><a href="http://www.biomedcentral.com/1471-2105/10/397" >www.biomedcentral.com/1471-2105/10/397</a></li>
</ol>
<p><b>Language and Skills</b>: Java</p>
<p><b>Potential Mentors</b>: Egon Willighagen, Jonathan Alvarsson</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>

<br />
<h2 dir="ltr" >Idea 16: Integration of Cytoscape into Bioclipse</h2>

<p>Goal: To integrate the new Cytoscape OSGi visualization widgets into Bioclipse for visualization of large interaction networks. Selecting protein and metabolites in the pathway visualization, will shows of that chemical entity in other Bioclipse components, using the Jmol and JChemPaint plugins for Bioclipse. Preferably, interaction can be scripted, allowing integration with other metabolomics features in Bioclipse.</p>

<br />
<p>Motivation: Bioclipse is a powerful platform combining a powerful scripting language to write data analysis and visualization workflows, which can easily be shared on, for example, MyExperiment.org [1]. While it has been used in drug discovery and predictive toxicology in pharmaceutical industry [2], it currently does not have functionality to visualize large graph structures as Cytoscape has.</p>

<br />
<p>Description: You will use Java to write a plugin for Bioclipse [3] that uses Cytoscape. No particular graph is aimed at, but the plugin will use the Eclipse selection event propagation mechanism [4] so that when users select protein and metabolites, Bioclipse will automatically visualize 2D or 3D structural information about those molecules (only the event creation has to be implemented). Preferably, a scripting language extension is written, to allow extracting information from the visualized graphs.</p>

<ol ><li><p><a href="http://myexperiment.org/" >myexperiment.org</a></p>

</li>
<li><p><a href="http://pubs.acs.org/doi/abs/10.1021/ci200242c" >pubs.acs.org/doi/abs/10.1021/ci200242c</a></p>

</li>
<li><p><a href="http://www.bioclipse.net/" >www.bioclipse.net</a></p>

</li>
<li><p><a href="http://www.eclipse.org/articles/Article-WorkbenchSelections/article.html" >www.eclipse.org/articles/Article-WorkbenchSelections/article.html</a></p>

</li>
</ol><br />
<a href="http://www.eclipse.org/articles/Article-WorkbenchSelections/article.html" ></a><p>Language and Skills: Java</p>

<p>Potential Mentors: Egon Willighagen, someone from Bioclipse team</p>

<p>Contact: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 17: Export as D3.js-Based Web Application</h2>
<p><b>Background</b>: From Cytoscape 3.2, users can <a href="http://wiki.cytoscape.org/Cytoscape_3/UserManual/Publish">export 
their network visualizations as a simple web application</a>. This feature uses <a href="http://js.cytoscape.org/">cytoscape.js</a> 
and other popular libraries such as <a href="https://angularjs.org/">Angular.js</a> to reproduce visualizations generated with 
Cytoscape 3 Desktop application using modern web technologies.</p>
<p><b>Goal</b>: This project is extending this idea. In addtion to exporting Cytoscape-generated visualizations as simple 
node-link diagrams, you will add new options for this feature, including:</p>
<ul>
<li>TreeMap</li>
<li>Dendrogram</li>
<li>Circle Packing</li>
<li>Node-Link Tree</li>
<li>Adj. Matrix</li>
</ul>
<p>This enables users to visualize the same data set in different ways. In this project, you need to extend the existing 
<a href="http://apps.cytoscape.org/apps/d3jsexporter">D3.js</a> Exporter App and add new export options. Note that you need 
both of Java and JavaScript knowledge to implement this feature. Also, we assume you are comfortable with modern web 
application development tools, such as command-line tools running on node.js, JavaScript frameworks like Angular.js, testing 
frameworks, and of course, <a href="http://d3js.org/">D3.js</a>.</p>
<p><b>Technology and Skills</b>: JavaScript, D3.js, Java, knowledge of modern web development tools such as node.js/grunt/gulp/yeoman.</p>
<p><b>Potential Mentors</b>: Kei Ono</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 18: BridgeDb config plugin in PathVisio</h2>

<p><b>Background</b>: <a href="http://www.bridgedb.org" >BridgeDb</a> is an identifier mapping framework used in PathVisio and WikiPathways. 
Currently it is possible to <a href="http://www.pathvisio.org/wiki/BridgeDbConfigPluginHelp" >load a mapping databases for gene products and metabolites</a>. 
This should be extended with interaction mapping databases.</p>
<p><b>Goal</b>: The existing <a href="http://www.pathvisio.org/plugin/bridgedbconfig-plugin/">BridgeDbConfig plugin</a> provides 
functionality to load more mapping databases or to use other webservices for the mappings. This functionality should be 
extended and integrated into the core PathVisio application as another core module. The configuration dialog has to be 
improved so it is more intuitive and user friendly. Furthermore the plugin should allow users to download and update BridgeDb 
mapping databases from within PathVisio. Currently the users have to download the databases first and then load them into 
PathVisio. This step should be automatized by the plugin.</p>

<p><b>Technology and Skills</b>: Java</p>
<p><b>Potential Mentors</b>:  Kutmon, Jonathan Melius</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com" >wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 19: Citing publications in biological pathways - PathVisio and WikiPathways</h2>

<p><b>Background</b>: <a href="http://www.pathvisio.org" >PathVisio</a> is a commonly-used pathway editor, visualization and analysis tool implemented in Java and is 
linked with the collaborative pathway database <a href="http://www.wikipathways.org" >WikiPathways</a>. Currently, PathVisio and WikiPathways 
allow users to add publication references to <a href="http://www.ncbi.nlm.nih.gov/pubmed">PubMed</a>.</p>
<p><b>Goal</b>: The goal of this project is to extend the functionality to allow the reference of other identifiers like DOIs, 
export references in standard formats like Bibtex or EndNote and provide statistics about citations used on WikiPathways 
(total number of citations, avg. number of citations per pathway, top cited papers/journals, ...).</p>

<p><b>Technology and Skills</b>: Java, PHP, Mediawiki</p>
<p><b>Potential Mentors</b>: Sravanthi Sinah, Martina Summer-Kutmon, Anwesha Bohler</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com" >wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 20: Enhanced Layout Algorithms</h2>
<p><b>Background</b>: As Cytoscape is used for increasingly large networks, the performance of our layout algorithms becomes 
a limiting factor. There are a number of layout algorithms that perform very well in real-world scenarios that have not been 
implemented in Cytoscape. For example, the <a href="http://yifanhu.net/PUB/graph_draw_small.pdf">Yifan Hu Multi-level layout 
algorithm</a>, a Fast Muiltipole Multilevel layout (See <a href="http://www.ogdf.net/doc-ogdf/classogdf_1_1_f_m_m_m_layout.html">OGDF</a> 
for example), Yifan Hu proportional, and potentially some of the better algorithms from Gephi. Cytoscape provides a 
framework for adding new layout algorithms as Apps, so this work could all be done as a series of separate apps.</p>

<p><b>Technology and Skills</b>: </p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 21: Compound Graph Visualization in Cytoscape 3</h2>

<p><b>Background</b>: Cytoscape 3 has introduced two concepts into the core, groups and annotations, that could be combined to provide a compound 
graph visualization. Currently groups can be either in a “collapsed” or “expanded” state. 
<p><b>Goal</b>: </p>For a compound graph, a new visualization state needs to be introduced that allows the compound node 
(group) to be visualized as a shape (concave or 
convex hull) around it’s children. Moving that shape would move all of it’s children, etc.</p>

<p><b>Technology and Skills</b>: Java</p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 22: Visualization of Time-series Cluster Data</h2>

<p><b>Background</b>: Currently, the clusterMaker plugin uses standard heat maps to visualize attribute cluster data (based on Java TreeView). 
Increasingly, time series data is becoming important for clustering applications and it would be useful to expand the provided
visualization options to explicitly support time series data. For these purposes, we’re thinking about time-series that have 
multiple conditions in each series, which means we need to move beyond the standard two-dimensional heat map. 
<p><b>Goal</b>: </p>This project 
can explore utilizing animation, small multiples, or other standard visualization techniques (or your own novel techniques) 
to provide new visualization opportunities to users. Ideally, this would be integrated with network-based animation tools 
such as DynNetwork or CyAnimator.</p>

<p><b>Technology and Skills</b>: Java</p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com" >cytoscape-discuss@googlegroups.com</a></p>

<br />

<h2 dir="ltr" >Idea 23: Develop a Converter between Visual and Textual Representations of Biological Pathways</h2>

<p><b>Goal</b>: <a href="http://www.pathwaycommons.org/">Pathway Commons</a> seeks to aggregate all pathway data available in literature, public databases, and 
curation/crowdsourced efforts. Our goal is to accelerate this process by leveraging pathway diagram editors that support a 
widely used visual representation for biological pathways.</p>
<p><b>Problem</b>: Pathway Commons data is represented in the <a href="http://www.ncbi.nlm.nih.gov/pubmed/20829833">Biological Pathway Exchange (BioPAX)</a> 
community standard format, 
but the BioPAX format lacks a simple graphical editor. On the other hand, the System Biology Graphical Notation (SBGN) is a 
visual representation for pathways that is supported in about 20 tools. We'd like you to build a converter between the BioPAX 
and SBGN formats.</p>
<p>A similar project has been developed with the editor CellDesigner (that shares a format similar to SBGN) making us believe 
that this should be a reasonably straight forward project; many of the mapping issues have already been addressed. The project 
will make use of two Java libraries <a href="http://www.biopax.org/paxtools.php">Paxtools</a> (for BioPAX) and 
<a href="http://www.sbgn.org/LibSBGN">LibSBGN</a> (for SBGN).</p

<p><b>Technology and Skills</b>: Java and XML</p>
<p><b>Potential Mentors</b>: Augustin Luna</p>
<p><b>Contact</b>: <a href="mailto:lunaa@cbio.mskcc.org" >Augustin Luna</a></p>

<br />

<h2 dir="ltr" >Idea 24: Touch-based Exploration of Large Data Collections and Networks</h2>

<p>The goal of this project is to develop a general, open source framework for performing iterative, faceted search on large data collections and networks specifically for touch-based input devices. The project will leverage ideas from existing desktop applications (see<a href="http://www.youtube.com/watch?v=BZuFUZpEZ-A" > Microsoft Pivot</a>) but will include experimentation and innovation in interaction techniques in order to create a more engaging user experience. Ultimately the framework will be applied in one of potentially many biological applications, including MedSavant, which is described below.</p>

<p>MedSavant is a high-performance search engine for genomic datasets, which is primarily aimed to help researchers identify genetic conditions that cause disease. The desktop client allows users to</p>

<p>incrementally construct faceted search queries and visualize their results virtually in realtime. Similar workflows enabled by the desktop client will be made possible by integration with the proposed</p>

<p>framework. For more information, visit <a href="http://www.medsavant.com" >www.medsavant.com</a>.</p>

<br />
<p>Language and Skills: Android or iOS development; AJAX, JSON, Java, and Javascript; UNIX</p>
<p>Potential Mentors: Marc Fiume, Marta Girdea</p>
<p>Contact: <a href="mailto:mfiume@cs.toronto.edu" >mfiume@cs.toronto.edu</a></p>
<br />
<h2 dir="ltr" >Idea 25: Improve the Text-Mining Functionality of Factoid</h2>

<p><b>Background</b>: <a href="http://pathwaycommons.github.io/factoid/" >Factoid</a> aims to be a tool that helps authors 
annotate biological interactions that occur in the text of academic publications in a machine-readable manner. Currently, 
Factoid uses basic entity-recognition to identify genes, proteins, biological processes to identify interactions between 
these entities. 
<p><b>Goal</b>: The goal of this project is to integrate more sophisticated algorithms into the extraction of 
interactions from text content. A starting point for some of these would be the algorithms outlined by the EBI website 
WhatIzIt.</p>
<p><b>Technology and Skills</b>: Experience connecting web services, Javascript</p>
<p><b>Potential Mentors</b>: Augustin Luna (MSKCC), Max Franz (U of Toronto) </p>
<p><b>Contact</b>: <a href="mailto:lunaa@cbio.mskcc.org" >Augustin Luna</a>, <a href="mailto:maxkfranz@gmail.com" >Max Franz</a></p>
<br />

<h2 dir="ltr">Idea 26: Optimize Semi-supervised Learning Algorithm for Genomic Datasets and Inject Prior Knowledge from 
Biological Pathways</h2>
<p><b>Background</b>: A major problem in bioinformatics is that the number of features collected by modern high-throughput 
technologies severely overwhelms the number of samples for which such features are available. In a typical dataset, it is not 
uncommon to have tens of thousands of molecular measurements, but only a few tens or hundreds of samples. This severe 
undersampling creates a significant challenge for researchers who want to mine such datasets for biological or clinical 
insight. For example, when looking to predict sensitivity of cancer cells to various compounds, it is not uncommon to come up 
with a collection of molecular features that are highly predictive on a given dataset, only to have the accuracy fail to 
generalize to new, previously unseen samples.</p>
<p>A natural way to approach the issue is to bring in additional samples to the dataset, e.g., other cancer cells from the same 
tissue in the compound sensitivity example above. However, the phenotypic information or "labels" for such samples may not be 
readily available. Semi-supervised learning is a branch of machine learning that deals with constructing predictive models 
using both labeled and unlabeled data. It is particularly well-suited to biological applications, because while there is 
typically only a small handful of labeled samples, the unlabeled data is aplenty. Furthermore, we can expect some shared 
underlying biology between labeled and unlabeled data, due the fact that molecular measurements capture some underlying 
activity of various signaling and metabolic pathways in a cell. Semi-supervised learning can tap into the vast amount of 
unlabeled data to uncover such pathways and then utilize the labeled samples to correlate these pathways to phenotypic 
outcomes.</p>
<p><b>Goal</b>: To implement a specific semi-supervised learning algorithm and to work with the mentor on modifying this 
algorithm to tailor to biological data. The algorithm was originally proposed by Ando & Zhang [1] and operates by 
constructing multiple predictive tasks from the unlabeled data and learning across all those tasks simultaneously. The 
intuition behind the algorithm is that, if a molecular feature X is highly correlated with the phenotype F, then learning to 
predict X is just as valuable as learning to predict F directly. The advantage of predicting X is the fact that it's observed 
across all data (both labeled and unlabeled), while the measurements of F are, by definition, available for the labeled data 
only. Continuing with the example above, if a compound targets the epidermal growth factor receptor (EGFR) in a cell, then 
predicting whether EGFR is highly expressed is a task that should be highly related to directly predicting the sensitivity of 
the cell to that compound.</p>
<p>To simultaneously learn across multiple tasks, Ando & Zhang's algorithm makes an assumption that the tasks share some 
underlying lower-dimensional representation. This provides a perfect place to inject prior biological knowledge about 
feature-feature relationships. Specifically, a sub-goal of the project is to modify the lower-dimensional constraints in the 
algorithm to align with known information about biological pathways, as taken from web resources, such as WikiPathways, 
BIOCARTA, PathwayCommons, etc.</p>
<p>The basic implementation of the algorithm is already in place. It remains to optimize the implementation for handling 
full-scale genomic datasets (~20k features and thousands of samples) by moving parts of the code from R to C / C++, and to 
implement the pathway-based component discussed above.</p>
<p>[1] <a href="http://www.jmlr.org/papers/volume6/ando05a/ando05a.pdf">Ando, R. K., & Zhang, T. (2005). A framework for learning predictive structures from multiple tasks and unlabeled data. The 
Journal of Machine Learning Research, 6, 1817-1853.</a></p>
<p>Technology and Skills: R, C / C++, Linear Algebra</p>
<p>Potential Mentors: Artem Sokolov (UCSC)</p>
<p>Contact: <a href="mailto:sokolov@soe.ucsc.edu">Artem Sokolov</a></p>

<br />
<h2 dir="ltr">Idea 27: Developing Web Interface For The Online Resource for Interface- Based Interaction Networks (TOR-IBIN) 
Using Cytoscape.js</h2>
<p><b>Background</b>: The protein-protein interaction (PPI) is a physical contact between two proteins, or more, resulted from 
biochemical reaction or electrostatic forces. The interaction usually takes place through an Interface, which is part of the 
protein (e.g. protein domain) with special molecular structure that enables and mediates the interaction between proteins. 
Interface-based PPI predictions help in understanding the mechanisms of interactions between proteins as well as helps in 
deciphering the functional roles of the proteins.</p>
<p><b>Focus</b>: TOR-IBIN is an online resource for gathering and storing the published interface-based (IB) protein-protein 
interaction networks (IN) predicted using any PPI prediction method. TOR-IBIN aims to make these networks available in a 
central repository with unified format and comprehensive search functions that maximize the use of these data. TOR-IBIN 
web-based interface aims to implement the browse and search functions and well as network visualization using Cytoscape.JS and 
network download in several formats including JSON and tab-delimited formats. TOR-IBIN is a relational database implemented in 
MySQL.
- The web interface and functionality will be similar to the web interface of the cancer variants database (Can-VD) 
canvd.baderlab.org.
</p>
<p><b>Goals</b>:
<ol>
<li>Develop a web-based interface for interface-based PPI data retrieval with a comprehensive search functions and data 
download options.</li>
<li>Develop a visualization and annotation module using Cytoscape.js to visualize the retrieved PPI data as a network with 
annotations that highlights the features of the predicted PPI such as the prediction confidence and validation...etc.</li>
</ol></p>

<p><b>Inputs</b>: The student will be provided with the database design and sample data as well as an introduction to the 
topic and the different possible visualizations and annotations needed. The student will have the freedom to choose among 
them or to propose other options.</p>

<p><b>Technology and Skills</b>: Javascript, JSON, PHP, JQuery</p>
<p><b>Potential Mentors</b>: Mohamed Helmy, Gary Bader</p>
<p><b>Contact</b>: <a href="mailto:mohamed.attiashahata@utoronto.ca">Mohamed Helmy</a>, <a href="mailto:gary.bader@utoronto.ca">Gary Bader</a></p>

<br />
<h2 dir="ltr">Idea 28: Full Compound Support in Cytoscape 3</h2>
<p><b>Background</b>: The notion of compound graphs is used to represent more complex types of relationships or varying levels 
of abstractions in data visualization. Networks from almost all domains need such abstractions and nesting, including 
biological networks. The recently developed standard notation of <a href="http://www.sbgn.org/">SBGN</a> makes heavy use of such structures for molecular 
complexes and cellular locations.</p>
<p><b>Goal</b>: We would like to first implement full compound support in <a href="http://www.cytoscape.org/">Cytoscape</a> 3 [2] core. This will not only allow users 
to nest nodes of a network but also automatically adjust size of a compound node as its child nodes get  moved, resized or  
deleted. Then, we'd like to implement a native spring embedder based layout algorithm named CoSE [1] that takes such structures 
into account during layout. A Java implementation of CoSE is already publicly available [2] and needs to be integrated into 
Cytoscape 3.</p>
<p>[1] U. Dogrusoz, E. Giral, A. Cetintas, A. Civril, and E. Demir, "A Layout Algorithm For Undirected Compound Graphs", 
Information Sciences, 179, pp. 980-994, 2009. <a href="http://www.cs.bilkent.edu.tr/~ivis/layout/cose-animated-demo/cose.html">
     http://www.cs.bilkent.edu.tr/~ivis/layout/cose-animated-demo/cose.html</a></p>
<p>[2] <a href="http://www.cs.bilkent.edu.tr/~ivis/chilay.html">http://www.cs.bilkent.edu.tr/~ivis/chilay.html</a></p>

<p><b>Technology and Skills</b>: Java, Data Structures and Algorithms</p>
<p><b>Potential Mentors</b>: Ugur Dogrusoz (Bilkent University & MSKCC) and Gerardo Huck (U of Toronto)</p>
<p><b>Contact</b>: <a href="mailto:ugur@cs.bilkent.edu.tr">ugur@cs.bilkent.edu.tr</a>, <a href="mailto:onur@cbio.mskcc.org">onur@cbio.mskcc.org</a></p>
<br />

<h2 dir="ltr">Idea 29: Extending and improving PathVisio Data Import</h2>
<p><b>Background</b>: <a href="http://pathvisio.org">PathVisio</a> is a pathway editor and analysis tool. PathVisio allows users to upload and visualize data on 
biological pathways. Currently, it is possible to upload only one dataset at a time as a comma separated file. </p>
<p><b>Goal</b>: This functionality should be extended to allow uploading a number of data files at the same time as well 
as uploading of multiple data file formats (for eg. Excel sheets) should be allowed.
This will involve extending the <a href="http://developers.pathvisio.org/browser/trunk/modules/org.pathvisio.gex">gex</a> and 
<a href="http://developers.pathvisio.org/browser/trunk/modules/org.pathvisio.data">data</a> modules  of PathVisio. These modules are part of the PathVisio core and take care 
of importing experimental data and mapping it to the elements in a pathway. </p>
<p>Features of the new plugin:
<ul>
<li>Uploading multiple datasets at once : these could be any kind of quantitative or qualitative data about any pathway 
element, such as genes, proteins, metabolites, and interactions.</li>
<li>Linking each individual dataset to a particular element on a pathway, for eg. linking a gene expression dataset to genes 
on pathways and a flux data set to interactions on pathways.</li>
<li>Importing data from multiple file formats (For eg: .xls, .ods).</li>
<li>Create a pop-up panel accessible from the menubar of the PathVisio main window to show:
     <ul>
          <li>all the currently loaded datasets</li>
          <li>the biological elements they are linked to</li>
          <li>the currently loaded annotation databases</li>
     </ul>
</ul></p>
<p><b>Technology and Skills</b>: Java</p>
<p><b>Potential Mentors</b>: Anwesha Bohler, Martina Summer-Kutmon</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>

<br />
<h2 dir="ltr">Idea 30: Collaborative System for Creating/Sharing Biological Icons</h2>
<p><b>Background</b>: <a href="http://wikipathways.org/index.php/WikiPathways:About">Biological pathways</a> provide intuitive 
diagrams of the interactions underlying biological processes. 
For example, the <a href="http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP706&label[]=L-DOPA&label[]=Dopamine&label[]=DDC&colors=yellow">SIDS Susceptibility Pathways</a> shows interactions between metabolites, genes and proteins, such as DDC 
catalyzing the conversion of L-DOPA to Dopamine (look for the items highlighted in yellow on the lower right). The site 
<a href="http://www.wikipathways.org">WikiPathways</a> is an open, collaborative platform dedicated to drawing and sharing biological pathways. These pathways often 
use icons to represent items like mitochondria or cardiac myocytes (heart muscle cells). The WikiPathways drawing editor has 
a limited set of icons built in, but it does not allow pathway authors to create and re-use custom icons. For example, the 
cardiac myocyte icon is repeated in the SIDS Susceptibility Pathways for <a href="http://wikipathways.org/index.php/Pathway:WP1266">mouse</a>
and for <a href="http://wikipathways.org/index.php/Pathway:WP706">human</a>, but the only way to re-use the 
icon is to manually redraw or copy/paste it. Updates to the icon in pathway do not appear in the other.</p>
<p><b>Goal</b>: Build a biological icon library that allows pathway authors to create icons, share them with other 
WikiPathways’ users and reuse them across pathways. It could work like Github for icons, allowing users to create, fork and 
merge their work, with the best icons rising to the top via a rating system. The library will work as a plugin for the current 
WikiPathways viewer/editor <a href="https://github.com/wikipathways/pathvisiojs">pathvisiojs</a>, allowing pathway authors to create new icons from scratch or to create icons from 
sections of existing pathways, e.g., the cardiac myocyte from the previously referenced pathways. It will have a backend 
component to store the icons and a front-end component to allow WikiPathways users to view and search for icons.</p>
<p><b>Links</b>: An <a href="https://code.google.com/p/svg-edit/">SVG drawing tool </a>with a large number of icons available might provide some ideas. 
</p>
<p><b>Technology and Skills</b>: JavaScript required. Some backend experience (PHP, Node.js or Python) to create a backend for saving the 
icons. SVG or Canvas a plus.
</p>
<p><b>Potential Mentors</b>: Anders Riutta, Alex Pico, Kristina Hanspers</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr">Idea 31: Biological Pathway Styling Themes</h2>
<p><b>Background</b>: Description: <a href="http://wikipathways.org/index.php/WikiPathways:About">Biological pathways</a> provide intuitive diagrams of the interactions underlying biological 
processes. For example, the <a href="http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP706&label[]=L-DOPA&label[]=Dopamine&label[]=DDC&colors=yellow">SIDS Susceptibility Pathways</a>
shows interactions between metabolites, genes and proteins, such as 
DDC catalyzing the conversion of L-DOPA to Dopamine (look for the items highlighted in yellow on the lower right). The site 
<a href="http://www.wikipathways.org">WikiPathways</a> is an open, collaborative platform dedicated to drawing and sharing biological pathways. These diagrams are viewed 
in many different environments, such as laptop screens, presentation slides and printed science journals. At WikiPathways today, 
the default styling theme for these diagrams is best suited for viewing by an individual on a computer screen, but it would be 
useful to export the diagrams with themes optimized for print and for presentations.</p>
<p><b>Goal</b>: Develop a custom theming system to allow for exporting pathways with themes optimized for print and for 
projected presentation slides. For example, the presentation export option might make pathway diagrams easier to read by 
removing some text, enlarging the remaining text and inverting the colors of the diagram. Optionally, the presentation option 
could integrate with one or more existing online presentation framework(s).</p>
<p><b>Links</b>:
<ul>
<li><a href="https://github.com/wikipathways/pathvisiojs">WikiPathways pathway viewer/editor</a></li>
<li>Example online presentation frameworks:</li>
<ul>
     <li><a href="https://github.com/hakimel/reveal.js">github.com/hakimel/reveal.js</a></li>
     <li><a href="https://github.com/bartaz/impress.js">github.com/bartaz/impress.js</a></li>
     <li><a href="https://docs.google.com/presentation">docs.google.com/presentation</a></li>
</ul>
</ul>
</p>
<p><b>Technology and Skills</b>: User experience, graphical design, CSS, some JavaScript.</p>
<p><b>Potential Mentors</b>: Anders Riutta, Alex Pico, Kristina Hanspers</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>
<br />
<h2 dir="ltr" >Idea 32: Details Panel for Semantic Terms</h2>
<p><b>Background</b>: <a href="http://wikipathways.org/index.php/WikiPathways:About">Biological pathways</a> provide intuitive 
diagrams of the interactions underlying biological processes. For 
example, the <a href="http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP706&label[]=L-DOPA&label[]=Dopamine&label[]=DDC&colors=yellow">SIDS Susceptibility Pathways</a> 
show interactions between metabolites, genes and proteins, such as DDC catalyzing 
the conversion of L-DOPA to Dopamine (look for the items highlighted in yellow on the lower right). The site <a href="http://www.wikipathways.org">WikiPathways</a>
is an open, collaborative platform dedicated to drawing and sharing biological pathways.</p>
<p>A common challenge in biological research is dealing with an ever-growing number of terms, some of them 
synonyms and others almost synonyms but with important differences in meaning. For instance, WikiPathways uses the term 
“GeneProduct” in its annotation panels (to see an example, click the “TNF” box in the upper left corner of the SIDS pathway). 
A WikiPathways user might be unsure of the exact definition of the “GeneProduct”, so it would be useful to point the user to 
the <a href="http://vocabularies.wikipathways.org/wp#GeneProduct">definition of the term</a> in the WikiPathways vocabulary. This could be done with a link or a mouseover display. Another 
example from the SIDS pathway is the Extracellular Space, which could display information from this IRI: 
<a href="http://identifiers.org/go/GO:0005615">href="http://identifiers.org/go/GO:0005615"</a>.</p>
<p>The first two examples are easy to display because each IRI has an HTML representation, but a more challenging example is 
Protein: <a href="http://www.biopax.org/release/biopax-level3.owl#Protein">http://www.biopax.org/release/biopax-level3.owl#Protein</a>. 
This IRI returns RDF but does not have an HTML 
representation, so it would be necessary to extract information from the RDF, such as the definition, and display it in a 
small, easy-to-read panel.</p>
<p><b>Goal</b>: Develop a browser-based display system for dereferenced IRIs. The system will use content-type negotiation to 
request an HTML representation of a resource, but it will be able to convert JSON-LD, RDF and OWL representations into an HTML 
representation for display in a small details panel or pop-over. Additionally, develop a plugin for the WikiPathways 
viewer/editor <a href="https://github.com/wikipathways/pathvisiojs">pathvisiojs</a> that integrates the display system. Optionally, if time allows, add functionality to allow 
pathvisiojs users to switch vocabularies for the GUI display, mapping from the built-in vocabularies to the user-preferred 
vocabularies with <a href="http://www.w3.org/TR/skos-primer/">SKOS</a> files.</p>
<p><b>Links</b>:
<ul>
<li><a href="https://semantic-mediawiki.org/">Project with related functionality</a></li>
<li><a href="https://github.com/alexmilowski/green-turtle">An RDFa implementation for browsers</a></li>
<li><a href="https://code.google.com/p/ubiquity-rdfa/wiki/TutorialIntroduction">Possibly relevant project</a></li>
</ul></p>
<p><b>Technology and Skills</b>: JavaScript, <a href="http://json-ld.org/">JSON-LD</a>, RDF, OWL, optionally SKOS</p>
<p><b>Potential Mentors</b>: Anders Riutta, Alex Pico, Kristina Hanspers</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-discuss@googlegroups.com" >wikipathways-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 33: Principal Component Analysis</h2>
<p><b>Background</b>: One of the common multidimensional analysis tools is principal component analysis. At this point, there 
is no good PCA tool integrated with Cytoscape.  The ideal tool would support performing the analysis, visualizing the results 
of the analysis using standard approaches, and then visualizing the results in the context of the Cytoscape network itself.
There are many tools that could assist in this task and the analysis itself is pretty well understood. The challenge to this 
idea is looking for ways to integrate the results of the analysis into Cytoscape.</p>
<p><b>Technology and Skills</b>: </p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 34: Incremental Loading of Networks</h2>
<p><b>Background</b>: Cytoscape now supports using JSON as a format for loading networks. In an effort to improve exploration 
of extremely large networks, it would be extremely useful to be able to load parts of the network on request.  One use case 
for this is loading networks from the Structure-Function Linkage Database, which creates Cytoscape networks representing 
similarity between groups of proteins.  Some of these networks are extremely large (millions of edges) and usually users are 
only interested in local neighborhoods of the network.  This project would involve writing a Cytoscape App that would load an 
abstraction of these networks (where multiple nodes are represented by a single "meta" node) into Cytoscape and then allow 
users to incrementally load all of the nodes within that metanode.</p>
<p><b>Technology and Skills</b>: </p>
<p><b>Potential Mentors</b>: Scooter Morris, Samad Lotia</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 35: Add t-SNE to clusterMaker</h2>
<p><b>Background</b>: t-SNE [1] is a technique for dimensionality reduction that is widely used.  
It would be extremely useful to add this algorithm into the existing clusterMaker app.</p>
<p>[1] <a href="http://lvdmaaten.github.io/tsne/">t-SNE</a></p>
<p><b>Technology and Skills</b>: </p>
<p><b>Potential Mentors</b>: Scooter Morris</p>
<p><b>Contact</b>: <a href="mailto:scooter@cgl.ucsf.edu">Scooter Morris</a></p>
<br />

<h2 dir="ltr" >Idea 36: cyRecommender - New Cytoscape 3 plugin for recommending network ontologies</h2>
<p><b>Background</b>: Cytoscape 3 is a powerful software for networks that provides diverse plugins in order to combine 
external knowledge with the network connectivity. There are a few plugins that allow to link ontology details with node labels. 
However, existing plugins require the user to decide what ontology or ontologies are best for each network, which may be a 
hard and time‑consuming task. The NCBO Recommender is a service developed by Stanford University that has been designed to 
recommend the most appropriate ontologies for a corpus or a list of keywords. Integrating the NCBO Recommender with Cytoscape 
will make it possible to automatically obtain the best ontological representation for a set of network nodes. The NCBO is 
developing a new version of the Recommender, which will be available in a few months. The proposed Cytoscape plugin will be 
one of the first tools using the new version of the Recommender.</p>
<p><b>Goal</b>:  To integrate the NCBO Recommender service into a new Cytoscape plugin. After selecting a set of network nodes, 
the Recommender will be invoked to obtain the best ontologies to represent the node labels, and the network will be presented 
using the recommended ontologies. The integration will consist on an easy to use graphical interface for any level of ontology
experience.</p>
<p><b>Description</b>: You will use Java to write an app for Cytoscape 3 [1] that will access the NCBO Recommender REST services. 
Your tool will make it possible to:
<ul>
<li>Get the keywords from the selected node labels, send them to NCBO Recommender[2] REST services [3], and obtain the recommended 
ontologies (the selection could be a subnetwork or the entire network).</li>
<li>Visualize the network using the recommended ontologies.</li>
<li>Save the recommendations as CSV or TAB-separated files.</li>
<li>Save the network with the new added ontology information for each node.</li>
</ul>
This work involves programming a new plugin that will interact with Cytoscape in the node selection, network plotting, etc. 
The code will be available as an open source GitHub project.</p>

<p>[1] <a href="http://www.cytoscape.org/cy3.html">Cytoscape 3</a></p>
<p>[2] <a href="http://bioportal.bioontology.org/recommender">NCBO Recommender</a></p>
<p>[3] <a href="http://data.bioontology.org/documentation">NCBO REST services</a></p>
<p><b>Technology and Skills</b>: Java, GitHub</p>
<p><b>Potential Mentors</b>: Cristian R Munteanu, Marcos Martinez, Alejandro Pazos</p>
<p><b>Contact</b>: <a href="mailto:muntisa@gmail.com">Cristian R Munteanu</a></p>
<br />

<h2 dir="ltr" >Idea 37: Advanced search on WikiPathways</h2>
<p><b>Background</b>: WikiPathways [1] is a collaborative pathway database that is build on Mediawiki. 
Currently our pathways can be annotated with ontology tags from the pathway ontology, cell type ontology and disease ontology.</p> 
<p><b>Goal</b>: The advanced search should provide an interface to search for pathways containing one or more pathway 
elements (genes, proteins, metabolites), for pathways that are annotated with specific ontology tags or pathways that contain 
a specific interactions. The advanced search allows users to filter pathways based on species, curation tags, authors, 
creation date, number of edits, etc.</p>
<p>[1] <a href="http://www.wikipathways.org">WikiPathways</a></p>

<p><b>Technology and Skills</b>: PHP, Mediawiki</p>
<p><b>Potential Mentors</b>: Martina Summer-Kutmon, Alex Pico</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 38: Over-representation module for WikiPathways</h2>
<p><b>Background</b>: <a href="http://www.pathvisio.org">PathVisio</a> is a commonly-used pathway editor, visualization and analysis tool 
implemented in Java. Currently, PathVisio provides an advanced statistics module that allows users to define criterias 
to perform pathway statistics which is sometimes not needed.</p>
<p><b>Goal</b>: In this project, the student will extend the statistics module with a basic over-representation methods 
that uses a gene list as input and provides ranked pathways as output. The genes in the pathways are highlighted when they 
are in the input gene list. Additionally this basic analysis module will also be integrated on <a href="http://www.wikipathways.org">WikiPathways</a>. The student 
will develop a page where the user can provide a list of genes, then the PathVisio 
statistics module gets called to calculate the over-representation and the result will be shown on WikiPathways. The pathways 
will be shown in the pathvisiojs viewer and the genes in the input list will be highlighted.</p>

<p><b>Technology and Skills</b>: Java, PHP</p>
<p><b>Potential Mentors</b>: Martina Summer-Kutmon, Anwesha Bohler</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br/ >

<h2 dir="ltr" >Idea 39: Tutorial mode for PathVisio</h2>
<p><b>Background</b>: PathVisio [1] is a commonly-used pathway editor, visualization and analysis tool 
implemented in Java. The analyses in PathVisio and the different advanced visualization methods can become very complex 
and having an integrated tutorial mode in the application is a powerful tool to give detailed instructions while performing 
a task.</p>
<p><b>Goal</b>: This will give the users a step-by-step guide with hints and additional information while learning how to use 
PathVisio. Documentation and tutorials are crucial elements of software development and the goal of this project is to 
implement a tutorial mode in PathVisio. PathVisio can be extended through plugins, so the tutorial mode should allow other 
modules to provide additional tutorials that can be selected if the plugins are installed.</p>
<p>[1] <a href="http://www.pathvisio.org">PathVisio</a></p>
<p><b>Technology and Skills</b>: Java, development of tutorials</p>
<p><b>Potential Mentors</b>: Martina Summer-Kutmon, Kristina Hanspers, Sravanthi Sinah</p>
<p><b>Contact</b>: <a href="mailto:wikipathways-devel@googlegroups.com">wikipathways-devel@googlegroups.com</a></p>
<br/ >

<h2 dir="ltr" >Idea 40: SIR model for CentiScaPe</h2>
<p><b>Background</b>: CentiScaPe is a Cytoscape plugin that allows to compute network centrality parameters. It is the most 
complete Cytoscape plugin for network centralities calculating Degree, Average Shortest Path, Eccentricity, Closeness, 
Betweenness, Centroid, Stress, Radiality, Eigenvector, Edge Betweenness, Bridging Centrality, for directed and undirected 
networks. CentiScaPe allows identifying network nodes that are relevant from both experimental and topological viewpoints. 
CentiScaPe also provides a Boolean logic- based tool that allows easy characterization of nodes whose topological relevance 
depends on more than one centrality. Finally, different graphic outputs and the included description of biological 
significance for each computed centrality facilitate the analysis by the end users not expert in graph theory, thus allowing 
easy node categorization and experimental prioritization.</p>
<p><b>Goal</b>: CentiScaPe can provide only static analysis of a network. We want to develop a tool that can compute 
semi-dynamic simulation, particularly the SIR model [1]. The SIR model is a Compartmental model used in epidemiology to 
simulate the establishment and spread of infectious diseases. It serve as a base mathematical framework for understanding the 
complex dynamics of such system. Population is divided in three states: susceptible to the infection of the pathogen (often 
denoted by S), infected by the pathogen (given the symbol I) and  recovered/removed/immune  (denoted R). The way that these 
compartments interact is often based on a network model, hence the idea to include the SIR model in the CentiScaPe tool for 
network analysis</p>
<p>[1] <a href="http://www.ncbi.nlm.nih.gov/pubmed/2059741">Kermack WO, McKendrick AG: A contribution to the mathematical theory of epidemics.
Proc Roy Soc Lond 1927, 115:700-721.</a></p>
<p><b>Technology and Skills</b>: Java</p>
<p><b>Potential Mentors</b>: Giovanni Scardoni (CBMC, University of Verona)</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<h2 dir="ltr" >Idea 41: Random networks for CentiScaPe</h2>
<p><b>Background</b>: CentScaPe is a Cytoscape app that allows, by using a very plain interface, to compute different network 
centrality parameters in single and in multiple networks at the same time. It allows to plot the results by using real and 
normalized values in different plots. Several centralities are available: Degree, Average Shortest Path, Eccentricity, 
Closeness, Betweenness, Centroid, Stress, Radiality, Eigenvector, Edge Betweenness, Bridging Centrality. Each one could be 
computed for directed and undirected networks; CentiScaPe gives the possibility to assign a weigth to the edges and, currently, 
we added a new feature that permits to assign a weight to each node by using an attribute.</p>
<p><b>Goal</b>: The new version permits an in-depth analysis of biological networks, from the topological point of view. The 
main issue with this methodology is that it requires a lot of literature mining in order to validate and verify the results 
that, in general, remain speculative. In order to overcome this limitation we decided to add a feature that allows to validate
the results by using random networks, i.e. by adding a sort of statistical evaluation. By randomizing a network we expect to 
highlight all the differences between the real and the random process[1]: this step gives strength to all the findings that 
come from a topological analysis and adds a statistical background that could integrate the biological information from 
experimental data or text mining.</p>
<p>We will integrate CentiScaPe with some algorithms that were already developed in an existing, but no more supported, plugin 
for the Cytoscape platform, version 2.6, that is RandomNetworks [2][3]. We will develop some algorithms that are able to generate 
newtorks by following a specific model: Watt-Strogatz, Erdos-Renyi, Barabasi-Albert and some others. Another interesting 
feature will be the degree-preserving algorithm that is able to randomize the edges by keeping the degree of each node fixed. 
Finally the comparison between the centralities before and after the randomization could be analysed by using the topological 
measures of CentiScaPe.</p>
<p>[1] <a href="http://www.ncbi.nlm.nih.gov/pubmed/12399590">Network Motifs: Simple Building Blocks of Complex Networks. Milo et al, Science 25 October 2002.</a></p>
<p>[2] <a href="http://apps.cytoscape.org/apps/randomnetworks">RandomNetworks App</a></p>
<p>[3] <a href="https://sites.google.com/site/randomnetworkplugin/">RandomNetworks Project Website</a></p>

<p><b>Technology and Skills</b>: Java</p>
<p><b>Potential Mentors</b>: Giovanni Scardoni (CBMC, University of Verona)</p>
<p><b>Contact</b>: <a href="mailto:cytoscape-discuss@googlegroups.com">cytoscape-discuss@googlegroups.com</a></p>
<br />

<!-- END PASTE -->

    </div> <!-- .span9 !-->
</div> <!-- .row !-->

<script src="../assets/js/jquery-1.11.0.min.js"></script>
<script src="../assets/js/jquery-ui.min.js"></script>
<script src="../assets/js/jquery.tocify.min.js"></script>
<script>
$("#nav").tocify({
    context: "#ideas",
    showAndHide: false,
    selectors: "h2",
});
</script>
</body>
</html>
